<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.3 Assessing clonal abundance | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</title>
  <meta name="description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  <meta name="generator" content="bookdown 0.18.3 and GitBook 2.6.7" />

  <meta property="og:title" content="3.3 Assessing clonal abundance | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.3 Assessing clonal abundance | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  
  <meta name="twitter:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

<meta name="author" content="Robert Bowman" />


<meta name="date" content="2020-05-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="post-processing.html"/>
<link rel="next" href="cohort-description.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreward</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="data-extraction-and-setup.html"><a href="data-extraction-and-setup.html"><i class="fa fa-check"></i><b>3</b> Data Extraction and Setup</a><ul>
<li class="chapter" data-level="3.1" data-path="tapestri-library.html"><a href="tapestri-library.html"><i class="fa fa-check"></i><b>3.1</b> Tapestri Package</a></li>
<li class="chapter" data-level="3.2" data-path="post-processing.html"><a href="post-processing.html"><i class="fa fa-check"></i><b>3.2</b> Post processing</a></li>
<li class="chapter" data-level="3.3" data-path="assessing-clonal-abundance.html"><a href="assessing-clonal-abundance.html"><i class="fa fa-check"></i><b>3.3</b> Assessing clonal abundance</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="cohort-description.html"><a href="cohort-description.html"><i class="fa fa-check"></i><b>4</b> Cohort Description</a></li>
<li class="chapter" data-level="5" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>5</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="Appendix.html"><a href="Appendix.html"><i class="fa fa-check"></i><b>6</b> Appendix</a><ul>
<li class="chapter" data-level="6.1" data-path="AppendixA.html"><a href="AppendixA.html"><i class="fa fa-check"></i><b>6.1</b> Import from Tapestri Insights</a></li>
<li class="chapter" data-level="6.2" data-path="functions.html"><a href="functions.html"><i class="fa fa-check"></i><b>6.2</b> Functions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="assessing-clonal-abundance" class="section level2">
<h2><span class="header-section-number">3.3</span> Assessing clonal abundance</h2>
<p>This will be the last code example that proceeds with our limited sample set. After this, the tutorial will procede with the data used in the manuscript. This analysis will focus only on samples that have greater than one mutation.</p>
<p>The general workflow is as follows * Select samples that have at least 2 mutaions * Clone assigment * Tally clonal abundance * Establish a clonal abundance confidence itnerval</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Select samples with at least 2 mutations </span>
clonal_sample_set &lt;-<span class="st"> </span><span class="kw">names</span>(final_NGTs)[<span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(final_NGTs,dim))[,<span class="dv">2</span>]<span class="op">&gt;</span><span class="dv">2</span>]

<span class="co">#Order columns based on computed_VAF, and assign a clone to each cell</span>
NGT_to_clone&lt;-<span class="kw">lapply</span>(final_NGTs[clonal_sample_set],<span class="cf">function</span>(y){
  bulk_VAF_order &lt;-<span class="kw">names</span>(<span class="kw">sort</span>(<span class="kw">colSums</span>(y[,<span class="op">-</span><span class="dv">1</span>]),<span class="dt">decreasing=</span><span class="ot">TRUE</span>))
  y[,<span class="kw">c</span>(<span class="st">&quot;Cell&quot;</span>,bulk_VAF_order)] <span class="op">%&gt;%</span><span class="kw">unite</span>(<span class="st">&quot;Clone&quot;</span>,<span class="kw">all_of</span>(<span class="st">`</span><span class="dt">bulk_VAF_order</span><span class="st">`</span>),<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>, <span class="dt">remove =</span> <span class="ot">FALSE</span>)
 })

<span class="co">#Tally clones</span>
clonal_abundance&lt;-<span class="st"> </span><span class="kw">lapply</span>(NGT_to_clone,<span class="cf">function</span>(x){
  x<span class="op">%&gt;%</span><span class="kw">count</span>(Clone,<span class="dt">name=</span><span class="st">&quot;Count&quot;</span>)<span class="op">%&gt;%</span><span class="kw">arrange</span>(Count)
 })

<span class="co"># Setup a resampling function to generate multiple clonal abundance tallies</span>
resample_fun&lt;-<span class="cf">function</span>(data){
  x &lt;-<span class="st"> </span>data[<span class="kw">sample</span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(data),<span class="dt">replace=</span><span class="ot">TRUE</span>),]
  <span class="kw">return</span>(<span class="kw">as.matrix</span>(x<span class="op">%&gt;%</span><span class="kw">count</span>(Clone,<span class="dt">name=</span><span class="st">&quot;Count&quot;</span>)<span class="op">%&gt;%</span><span class="kw">arrange</span>(Count)))
}

replicates &lt;-<span class="st"> </span><span class="dv">100</span> <span class="co"># we did 10,000. Keeping it low here for run time.</span>
clone_cutoff &lt;-<span class="st"> </span><span class="dv">10</span> <span class="co"># minimum number of cells to keep a clone</span>
clonal_abundance_boot_CI &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">names</span>(NGT_to_clone),<span class="cf">function</span>(sample_to_test){
    test&lt;-<span class="kw">replicate</span>(<span class="dt">n=</span>replicates,<span class="kw">resample_fun</span>(NGT_to_clone[[sample_to_test]]),<span class="dt">simplify =</span> <span class="st">&quot;array&quot;</span>)
    <span class="cf">if</span>(<span class="kw">class</span>(test)<span class="op">==</span><span class="st">&quot;list&quot;</span>){
      y &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">lapply</span>(test,data.frame),<span class="dv">1</span><span class="op">:</span>replicates) <span class="op">%&gt;%</span>
<span class="st">           </span><span class="kw">imap</span>(<span class="dt">.x =</span> ., <span class="op">~</span><span class="st"> </span><span class="kw">set_names</span>(.x, <span class="kw">c</span>(<span class="st">&quot;Clone&quot;</span>, .y))) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">           </span>purrr<span class="op">::</span><span class="kw">reduce</span>(full_join, <span class="dt">by =</span> <span class="st">&quot;Clone&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">           </span><span class="kw">mutate_if</span>(<span class="kw">names</span>(.)<span class="op">!=</span><span class="st">&quot;Clone&quot;</span>,as.numeric)<span class="op">%&gt;%</span>
<span class="st">           </span><span class="kw">mutate_each</span>(<span class="kw">funs</span>(<span class="kw">replace</span>(., <span class="kw">is.na</span>(.), <span class="dv">0</span>)))
      }
    <span class="cf">if</span>(<span class="kw">class</span>(test)<span class="op">==</span><span class="st">&quot;array&quot;</span>){
      y &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">apply</span>(test,<span class="dv">3</span>,data.frame),<span class="dv">1</span><span class="op">:</span>replicates) <span class="op">%&gt;%</span>
<span class="st">           </span><span class="kw">imap</span>(<span class="dt">.x =</span> ., <span class="op">~</span><span class="st"> </span><span class="kw">set_names</span>(.x, <span class="kw">c</span>(<span class="st">&quot;Clone&quot;</span>, .y))) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">           </span>purrr<span class="op">::</span><span class="kw">reduce</span>(full_join, <span class="dt">by =</span> <span class="st">&quot;Clone&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">           </span><span class="kw">mutate_if</span>(<span class="kw">names</span>(.)<span class="op">!=</span><span class="st">&quot;Clone&quot;</span>,as.numeric)<span class="op">%&gt;%</span>
<span class="st">           </span><span class="kw">mutate_each</span>(<span class="kw">funs</span>(<span class="kw">replace</span>(., <span class="kw">is.na</span>(.), <span class="dv">0</span>)))
      }
    z &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">apply</span>(y<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="op">-</span>Clone),<span class="dv">1</span>,<span class="cf">function</span>(p){
            <span class="kw">quantile</span>(p,<span class="dt">probs=</span><span class="kw">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>))
         })),<span class="st">&quot;Clone&quot;</span>=y<span class="op">$</span>Clone)
    set &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">inner_join</span>(<span class="kw">data.frame</span>(clonal_abundance[[sample_to_test]]),z,<span class="dt">by=</span><span class="st">&quot;Clone&quot;</span>)),
                  <span class="kw">c</span>(<span class="st">&quot;Clone&quot;</span>,<span class="st">&quot;Count&quot;</span>,<span class="st">&quot;LCI&quot;</span>,<span class="st">&quot;UCI&quot;</span>))<span class="op">%&gt;%</span><span class="kw">filter</span>(LCI<span class="op">&gt;=</span>clone_cutoff)
})</code></pre></div>
<pre><code>## Warning: funs() is soft deprecated as of dplyr 0.8.0
## Please use a list of either functions or lambdas: 
## 
##   # Simple named list: 
##   list(mean = mean, median = median)
## 
##   # Auto named with `tibble::lst()`: 
##   tibble::lst(mean, median)
## 
##   # Using lambdas
##   list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))
## This warning is displayed once per session.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(clonal_abundance_boot_CI) &lt;-<span class="kw">names</span>(clonal_abundance)</code></pre></div>
<p>Now that we have a set of clones that we believe reproducibily have at least 10 cells, we remove cells and variants that are no longer represented at sufficient coverage.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">clone_filtered_NGTs &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">lapply</span>(<span class="kw">names</span>(clonal_abundance_boot_CI),<span class="cf">function</span>(sample_to_test){

  ## Determine if there are any clones left to process
  <span class="cf">if</span>(<span class="kw">nrow</span>(clonal_abundance_boot_CI[[sample_to_test]])<span class="op">==</span><span class="dv">0</span>) {
    <span class="kw">return</span>(<span class="st">&quot;No clones after boostrapping&quot;</span>)
  }
  
  ## Determine if there are any mutations that are no longer found in a stable clone
  clone_matrix&lt;-<span class="kw">as.matrix</span>(<span class="kw">do.call</span>(rbind,
                                  <span class="kw">strsplit</span>(clonal_abundance_boot_CI[[sample_to_test]][,<span class="st">&quot;Clone&quot;</span>],
                                           <span class="dt">split=</span><span class="st">&quot;_&quot;</span>)))
  <span class="kw">mode</span>(clone_matrix) &lt;-<span class="st"> &quot;numeric&quot;</span>
  <span class="kw">colnames</span>(clone_matrix) &lt;-<span class="kw">colnames</span>(NGT_to_clone[[sample_to_test]])[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]
  variants_to_remove&lt;-<span class="kw">names</span>(<span class="kw">which</span>(<span class="kw">colSums</span>(clone_matrix)<span class="op">==</span><span class="dv">0</span>))
  <span class="kw">print</span>(variants_to_remove)      
 
  ## Check other conditions of interest that might remove sample from further processing
  <span class="cf">if</span>(<span class="kw">nrow</span>(clone_matrix)<span class="op">==</span><span class="dv">1</span>) {
    <span class="kw">return</span>(<span class="st">&quot;Only 1 clone left&quot;</span>)
  }<span class="cf">else</span>  <span class="cf">if</span>(<span class="kw">length</span>(<span class="kw">setdiff</span>(<span class="kw">colnames</span>(clone_matrix),<span class="kw">c</span>(variants_to_remove)))<span class="op">&lt;=</span><span class="dv">1</span>){
    <span class="kw">return</span>(<span class="st">&quot;Removed all but 1 variant&quot;</span>)
  }<span class="cf">else</span> {
  
    <span class="co"># Select only clones that survive the bootstrapping, and remove variants that fall out</span>
    NGT_to_clone_subset &lt;-<span class="st"> </span>NGT_to_clone[[sample_to_test]]<span class="op">%&gt;%</span>
<span class="st">                            </span><span class="kw">filter</span>(Clone<span class="op">%in%</span>clonal_abundance_boot_CI[[sample_to_test]]<span class="op">$</span>Clone)<span class="op">%&gt;%</span>
<span class="st">                            </span><span class="kw">select</span>(<span class="op">!</span><span class="st">`</span><span class="dt">variants_to_remove</span><span class="st">`</span>) 
    
    <span class="co">#Create a key for the new and old clone names after removing variants that are no longer present</span>
    clone_key &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;New&quot;</span>=<span class="kw">apply</span>(<span class="kw">data.frame</span>(clone_matrix)<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="op">!</span>variants_to_remove),<span class="dt">MARGIN=</span><span class="dv">1</span>,
                                        <span class="cf">function</span>(x){ <span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)}),
                            <span class="st">&quot;Old&quot;</span>=<span class="kw">apply</span>(<span class="kw">data.frame</span>(clone_matrix),<span class="dt">MARGIN=</span><span class="dv">1</span>,
                                        <span class="cf">function</span>(x){ <span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)}))
    
    <span class="co">#If there are any variants to remove and clones that need to be renamed</span>
    <span class="cf">if</span>(<span class="kw">any</span>(clone_key<span class="op">$</span>New<span class="op">!=</span>clone_key<span class="op">$</span>Old)){
        NGT_to_clone_subset<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">sapply</span>(NGT_to_clone_subset<span class="op">$</span>Clone,<span class="cf">function</span>(x){
                                          clone_key<span class="op">$</span>New[<span class="kw">match</span>(x,clone_key<span class="op">$</span>Old)]})
    }
    <span class="kw">return</span>(NGT_to_clone_subset)
  }
}),<span class="kw">names</span>(clonal_abundance_boot_CI))</code></pre></div>
<pre><code>## character(0)</code></pre>
<pre><code>## Note: Using an external vector in selections is ambiguous.
## ℹ Use `all_of(variants_to_remove)` instead of `variants_to_remove` to silence this message.
## ℹ See &lt;https://tidyselect.r-lib.org/reference/faq-external-vector.html&gt;.
## This message is displayed once per session.</code></pre>
<pre><code>## character(0)
## character(0)
## character(0)</code></pre>
<p>One last step I find useful is to explicitly state the genotype of each mutation in each clone. This si useful for the clonotype plots we’ll make later.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">clonal_architecture &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">lapply</span>(<span class="kw">names</span>(clonal_abundance_boot_CI),<span class="cf">function</span>(sample_to_test){

  clonal_architecture&lt;-clone_filtered_NGTs[[sample_to_test]]<span class="op">%&gt;%</span>
<span class="st">                                                </span><span class="kw">select</span>(<span class="op">!</span>Cell)<span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">                                                </span><span class="kw">distinct</span>()<span class="op">%&gt;%</span>
<span class="st">                                                </span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="op">!</span>Clone,<span class="dt">names_to=</span><span class="st">&quot;Mutant&quot;</span>,<span class="dt">values_to=</span><span class="st">&quot;Genotype&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                                                </span><span class="kw">mutate</span>(<span class="dt">Genotype=</span><span class="kw">ifelse</span>(Genotype<span class="op">==</span><span class="dv">3</span>,<span class="ot">NA</span>,
                                                                   <span class="kw">ifelse</span>(Genotype<span class="op">==</span><span class="dv">0</span>,<span class="st">&quot;WT&quot;</span>,
                                                                   <span class="kw">ifelse</span>(Genotype<span class="op">==</span><span class="dv">1</span>,<span class="st">&quot;Heterozygous&quot;</span>,                                                                                   <span class="kw">ifelse</span>(Genotype<span class="op">==</span><span class="dv">2</span>,<span class="st">&quot;Homozygous&quot;</span>,<span class="ot">NA</span>)))))
  
}), <span class="kw">names</span>(clonal_abundance_boot_CI))</code></pre></div>
<p>Lastly, we are going to package everything together into a list format for easy access later.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">final_sample_summary&lt;-<span class="kw">setNames</span>(<span class="kw">lapply</span>(<span class="kw">names</span>(clonal_architecture),<span class="cf">function</span>(sample){
   <span class="kw">return</span>(<span class="kw">list</span>(<span class="st">&quot;Clones&quot;</span>=clonal_abundance_boot_CI[[sample]],
               <span class="st">&quot;NGT&quot;</span>=clone_filtered_NGTs[[sample]],
               <span class="st">&quot;Architecture&quot;</span>=clonal_architecture[[sample]]))
}),<span class="kw">names</span>(clonal_abundance_boot_CI))

<span class="kw">saveRDS</span>(final_sample_summary,<span class="dt">file=</span><span class="st">&quot;./analysis/final_sample_summary.rds&quot;</span>)</code></pre></div>


</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="post-processing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cohort-description.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scDNA.pdf", "scDNA.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
