<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Cohort Description | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</title>
  <meta name="description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  <meta name="generator" content="bookdown 0.18.3 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Cohort Description | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Cohort Description | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  
  <meta name="twitter:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

<meta name="author" content="Robert Bowman" />


<meta name="date" content="2020-05-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="assessing-clonal-abundance.html"/>
<link rel="next" href="final-words.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreward</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="data-extraction-and-setup.html"><a href="data-extraction-and-setup.html"><i class="fa fa-check"></i><b>3</b> Data Extraction and Setup</a><ul>
<li class="chapter" data-level="3.1" data-path="tapestri-library.html"><a href="tapestri-library.html"><i class="fa fa-check"></i><b>3.1</b> Tapestri Package</a></li>
<li class="chapter" data-level="3.2" data-path="post-processing.html"><a href="post-processing.html"><i class="fa fa-check"></i><b>3.2</b> Post processing</a></li>
<li class="chapter" data-level="3.3" data-path="assessing-clonal-abundance.html"><a href="assessing-clonal-abundance.html"><i class="fa fa-check"></i><b>3.3</b> Assessing clonal abundance</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="cohort-description.html"><a href="cohort-description.html"><i class="fa fa-check"></i><b>4</b> Cohort Description</a></li>
<li class="chapter" data-level="5" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>5</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="Appendix.html"><a href="Appendix.html"><i class="fa fa-check"></i><b>6</b> Appendix</a><ul>
<li class="chapter" data-level="6.1" data-path="AppendixA.html"><a href="AppendixA.html"><i class="fa fa-check"></i><b>6.1</b> Import from Tapestri Insights</a></li>
<li class="chapter" data-level="6.2" data-path="functions.html"><a href="functions.html"><i class="fa fa-check"></i><b>6.2</b> Functions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cohort-description" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Cohort Description</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Scripts/mBio_processing_scripts.R&quot;</span>)

final_NGTs&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/final_NGTs.rds&quot;</span>)
pheno_mut_melted&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/pheno_mut_melted.rds&quot;</span>)
final_sample_set&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/final_sample_set.rds&quot;</span>)


final_mut_melt &lt;-pheno_mut_melted<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample.ID<span class="op">%in%</span>final_sample_set)
final_mut_melt<span class="op">$</span>Gene&lt;-<span class="st"> </span><span class="kw">factor</span>(final_mut_melt<span class="op">$</span>Gene,<span class="dt">levels=</span><span class="kw">names</span>(<span class="kw">sort</span>(<span class="kw">table</span>(final_mut_melt<span class="op">$</span>Gene), <span class="dt">decreasing=</span><span class="ot">TRUE</span>)))

## Total number of mutations identified
gg_mut_count&lt;-<span class="kw">ggplot</span>(final_mut_melt,<span class="kw">aes</span>(<span class="dt">x=</span>Gene))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;count&quot;</span>)<span class="op">+</span>
<span class="st">  </span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span> <span class="dv">8</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of mutations&quot;</span>)<span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">45</span>, <span class="dt">hjust=</span><span class="dv">1</span>,<span class="dt">vjust=</span><span class="dv">1</span>))

## Melt to get a tally of how many mutations per patient
melted_mut_mat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">count</span>(final_mut_melt, Gene, Sample.ID))
melted_mut_mat<span class="op">$</span>Gene&lt;-<span class="st"> </span><span class="kw">factor</span>(melted_mut_mat<span class="op">$</span>Gene,<span class="dt">levels=</span><span class="kw">names</span>(<span class="kw">sort</span>(<span class="kw">table</span>(melted_mut_mat<span class="op">$</span>Gene),<span class="dt">decreasing=</span><span class="ot">TRUE</span>)))
gg_mut_patient&lt;-<span class="kw">ggplot</span>(melted_mut_mat,<span class="kw">aes</span>(<span class="dt">x=</span>Gene))<span class="op">+</span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;count&quot;</span>)<span class="op">+</span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span> <span class="dv">8</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of patients with mutation&quot;</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">45</span>,<span class="dt">hjust=</span><span class="dv">1</span>,<span class="dt">vjust=</span><span class="dv">1</span>))

## Total nunber of mutant genes per patient
gg_mutated_genes_per_patient&lt;-melted_mut_mat<span class="op">%&gt;%</span><span class="kw">group_by</span>(Sample.ID)<span class="op">%&gt;%</span>tally<span class="op">%&gt;%</span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n))<span class="op">+</span><span class="kw">geom_bar</span>()<span class="op">+</span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span> <span class="dv">8</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Mutant genes per patient&quot;</span>)<span class="op">+</span><span class="kw">xlab</span>(<span class="st">&quot;Number of genes&quot;</span>)

## Total number of mutations per patient
melted_variant_mat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">count</span>(final_mut_melt, protein, Sample.ID))
melted_variant_mat<span class="op">$</span>protein&lt;-<span class="st"> </span><span class="kw">factor</span>(melted_variant_mat<span class="op">$</span>protein, <span class="dt">levels=</span><span class="kw">names</span>(<span class="kw">sort</span>(<span class="kw">table</span>(melted_variant_mat<span class="op">$</span>protein), <span class="dt">decreasing=</span><span class="ot">TRUE</span>)))
gg_mutations_per_patient&lt;-melted_variant_mat<span class="op">%&gt;%</span><span class="kw">group_by</span>(Sample.ID)<span class="op">%&gt;%</span>tally<span class="op">%&gt;%</span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n))<span class="op">+</span><span class="kw">geom_bar</span>()<span class="op">+</span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span> <span class="dv">10</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Variants per patient&quot;</span>)<span class="op">+</span><span class="kw">xlab</span>(<span class="st">&quot;Number of variants&quot;</span>)

### Cells per sample
gg_cells_per_patient&lt;-<span class="kw">data.frame</span>(<span class="st">&quot;Cells&quot;</span>=<span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(final_NGTs,nrow)))<span class="op">%&gt;%</span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>Cells))<span class="op">+</span><span class="kw">geom_histogram</span>(<span class="dt">binwidth =</span> <span class="dv">100</span>)<span class="op">+</span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span> <span class="dv">10</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Informative Cells per Sample&quot;</span>) <span class="co">#+scale_x_continuous(breaks=c(0,500,1000,2000,4000,8000))</span>

### create matrix for oncoprint
mut_mat &lt;-<span class="st"> </span><span class="kw">table</span>(melted_mut_mat<span class="op">$</span>Sample.ID,melted_mut_mat<span class="op">$</span>Gene)
mut_mat &lt;-<span class="st"> </span>mut_mat[,<span class="kw">colSums</span>(mut_mat)<span class="op">!=</span><span class="dv">0</span>]
tronco_mat &lt;-<span class="kw">import.genotypes</span>(mut_mat, <span class="dt">event.type=</span><span class="st">&#39;myVariant&#39;</span>)
grob_oncoprint&lt;-<span class="kw">grid.grabExpr</span>(<span class="kw">oncoprint</span>(tronco_mat))

### Prepare matrix for co occurence map
grob_corrplot&lt;-<span class="kw">generate_and_plot_cooccurence</span>(mut_mat)

<span class="co">#EF1A</span>
<span class="kw">ggsave</span>(gg_mut_count<span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>)), <span class="dt">width=</span><span class="fl">3.75</span>,<span class="dt">height=</span><span class="fl">2.25</span>,
        <span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/Figures/EF1A-mutation_count.pdf&quot;</span>)

<span class="co">#EF1B</span>
<span class="kw">ggsave</span>(gg_mut_patient<span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>)), <span class="dt">width=</span><span class="fl">3.75</span>,<span class="dt">height=</span><span class="fl">2.25</span>,
       <span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/Figures/EF1B-number_of_patients_with_mutation.pdf&quot;</span>)

<span class="co">#EF1C</span>
<span class="kw">ggsave</span>(gg_mutated_genes_per_patient<span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>)), <span class="dt">width=</span><span class="fl">3.75</span>,<span class="dt">height=</span><span class="fl">2.25</span>,
       <span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/Figures/EF1C-mutant_genes_per_patient.pdf&quot;</span>)

<span class="co">#EF1D</span>
<span class="kw">ggsave</span>(gg_mutations_per_patient<span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust =</span> <span class="fl">0.5</span>)), <span class="dt">width=</span><span class="fl">3.75</span>,<span class="dt">height=</span><span class="fl">2.25</span>,
       <span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/Analysis/2020/January/Figures/EF1D-variants_per_patient.pdf&quot;</span>)




final_mut_melt<span class="op">$</span>Variant_Class&lt;-<span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;fs</span><span class="ch">\\</span><span class="st">*&quot;</span>,final_mut_melt<span class="op">$</span>protein),<span class="st">&quot;Indel&quot;</span>,
                                 <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;INS_&quot;</span>,final_mut_melt<span class="op">$</span>protein),<span class="st">&quot;Indel&quot;</span>,
                                        <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;ins&quot;</span>,final_mut_melt<span class="op">$</span>protein),<span class="st">&quot;Indel&quot;</span>,
                                               <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;ext&quot;</span>,final_mut_melt<span class="op">$</span>protein),<span class="st">&quot;Indel&quot;</span>,
                                                      <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;del&quot;</span>,final_mut_melt<span class="op">$</span>protein),<span class="st">&quot;Indel&quot;</span>,
                                                           <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">*$&quot;</span>,final_mut_melt<span class="op">$</span>protein),<span class="st">&quot;Nonsense&quot;</span>,<span class="st">&quot;Missense&quot;</span>))))))
final_mut_melt<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">as.character</span>(final_mut_melt<span class="op">$</span>Gene)
final_mut_melt<span class="op">$</span>Variant_Class &lt;-<span class="st"> </span><span class="kw">as.character</span>(final_mut_melt<span class="op">$</span>Variant_Class)
final_mut_melt<span class="op">$</span>Sample.ID &lt;-<span class="st"> </span><span class="kw">as.character</span>(final_mut_melt<span class="op">$</span>Sample.ID)


mut_mat_wide&lt;-<span class="kw">data.frame</span>(final_mut_melt<span class="op">%&gt;%</span><span class="kw">group_by</span>(Sample.ID)<span class="op">%&gt;%</span><span class="kw">pivot_wider</span>(<span class="dt">id_cols=</span>Sample.ID, <span class="dt">names_from =</span> Gene, <span class="dt">values_from =</span> Variant_Class,<span class="dt">values_fn =</span> <span class="kw">list</span>(<span class="dt">Variant_Class =</span> list)),<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
mut_mat_wide[<span class="kw">is.na</span>(mut_mat_wide)] =<span class="st"> &quot; &quot;</span>

mut_mat_wide_storage &lt;-<span class="st"> </span><span class="kw">list</span>()
<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span><span class="kw">ncol</span>(mut_mat_wide)){
  mut_mat_wide_storage[[i]]&lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(mut_mat_wide[,i],<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;;&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;;&quot;</span>)}))
}
mut_mat_wide_storage[[<span class="dv">1</span>]] &lt;-<span class="st"> </span>mut_mat_wide[,<span class="dv">1</span>]
final_mat&lt;-<span class="kw">do.call</span>(cbind,mut_mat_wide_storage)
<span class="kw">colnames</span>(final_mat) &lt;-<span class="st"> </span><span class="kw">colnames</span>(mut_mat_wide)
<span class="kw">rownames</span>(final_mat) &lt;-<span class="st"> </span>final_mat[,<span class="dv">1</span>]
final_mat &lt;-<span class="st"> </span><span class="kw">t</span>(final_mat[,<span class="op">-</span><span class="dv">1</span>])



col =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Indel&quot;</span> =<span class="st"> &quot;darkorchid2&quot;</span>, <span class="st">&quot;Nonsense&quot;</span> =<span class="st"> &quot;black&quot;</span>, <span class="st">&quot;Missense&quot;</span> =<span class="st"> &quot;darkgreen&quot;</span>)
alter_fun =<span class="st"> </span><span class="kw">list</span>(
  <span class="dt">background =</span> <span class="cf">function</span>(x, y, w, h) {
    <span class="kw">grid.rect</span>(x, y, w, h<span class="op">-</span><span class="kw">unit</span>(<span class="fl">0.25</span>, <span class="st">&quot;mm&quot;</span>), 
              <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> <span class="st">&quot;#CCCCCC&quot;</span>, <span class="dt">col =</span> <span class="ot">NA</span>))
  },
  <span class="co"># big blue</span>
  <span class="dt">Indel =</span> <span class="cf">function</span>(x, y, w, h) {
    <span class="kw">grid.rect</span>(x, y, w<span class="op">-</span><span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), h<span class="op">-</span><span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), 
              <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> col[<span class="st">&quot;Indel&quot;</span>], <span class="dt">col =</span> <span class="ot">NA</span>))
  },
  <span class="co"># bug red</span>
  <span class="dt">Nonsense =</span> <span class="cf">function</span>(x, y, w, h) {
    <span class="kw">grid.rect</span>(x, y, w<span class="op">-</span><span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), h<span class="op">-</span><span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), 
              <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> col[<span class="st">&quot;Nonsense&quot;</span>], <span class="dt">col =</span> <span class="ot">NA</span>))
  },
  <span class="co"># small green</span>
  <span class="dt">Missense =</span> <span class="cf">function</span>(x, y, w, h) {
    <span class="kw">grid.rect</span>(x, y, w<span class="op">-</span><span class="kw">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;mm&quot;</span>), h<span class="op">*</span><span class="fl">0.33</span>, 
              <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fill =</span> col[<span class="st">&quot;Missense&quot;</span>], <span class="dt">col =</span> <span class="ot">NA</span>))
  }
)


<span class="kw">library</span>(ComplexHeatmap)

sample_key&lt;-<span class="kw">read.delim</span>(<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/From_LAM/OS_and_Cheatsheet/Mission_Bio_MRN_RunID_Samples.txt&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)
sample_clinical&lt;-<span class="kw">read.delim</span>(<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/From_LAM/OS_and_Cheatsheet/sample_clinical_information.txt&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)

merged_set&lt;-<span class="kw">full_join</span>(sample_key,sample_clinical,<span class="dt">by=</span><span class="st">&quot;Molecular_accession_num&quot;</span>)

missing_patient_info&lt;-<span class="kw">setdiff</span>(<span class="kw">colnames</span>(final_mat),merged_set<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample.Name<span class="op">%in%</span><span class="kw">colnames</span>(final_mat))<span class="op">%&gt;%</span><span class="kw">distinct</span>(Sample.Name)<span class="op">%&gt;%</span><span class="kw">pull</span>(Sample.Name))
complete_patient_info&lt;-<span class="kw">intersect</span>(<span class="kw">colnames</span>(final_mat),merged_set<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample.Name<span class="op">%in%</span><span class="kw">colnames</span>(final_mat))<span class="op">%&gt;%</span><span class="kw">distinct</span>(Sample.Name)<span class="op">%&gt;%</span><span class="kw">pull</span>(Sample.Name))


set&lt;-<span class="kw">rbind</span>(<span class="kw">data.frame</span>(merged_set<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample.Name<span class="op">%in%</span>complete_patient_info)<span class="op">%&gt;%</span><span class="kw">select</span>(Dx,Disease.Status,Sample.Name)),
      <span class="kw">data.frame</span>(<span class="st">&quot;Dx&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^CH&quot;</span>,missing_patient_info),<span class="st">&quot;CH&quot;</span>,<span class="st">&quot;AML&quot;</span>),
                 <span class="st">&quot;Disease.Status&quot;</span>=<span class="ot">NA</span>,
                <span class="st">&quot;Sample.Name&quot;</span>=missing_patient_info))

set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;S5522&quot;</span>),<span class="st">&quot;Dx&quot;</span>] &lt;-<span class="st"> &quot;tAML&quot;</span>
set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;F7469&quot;</span>,<span class="st">&quot;F7469dnew&quot;</span>,<span class="st">&quot;MA4244B&quot;</span>,<span class="st">&quot;MA6300B&quot;</span>,<span class="st">&quot;MA6363C&quot;</span>),<span class="st">&quot;Dx&quot;</span>] &lt;-<span class="st"> &quot;sAML&quot;</span>
set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;RF2768&quot;</span>,<span class="st">&quot;R6501&quot;</span>,<span class="st">&quot;CB0346&quot;</span>,<span class="st">&quot;CB4833&quot;</span>,<span class="st">&quot;E3083new&quot;</span>,<span class="st">&quot;E6502&quot;</span>,<span class="st">&quot;E6522&quot;</span>,<span class="st">&quot;E8261&quot;</span>,<span class="st">&quot;F5348&quot;</span>,<span class="st">&quot;R2278new&quot;</span>),<span class="st">&quot;Dx&quot;</span>] &lt;-<span class="st"> &quot;AML&quot;</span>

set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;S5046&quot;</span>,<span class="st">&quot;S5331&quot;</span>,<span class="st">&quot;S4410&quot;</span>,<span class="st">&quot;S6225&quot;</span>,<span class="st">&quot;S5522&quot;</span>,<span class="st">&quot;CB4833&quot;</span>,<span class="st">&quot;E3083new&quot;</span>,<span class="st">&quot;E6502&quot;</span>),<span class="st">&quot;Disease.Status&quot;</span>] &lt;-<span class="st"> &quot;Relapse&quot;</span>
set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;S3326&quot;</span>,<span class="st">&quot;S5346&quot;</span>,<span class="st">&quot;S5480&quot;</span>,<span class="st">&quot;S5941&quot;</span>,<span class="st">&quot;Meyer_1&quot;</span>,<span class="st">&quot;17-005&quot;</span>,<span class="st">&quot;17-006&quot;</span>,<span class="st">&quot;17-020&quot;</span>,<span class="st">&quot;17-021&quot;</span>),<span class="st">&quot;Disease.Status&quot;</span>] &lt;-<span class="st"> &quot;Newly Diagnosed&quot;</span>
set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;RF2768&quot;</span>,<span class="st">&quot;R6501&quot;</span>,<span class="st">&quot;CB0346&quot;</span>,<span class="st">&quot;F5348&quot;</span>,<span class="st">&quot;R2278new&quot;</span>),<span class="st">&quot;Disease.Status&quot;</span>] &lt;-<span class="st"> &quot;Newly Diagnosed&quot;</span>
set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;E6522&quot;</span>,<span class="st">&quot;F7469&quot;</span>,<span class="st">&quot;F7469dnew&quot;</span>,<span class="st">&quot;MA6300B&quot;</span>,<span class="st">&quot;MA6363C&quot;</span>),<span class="st">&quot;Disease.Status&quot;</span>] &lt;-<span class="st"> &quot;Newly Transformed&quot;</span>
set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;E8261&quot;</span>,<span class="st">&quot;MA4244B&quot;</span>),<span class="st">&quot;Disease.Status&quot;</span>] &lt;-<span class="st"> &quot;Refractory&quot;</span>


set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;MA0092A&quot;</span>,<span class="st">&quot;MA1715B&quot;</span>,<span class="st">&quot;MA4244A&quot;</span>,<span class="st">&quot;MA4849B&quot;</span>,<span class="st">&quot;MA6300A&quot;</span>,<span class="st">&quot;MA6363B&quot;</span>,<span class="st">&quot;MA9521A&quot;</span>,<span class="st">&quot;MA9521B&quot;</span>),<span class="st">&quot;Dx&quot;</span>] &lt;-<span class="st"> &quot;MPN&quot;</span>
set[set<span class="op">$</span>Sample.Name<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;MA0092A&quot;</span>,<span class="st">&quot;MA1715B&quot;</span>,<span class="st">&quot;MA4244A&quot;</span>,<span class="st">&quot;MA4849B&quot;</span>,<span class="st">&quot;MA6300A&quot;</span>,<span class="st">&quot;MA6363B&quot;</span>,<span class="st">&quot;MA9521A&quot;</span>,<span class="st">&quot;MA9521B&quot;</span>),<span class="st">&quot;Disease.Status&quot;</span>] &lt;-<span class="st"> &quot;Chronic Stage&quot;</span>

set_dedup&lt;-set[<span class="op">!</span><span class="kw">duplicated</span>(<span class="kw">as.matrix</span>(set<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample.Name<span class="op">%in%</span><span class="kw">colnames</span>(final_mat)))),]

set_dedup&lt;-<span class="kw">read.delim</span>(<span class="dt">file=</span><span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/From_LAM/OS_and_Cheatsheet/BB_merged_sample_clinical_information.txt&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">row.names=</span><span class="dv">4</span>)[,<span class="op">-</span><span class="dv">1</span>]

<span class="kw">colnames</span>(set_dedup) &lt;-<span class="kw">c</span>(<span class="st">&quot;Diagnosis&quot;</span>,<span class="st">&quot;Status&quot;</span>)
set_dedup&lt;-<span class="st"> </span>set_dedup[<span class="kw">colnames</span>(final_mat),]
set_dedup[<span class="kw">is.na</span>(set_dedup[,<span class="dv">2</span>]),<span class="st">&quot;Status&quot;</span>] &lt;-<span class="st"> &quot;Other&quot;</span>

color_set &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="st">&quot;Diagnosis&quot;</span> =<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">okabe</span>(<span class="dt">n=</span><span class="kw">length</span>(<span class="kw">unique</span>(set_dedup[,<span class="dv">1</span>]))),<span class="kw">c</span>( <span class="st">&quot;CH&quot;</span>,<span class="st">&quot;MDS&quot;</span>,<span class="st">&quot;MPN&quot;</span>, <span class="st">&quot;AML&quot;</span>, <span class="st">&quot;sAML&quot;</span> ,<span class="st">&quot;tAML&quot;</span>,<span class="st">&quot;Other&quot;</span>) ),
                       <span class="st">&quot;Status&quot;</span> =<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">tol</span>(<span class="dt">n=</span>(<span class="kw">length</span>(<span class="kw">unique</span>(set_dedup[,<span class="dv">2</span>])))),<span class="kw">c</span>(<span class="st">&quot;Newly Diagnosed&quot;</span>,<span class="st">&quot;Relapse/Refractory&quot;</span>,<span class="st">&quot;Newly Transformed&quot;</span>,<span class="st">&quot;Persistent&quot;</span>,<span class="st">&quot;Chronic Stage&quot;</span>, <span class="st">&quot;Other&quot;</span>)))                       
                                                    
top_annotation=<span class="kw">HeatmapAnnotation</span>(<span class="dt">cbar =</span> <span class="kw">anno_oncoprint_barplot</span>(),
                  <span class="dt">df=</span>set_dedup[,<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>],
                  <span class="dt">col =</span><span class="kw">list</span>(<span class="st">&quot;Diagnosis&quot;</span>=color_set[[<span class="dv">1</span>]],
                            <span class="st">&quot;Status&quot;</span>=color_set[[<span class="dv">2</span>]])
                  )



column_title =<span class="st"> &quot;&quot;</span>
heatmap_legend_param =<span class="st"> </span><span class="kw">list</span>(<span class="dt">title =</span> <span class="st">&quot;Alternations&quot;</span>, 
                            <span class="dt">at =</span> <span class="kw">c</span>(<span class="st">&quot;Indel&quot;</span>, <span class="st">&quot;Nonsense&quot;</span>, <span class="st">&quot;Missense&quot;</span>), 
                            <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Indel&quot;</span>, <span class="st">&quot;Nonsense&quot;</span>, <span class="st">&quot;Missense&quot;</span>))
<span class="kw">oncoPrint</span>(final_mat,
          <span class="dt">alter_fun =</span> alter_fun, 
          <span class="dt">col =</span> col, 
          <span class="dt">top_annotation=</span>top_annotation,
         <span class="co">#heatmap_legend_param = list(direction = &quot;horizontal&quot;),</span>
          <span class="co">#column_title = column_title# </span>
          <span class="dt">heatmap_legend_param =</span> heatmap_legend_param)</code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="assessing-clonal-abundance.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="final-words.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scDNA.pdf", "scDNA.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
