<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4.1 Cohort Description: EF1A-D | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</title>
  <meta name="description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  <meta name="generator" content="bookdown 0.18.3 and GitBook 2.6.7" />

  <meta property="og:title" content="4.1 Cohort Description: EF1A-D | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4.1 Cohort Description: EF1A-D | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  
  <meta name="twitter:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

<meta name="author" content="Robert Bowman" />


<meta name="date" content="2020-05-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cohort-description-and-exploration.html"/>
<link rel="next" href="oncoprint-figure-1a.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreward</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="data-extraction-and-setup.html"><a href="data-extraction-and-setup.html"><i class="fa fa-check"></i><b>3</b> Data Extraction and Setup</a><ul>
<li class="chapter" data-level="3.1" data-path="tapestri-library.html"><a href="tapestri-library.html"><i class="fa fa-check"></i><b>3.1</b> Tapestri Package</a></li>
<li class="chapter" data-level="3.2" data-path="post-processing.html"><a href="post-processing.html"><i class="fa fa-check"></i><b>3.2</b> Post processing</a></li>
<li class="chapter" data-level="3.3" data-path="assessing-clonal-abundance.html"><a href="assessing-clonal-abundance.html"><i class="fa fa-check"></i><b>3.3</b> Assessing clonal abundance</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="cohort-description-and-exploration.html"><a href="cohort-description-and-exploration.html"><i class="fa fa-check"></i><b>4</b> Cohort description and exploration</a><ul>
<li class="chapter" data-level="4.1" data-path="cohort-description-ef1a-d.html"><a href="cohort-description-ef1a-d.html"><i class="fa fa-check"></i><b>4.1</b> Cohort Description: EF1A-D</a></li>
<li class="chapter" data-level="4.2" data-path="oncoprint-figure-1a.html"><a href="oncoprint-figure-1a.html"><i class="fa fa-check"></i><b>4.2</b> Oncoprint: Figure 1A</a></li>
<li class="chapter" data-level="4.3" data-path="sample-clonality-figure-1ce-figure-2ab.html"><a href="sample-clonality-figure-1ce-figure-2ab.html"><i class="fa fa-check"></i><b>4.3</b> Sample clonality: Figure 1C,E Figure 2A,B</a></li>
<li class="chapter" data-level="4.4" data-path="clonograph-figure-1d.html"><a href="clonograph-figure-1d.html"><i class="fa fa-check"></i><b>4.4</b> Clonograph: Figure 1D</a></li>
<li class="chapter" data-level="4.5" data-path="gene-clonality-figure-2.html"><a href="gene-clonality-figure-2.html"><i class="fa fa-check"></i><b>4.5</b> Gene Clonality: Figure 2</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>5</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="Appendix.html"><a href="Appendix.html"><i class="fa fa-check"></i><b>6</b> Appendix</a><ul>
<li class="chapter" data-level="6.1" data-path="AppendixA.html"><a href="AppendixA.html"><i class="fa fa-check"></i><b>6.1</b> Import from Tapestri Insights</a></li>
<li class="chapter" data-level="6.2" data-path="functions.html"><a href="functions.html"><i class="fa fa-check"></i><b>6.2</b> Functions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cohort-description-ef1a-d" class="section level2">
<h2><span class="header-section-number">4.1</span> Cohort Description: EF1A-D</h2>
<p>Now we can start to make the plots that are in Extended Figure 1. Starting with the total number of mutations identified per gene</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Set the levels of the Gene column from most to least prevalent for plotting purposes
final_mut_melt<span class="op">$</span>Gene&lt;-<span class="st"> </span><span class="kw">factor</span>(final_mut_melt<span class="op">$</span>Gene,<span class="dt">levels=</span><span class="kw">names</span>(<span class="kw">sort</span>(<span class="kw">table</span>(final_mut_melt<span class="op">$</span>Gene), <span class="dt">decreasing=</span><span class="ot">TRUE</span>)))

gg_mut_count&lt;-<span class="kw">ggplot</span>(final_mut_melt,<span class="kw">aes</span>(<span class="dt">x=</span>Gene))<span class="op">+</span>
<span class="st">                        </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;count&quot;</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span> <span class="dv">10</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of mutations&quot;</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">45</span>, <span class="dt">hjust=</span><span class="dv">1</span>,<span class="dt">vjust=</span><span class="dv">1</span>),
                              <span class="dt">plot.title=</span><span class="kw">element_text</span>(<span class="dt">hjust=</span><span class="fl">0.5</span>))<span class="op">+</span>
<span class="st">                        </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</code></pre></div>
<p>Total number of patients mutated for each gene</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## tally of how many mutations per patient
melted_mut_mat&lt;-<span class="st"> </span>final_mut_melt<span class="op">%&gt;%</span><span class="kw">count</span>(Gene, Sample)

## Set the levels of the Gene column from most to least prevalent for plotting purposes
melted_mut_mat<span class="op">$</span>Gene&lt;-<span class="st"> </span><span class="kw">factor</span>(melted_mut_mat<span class="op">$</span>Gene,<span class="dt">levels=</span><span class="kw">names</span>(<span class="kw">sort</span>(<span class="kw">table</span>(melted_mut_mat<span class="op">$</span>Gene),<span class="dt">decreasing=</span><span class="ot">TRUE</span>)))

gg_mut_patient&lt;-<span class="kw">ggplot</span>(melted_mut_mat,<span class="kw">aes</span>(<span class="dt">x=</span>Gene))<span class="op">+</span>
<span class="st">                      </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;count&quot;</span>)<span class="op">+</span>
<span class="st">                      </span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span><span class="dv">10</span>)<span class="op">+</span>
<span class="st">                      </span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of patients with mutation&quot;</span>)<span class="op">+</span>
<span class="st">                      </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">45</span>,<span class="dt">hjust=</span><span class="dv">1</span>,<span class="dt">vjust=</span><span class="dv">1</span>),
                            <span class="dt">plot.title=</span><span class="kw">element_text</span>(<span class="dt">hjust=</span><span class="fl">0.5</span>))<span class="op">+</span>
<span class="st">                          </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))</code></pre></div>
<p>Number of mutated genes per patient</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gg_mutated_genes_per_patient&lt;-final_mut_melt<span class="op">%&gt;%</span>
<span class="st">                                          </span><span class="kw">distinct</span>(Sample,Gene)<span class="op">%&gt;%</span>
<span class="st">                                          </span><span class="kw">group_by</span>(Sample)<span class="op">%&gt;%</span>
<span class="st">                                          </span>tally<span class="op">%&gt;%</span>
<span class="st">                                          </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n))<span class="op">+</span><span class="kw">geom_bar</span>()<span class="op">+</span>
<span class="st">                                          </span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span>
<span class="st">                                          </span><span class="kw">xlab</span>(<span class="st">&quot;Number of genes&quot;</span>)<span class="op">+</span>
<span class="st">                                          </span><span class="kw">ggtitle</span>(<span class="st">&quot;Mutant genes per patient&quot;</span>)<span class="op">+</span>
<span class="st">                                          </span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span> <span class="dv">10</span>)<span class="op">+</span>
<span class="st">                                          </span><span class="kw">theme</span>(<span class="dt">plot.title=</span><span class="kw">element_text</span>(<span class="dt">hjust=</span><span class="fl">0.5</span>))<span class="op">+</span>
<span class="st">                                          </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="op">+</span>
<span class="st">                                          </span><span class="kw">scale_x_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),<span class="dt">n.breaks=</span><span class="dv">8</span>)</code></pre></div>
<p>Total number of mutations per patient</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gg_mutations_per_patient&lt;-<span class="st"> </span>final_mut_melt<span class="op">%&gt;%</span>
<span class="st">                                        </span><span class="kw">group_by</span>(Sample)<span class="op">%&gt;%</span>
<span class="st">                                        </span>tally<span class="op">%&gt;%</span>
<span class="st">                                        </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>n))<span class="op">+</span><span class="kw">geom_bar</span>()<span class="op">+</span>
<span class="st">                                        </span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span> <span class="dv">10</span>)<span class="op">+</span>
<span class="st">                                        </span><span class="kw">theme</span>(<span class="dt">plot.title=</span><span class="kw">element_text</span>(<span class="dt">hjust=</span><span class="fl">0.5</span>))<span class="op">+</span>
<span class="st">                                        </span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Variants per patient&quot;</span>)<span class="op">+</span><span class="kw">xlab</span>(<span class="st">&quot;Number of variants&quot;</span>)<span class="op">+</span><span class="st">                              </span>
<span class="st">                                        </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="op">+</span>
<span class="st">                                        </span><span class="kw">scale_x_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),<span class="dt">n.breaks=</span><span class="dv">6</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(cowplot)
<span class="kw">plot_grid</span>(gg_mut_count,gg_mut_patient,
          gg_mutated_genes_per_patient,gg_mutations_per_patient,
          <span class="dt">ncol=</span><span class="dv">2</span>,<span class="dt">align=</span><span class="st">&quot;hv&quot;</span>,<span class="dt">axis=</span><span class="st">&quot;ltrb&quot;</span>,
          <span class="dt">labels =</span> <span class="st">&quot;AUTO&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-11"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-11-1.png" alt="Extended Figure 1A-D" width="864" />
<p class="caption">
Figure 4.1: Extended Figure 1A-D
</p>
</div>
<p>An extra plot worth noting is below, which plots the number of cells per sample, after we filterd out all of the “3” uniformative genotyped cells.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">### Cells per sample
<span class="kw">data.frame</span>(<span class="st">&quot;Cells&quot;</span>=<span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(final_NGTs,nrow)))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>Cells))<span class="op">+</span><span class="kw">geom_histogram</span>(<span class="dt">binwidth =</span> <span class="dv">100</span>)<span class="op">+</span>
<span class="st">                                      </span><span class="kw">theme_classic</span>(<span class="dt">base_size =</span> <span class="dv">10</span>)<span class="op">+</span>
<span class="st">                                      </span><span class="kw">theme</span>(<span class="dt">plot.title=</span><span class="kw">element_text</span>(<span class="dt">hjust=</span><span class="fl">0.5</span>))<span class="op">+</span>
<span class="st">                                      </span><span class="kw">ylab</span>(<span class="st">&quot;Count&quot;</span>)<span class="op">+</span><span class="kw">ggtitle</span>(<span class="st">&quot;Informative Cells per Sample&quot;</span>)<span class="op">+</span>
<span class="st">                                      </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="op">+</span>
<span class="st">                                      </span><span class="kw">scale_x_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>),<span class="dt">n.breaks=</span><span class="dv">8</span>)</code></pre></div>
<p><img src="scDNA_files/figure-html/unnamed-chunk-12-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>Next we want to make the co-occurence matrix on a sample level</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(cooccur)

### create matrix for oncoprint
mut_mat &lt;-<span class="st"> </span><span class="kw">table</span>(melted_mut_mat<span class="op">$</span>Sample,melted_mut_mat<span class="op">$</span>Gene)

### Prepare matrix for co occurence map
cooccur_mat &lt;-<span class="st"> </span><span class="kw">cooccur</span>(<span class="dt">mat=</span><span class="kw">t</span>(mut_mat), <span class="dt">type=</span><span class="st">&quot;spp_site&quot;</span>,
                       <span class="dt">only_effects =</span> <span class="ot">FALSE</span>,<span class="dt">eff_matrix=</span><span class="ot">TRUE</span>,
                       <span class="dt">thresh=</span><span class="ot">FALSE</span>, <span class="dt">eff_standard=</span><span class="ot">FALSE</span>,<span class="dt">spp_names=</span><span class="ot">TRUE</span>)<span class="op">$</span>results

## Denote which interactions are significantly inclusive or exclusive 
<span class="co"># The &#39;add_row&#39; function generates a new line, but it gets removed later.</span>
<span class="co"># This is helpful for setting the order of the gene labels below. </span>
cooccur_data_mat &lt;-<span class="st"> </span>cooccur_mat<span class="op">%&gt;%</span>
<span class="st">                        </span><span class="kw">mutate</span>(<span class="dt">score=</span><span class="kw">ifelse</span>(p_lt<span class="op">&lt;=</span><span class="fl">0.05</span>,<span class="op">-</span><span class="dv">1</span>,
                                            <span class="kw">ifelse</span>(p_gt<span class="op">&lt;=</span><span class="fl">0.05</span>,<span class="dv">1</span>,<span class="dv">0</span>))) <span class="op">%&gt;%</span>
<span class="st">                        </span><span class="kw">select</span>(sp1_name,sp2_name,score)<span class="op">%&gt;%</span>
<span class="st">                        </span><span class="kw">add_row</span>(<span class="dt">sp2_name=</span><span class="kw">setdiff</span>(.<span class="op">$</span>sp1_name,.<span class="op">$</span>sp2_name),
                                <span class="dt">sp1_name=</span><span class="kw">setdiff</span>(.<span class="op">$</span>sp2_name,.<span class="op">$</span>sp1_name),
                                <span class="dt">score=</span><span class="dv">0</span>)

<span class="co">#check out the final that we added so we can remove it later</span>
<span class="kw">tail</span>(cooccur_data_mat) 

<span class="co"># Order the genes in a coherent pattern for triangle strucutre of graph.</span>
cooccur_data_mat<span class="op">$</span>sp1_name&lt;-<span class="kw">factor</span>(cooccur_data_mat<span class="op">$</span>sp1_name,
                                  <span class="dt">levels=</span><span class="kw">unique</span>(cooccur_data_mat<span class="op">$</span>sp1_name))
cooccur_data_mat<span class="op">$</span>sp2_name&lt;-<span class="kw">factor</span>(cooccur_data_mat<span class="op">$</span>sp2_name,
                                  <span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">levels</span>(cooccur_data_mat<span class="op">$</span>sp1_name)))
  
<span class="co"># Triangle heatmap to compare cohorts</span>
grob_corrplot&lt;-<span class="kw">ggplot</span>(cooccur_data_mat<span class="op">%&gt;%</span><span class="kw">filter</span>(sp1_name<span class="op">!=</span><span class="st">&quot;BRAF&quot;</span>),<span class="kw">aes</span>(<span class="dt">x=</span>sp1_name,<span class="dt">y=</span>sp2_name))<span class="op">+</span>
<span class="st">                      </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> <span class="kw">factor</span>(score)), <span class="dt">color=</span><span class="st">&#39;grey90&#39;</span>) <span class="op">+</span>
<span class="st">                      </span><span class="kw">scale_fill_manual</span>(<span class="dt">name=</span><span class="st">&quot;Correlation&quot;</span>,
                                        <span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;-1&quot;</span>=<span class="st">&quot;firebrick3&quot;</span>,
                                                 <span class="st">&quot;0&quot;</span>=<span class="st">&quot;white&quot;</span>,
                                                 <span class="st">&quot;1&quot;</span>=<span class="st">&quot;steelblue2&quot;</span>),
                                        <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Mutually Exclusive&quot;</span>,
                                                 <span class="st">&quot;Not Significant&quot;</span>,
                                                 <span class="st">&quot;Mutually Inclusive&quot;</span>))<span class="op">+</span>
<span class="st">                      </span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">10</span>)<span class="op">+</span>
<span class="st">                      </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span>
<span class="st">                      </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">45</span>,<span class="dt">hjust=</span><span class="dv">1</span>,<span class="dt">vjust=</span><span class="dv">1</span>),
                            <span class="dt">axis.line =</span> <span class="kw">element_blank</span>(),
                            <span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>), 
                            <span class="dt">legend.justification =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),
                            <span class="dt">legend.direction =</span> <span class="st">&quot;vertical&quot;</span>)<span class="op">+</span>
<span class="st">                      </span><span class="kw">theme</span>(<span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>,<span class="st">&quot;line&quot;</span>))

grob_corrplot</code></pre></div>
<p><img src="scDNA_files/figure-html/unnamed-chunk-13-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cohort-description-and-exploration.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="oncoprint-figure-1a.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scDNA.pdf", "scDNA.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
