<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.2 Figure 2: Clonality | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</title>
  <meta name="description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  <meta name="generator" content="bookdown 0.18.3 and GitBook 2.6.7" />

  <meta property="og:title" content="3.2 Figure 2: Clonality | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.2 Figure 2: Clonality | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  
  <meta name="twitter:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

<meta name="author" content="Robert Bowman" />


<meta name="date" content="2020-05-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-1-cohort-characterization.html"/>
<link rel="next" href="final-words.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreward</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="data-extraction-and-setup.html"><a href="data-extraction-and-setup.html"><i class="fa fa-check"></i><b>2.1</b> Data Extraction and Setup</a></li>
<li class="chapter" data-level="2.2" data-path="tapestri-library.html"><a href="tapestri-library.html"><i class="fa fa-check"></i><b>2.2</b> Tapestri Package</a></li>
<li class="chapter" data-level="2.3" data-path="post-processing.html"><a href="post-processing.html"><i class="fa fa-check"></i><b>2.3</b> Post processing</a></li>
<li class="chapter" data-level="2.4" data-path="assessing-clonal-abundance.html"><a href="assessing-clonal-abundance.html"><i class="fa fa-check"></i><b>2.4</b> Assessing clonal abundance</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="manuscript-analysis.html"><a href="manuscript-analysis.html"><i class="fa fa-check"></i><b>3</b> Manuscript analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html"><i class="fa fa-check"></i><b>3.1</b> Figure 1: Cohort characterization</a><ul>
<li class="chapter" data-level="3.1.1" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#cohort-characteristics-ef1a-d"><i class="fa fa-check"></i><b>3.1.1</b> Cohort characteristics (EF1A-D)</a></li>
<li class="chapter" data-level="3.1.2" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#mutation-co-occurence"><i class="fa fa-check"></i><b>3.1.2</b> Mutation Co-occurence</a></li>
<li class="chapter" data-level="3.1.3" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#oncoprint-figure-1a"><i class="fa fa-check"></i><b>3.1.3</b> Oncoprint: Figure 1A</a></li>
<li class="chapter" data-level="3.1.4" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#sample-clonality-figure-1ce-figure-2ab"><i class="fa fa-check"></i><b>3.1.4</b> Sample clonality: Figure 1C,E Figure 2A,B</a></li>
<li class="chapter" data-level="3.1.5" data-path="figure-1-cohort-characterization.html"><a href="figure-1-cohort-characterization.html#clonograph-figure-1d"><i class="fa fa-check"></i><b>3.1.5</b> Clonograph: Figure 1D</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html"><i class="fa fa-check"></i><b>3.2</b> Figure 2: Clonality</a><ul>
<li class="chapter" data-level="3.2.1" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#clonality-in-disease-states"><i class="fa fa-check"></i><b>3.2.1</b> Clonality in disease states</a></li>
<li class="chapter" data-level="3.2.2" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#clonality-in-co-mutational-groups"><i class="fa fa-check"></i><b>3.2.2</b> Clonality in co-mutational groups</a></li>
<li class="chapter" data-level="3.2.3" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#clonality-assocation-with-vaf"><i class="fa fa-check"></i><b>3.2.3</b> Clonality assocation with VAF</a></li>
<li class="chapter" data-level="3.2.4" data-path="figure-2-clonality.html"><a href="figure-2-clonality.html#co-mutation-and-clonality"><i class="fa fa-check"></i><b>3.2.4</b> Co-mutation and clonality</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>4</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="5" data-path="Appendix.html"><a href="Appendix.html"><i class="fa fa-check"></i><b>5</b> Appendix</a><ul>
<li class="chapter" data-level="5.1" data-path="AppendixA.html"><a href="AppendixA.html"><i class="fa fa-check"></i><b>5.1</b> Import from Tapestri Insights</a></li>
<li class="chapter" data-level="5.2" data-path="functions.html"><a href="functions.html"><i class="fa fa-check"></i><b>5.2</b> Functions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-2-clonality" class="section level2">
<h2><span class="header-section-number">3.2</span> Figure 2: Clonality</h2>
<p>We next wanted to determine if a given gene is likely to be found in the dominant clone or only ib subclones. Of course, nearly all mutations found in the dominant clone were also found in subclones. We attempted to find mutations that were exclusively found in the dominant clone, but did not observe any. This is likely complicated by the fact that many subclones are not the most “genetically dense” clone as explored in Figure 3 of the manuscript. We decided to address that with the markov models described elsewhere, and instead classified a mutation on whether it was present present or absent in the dominant clone.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(tidyr)
<span class="kw">library</span>(dplyr)
<span class="kw">library</span>(RColorBrewer)
<span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(pals)
<span class="kw">library</span>(cowplot)
<span class="kw">setwd</span>(<span class="st">&quot;/Users/bowmanr/Projects/scDNA&quot;</span>)
final_sample_summary&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;/Users/bowmanr/Projects/scDNA/data/final_sample_summary.rds&quot;</span>)
pheno&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;./data/pheno.rds&quot;</span>)


clone_size_by_gene&lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(<span class="kw">names</span>(final_sample_summary),<span class="cf">function</span>(x){
  <span class="co"># select the clones</span>
  clones&lt;-final_sample_summary[[x]]<span class="op">$</span>Clones<span class="op">%&gt;%</span><span class="kw">select</span>(Clone)
  
  <span class="co"># Compute the relative clone size</span>
  Clone_size&lt;-final_sample_summary[[x]]<span class="op">$</span>Clones<span class="op">%&gt;%</span><span class="kw">transmute</span>(<span class="dt">Clone_size=</span>Count<span class="op">/</span><span class="kw">sum</span>(Count))<span class="op">%&gt;%</span><span class="kw">pull</span>(Clone_size)
  <span class="kw">names</span>(Clone_size) &lt;-<span class="st"> </span>clones[,<span class="st">&quot;Clone&quot;</span>]
  
  <span class="co"># Extract the mutations</span>
  mutations &lt;-<span class="st"> </span><span class="kw">colnames</span>(final_sample_summary[[x]]<span class="op">$</span>NGT<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="op">!</span>Clone))
  
  <span class="co"># Identify the dominant clone, which is the last one in this data frame</span>
  dominant_clone &lt;-<span class="st"> </span>clones[<span class="kw">nrow</span>(clones),]
  
  <span class="co"># Compute the bulk VAF for each mutation</span>
  VAFs &lt;-<span class="st"> </span>final_sample_summary[[x]]<span class="op">$</span>NGT<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="st">`</span><span class="dt">mutations</span><span class="st">`</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">summarise_all</span>(<span class="kw">funs</span>(<span class="kw">mean</span>(.)<span class="op">/</span><span class="dv">2</span>))

  <span class="co"># Separate the clone into a deduplicated NGT matrix</span>
  mut_mat&lt;-final_sample_summary[[x]]<span class="op">$</span>Clones<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">select</span>(Clone)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">separate</span>(<span class="dt">col=</span>Clone,<span class="dt">into=</span>mutations,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)
  
  <span class="co"># Create a composite data frame and turn it into long form</span>
  <span class="kw">data.frame</span>(clones,Clone_size,mut_mat,<span class="st">&quot;Sample&quot;</span>=x)<span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="kw">all_of</span>(mutations),<span class="dt">names_to=</span><span class="st">&quot;Variant&quot;</span>, <span class="dt">values_to=</span><span class="st">&quot;Genotype&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">            </span>
<span class="st">            </span><span class="kw">filter</span>(Genotype<span class="op">!=</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="st"> </span><span class="co"># remove WT entries</span>
<span class="st">            </span><span class="kw">separate</span>(<span class="dt">col=</span>Variant, <span class="dt">into=</span><span class="st">&quot;Gene&quot;</span>,<span class="dt">extra=</span><span class="st">&quot;drop&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\\</span><span class="st">.|_&quot;</span>,<span class="dt">remove=</span><span class="ot">FALSE</span>)<span class="op">%&gt;%</span><span class="st"> </span><span class="co"># For later useage in plotting</span>
<span class="st">            </span><span class="kw">group_by</span>(Variant)<span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">            </span><span class="kw">filter</span>(Clone_size<span class="op">==</span><span class="kw">max</span>(Clone_size))<span class="op">%&gt;%</span><span class="st"> </span><span class="co">#identify largest clone</span>
<span class="st">            </span><span class="kw">mutate</span>(<span class="dt">Clonality=</span><span class="kw">case_when</span>(
                          Clone<span class="op">==</span><span class="st">`</span><span class="dt">dominant_clone</span><span class="st">`</span><span class="op">~</span><span class="st">&quot;Dominant&quot;</span>,
                          Clone<span class="op">!=</span><span class="st">`</span><span class="dt">dominant_clone</span><span class="st">`</span><span class="op">~</span><span class="st">&quot;Subclone&quot;</span>))<span class="op">%&gt;%</span><span class="st"> </span><span class="co">#label clones</span>
<span class="st">            </span><span class="kw">inner_join</span>(<span class="kw">data.frame</span>(<span class="st">&quot;VAF&quot;</span>=<span class="kw">t</span>(VAFs),
                                  <span class="st">&quot;Variant&quot;</span>=<span class="kw">names</span>(VAFs))) <span class="co"># merge with bulk VAF info</span>
}))

<span class="co"># Tally the number of times a gene is in the dominant and subclone</span>
tally_set&lt;-<span class="kw">data.frame</span>(<span class="kw">table</span>(clone_size_by_gene<span class="op">$</span>Gene,
                           clone_size_by_gene<span class="op">$</span>Clonality))<span class="op">%&gt;%</span>
<span class="st">                      </span><span class="kw">pivot_wider</span>(<span class="dt">names_from=</span>Var2,<span class="dt">values_from=</span>Freq)<span class="op">%&gt;%</span>
<span class="st">                      </span><span class="kw">mutate</span>(<span class="dt">Ratio=</span>Dominant<span class="op">/</span>(Subclone<span class="op">+</span>Dominant))<span class="op">%&gt;%</span><span class="st"> </span><span class="co">#calculate the dominant ratio</span>
<span class="st">                      </span><span class="kw">arrange</span>(Ratio) 
  
<span class="co"># For plotting purposes establish order of the y axis</span>
clone_size_by_gene<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">factor</span>(clone_size_by_gene<span class="op">$</span>Gene, <span class="dt">levels=</span>tally_set<span class="op">$</span>Var1)

<span class="co"># Linde and I spent too much time picking the exact shade of red we wanted for this paper....</span>
color_red&lt;-<span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>]

<span class="co"># For plotting purposes establish order of stacked bars</span>
clone_size_by_gene<span class="op">$</span>Clonality&lt;-<span class="kw">factor</span>(clone_size_by_gene<span class="op">$</span>Clonality,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;Subclone&quot;</span>,<span class="st">&quot;Dominant&quot;</span>))

<span class="co">#plot the data</span>
ggA&lt;-<span class="kw">ggplot</span>(<span class="kw">tally</span>(clone_size_by_gene<span class="op">%&gt;%</span><span class="kw">group_by</span>(Gene,Clonality)),
            <span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">factor</span>(Gene),<span class="dt">fill=</span>Clonality,<span class="dt">y=</span>n,<span class="dt">label=</span>n))<span class="op">+</span>
<span class="st">            </span><span class="kw">guides</span>(<span class="dt">fill=</span><span class="ot">FALSE</span>,<span class="dt">color=</span><span class="ot">FALSE</span>)<span class="op">+</span>
<span class="st">            </span><span class="kw">scale_y_continuous</span>( <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.0</span>))<span class="op">+</span><span class="st"> </span><span class="co">#removes white space near the axis of the bars</span>
<span class="st">            </span><span class="kw">geom_bar</span>(<span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>,<span class="dt">position=</span><span class="st">&quot;fill&quot;</span>)<span class="op">+</span>
<span class="st">            </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span><span class="kw">coord_flip</span>()<span class="op">+</span>
<span class="st">            </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;Dominant&quot;</span>=color_red,
                                       <span class="st">&quot;Subclone&quot;</span>=<span class="st">&quot;grey80&quot;</span>))<span class="op">+</span>
<span class="st">            </span><span class="kw">ylab</span>(<span class="st">&quot;Fraction of mutant clones </span><span class="ch">\n</span><span class="st"> with mutation in dominant clone&quot;</span>)<span class="op">+</span>
<span class="st">            </span><span class="kw">theme_bw</span>(<span class="dt">base_size=</span><span class="dv">8</span>)<span class="op">+</span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)
  
ggB&lt;-<span class="kw">ggplot</span>(clone_size_by_gene, 
            <span class="kw">aes</span>(<span class="dt">y=</span>Clone_size, <span class="dt">x=</span>Gene, <span class="dt">fill=</span>Gene)) <span class="op">+</span>
<span class="st">            </span><span class="kw">geom_boxplot</span>(<span class="dt">alpha =</span> <span class="fl">0.5</span>,<span class="dt">outlier.shape =</span> <span class="ot">NA</span>)<span class="op">+</span>
<span class="st">            </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">color=</span>Clonality,<span class="dt">group=</span>Clonality), 
                       <span class="dt">position =</span> <span class="kw">position_jitterdodge</span>(), <span class="dt">size=</span><span class="fl">0.3</span>)<span class="op">+</span>
<span class="st">            </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">tol.rainbow</span>(<span class="dt">n=</span><span class="kw">length</span>(<span class="kw">levels</span>(clone_size_by_gene<span class="op">$</span>Gene))))<span class="op">+</span>
<span class="st">            </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;Dominant&quot;</span>=color_red,
                                        <span class="st">&quot;Subclone&quot;</span>=<span class="st">&quot;grey20&quot;</span>))<span class="op">+</span>
<span class="st">            </span><span class="kw">coord_flip</span>()<span class="op">+</span>
<span class="st">            </span><span class="kw">theme_bw</span>(<span class="dt">base_size=</span><span class="dv">8</span>)<span class="op">+</span><span class="kw">guides</span>(<span class="dt">fill=</span><span class="ot">FALSE</span>,<span class="dt">color=</span><span class="ot">FALSE</span>)<span class="op">+</span>
<span class="st">            </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_blank</span>(),
                  <span class="dt">axis.ticks.y  =</span> <span class="kw">element_blank</span>(),
                  <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>())<span class="op">+</span>
<span class="st">            </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.05</span>)) <span class="op">+</span>
<span class="st">            </span><span class="kw">ylab</span>(<span class="st">&quot;Fraction of cells </span><span class="ch">\n</span><span class="st"> in largest mutant clone&quot;</span>)<span class="op">+</span>
<span class="st">            </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)

spacer &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(<span class="ot">NULL</span>) <span class="co"># plot looks better with a little spacer</span>

<span class="kw">plot_grid</span>(ggA,spacer,ggB,<span class="dt">align=</span><span class="st">&quot;h&quot;</span>,<span class="dt">axis=</span><span class="st">&quot;tb&quot;</span>,
          <span class="dt">ncol=</span><span class="dv">3</span>,<span class="dt">rel_widths=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="fl">0.05</span>,<span class="dv">1</span>))</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-27"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-27-1.png" alt="Miles et al. Figure 2C" width="672" />
<p class="caption">
Figure 3.6: Miles et al. Figure 2C
</p>
</div>
<div id="clonality-in-disease-states" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Clonality in disease states</h3>
<p>We can look at this in a little more resolution on individual genes across disease course.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(magrittr)

<span class="co"># Incorporate Diagnosis and disease state</span>
clone_size_by_gene_Dx&lt;-<span class="kw">inner_join</span>(clone_size_by_gene,pheno)

<span class="co"># We focused on a subset of genes</span>
genes_of_interest &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;TET2&quot;</span>,<span class="st">&quot;ASXL1&quot;</span>,<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;IDH2&quot;</span>,
                       <span class="st">&quot;JAK2&quot;</span>,<span class="st">&quot;NRAS&quot;</span>,<span class="st">&quot;KRAS&quot;</span>,<span class="st">&quot;FLT3&quot;</span>,<span class="st">&quot;NPM1&quot;</span>)

<span class="co"># We had an interest in DNMT3A R882 point mutants, so we can extract those out</span>
clone_size_by_gene_Dx<span class="op">%&lt;&gt;%</span><span class="kw">mutate</span>(<span class="dt">Gene=</span><span class="kw">case_when</span>(
                          <span class="kw">grepl</span>(<span class="st">&quot;DNMT3A.p.R882&quot;</span>,Variant)<span class="op">~</span><span class="st">&quot;DNMT3A.p.R882&quot;</span>,
                          <span class="ot">TRUE</span><span class="op">~</span><span class="kw">as.character</span>(Gene)))

clone_size_by_gene_Dx<span class="op">$</span>Dx &lt;-<span class="st"> </span><span class="kw">factor</span>(clone_size_by_gene_Dx<span class="op">$</span>Dx,
                                   <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;CH&quot;</span>,<span class="st">&quot;MPN&quot;</span>,<span class="st">&quot;Other&quot;</span>,<span class="st">&quot;sAML&quot;</span>,<span class="st">&quot;tAML&quot;</span>,<span class="st">&quot;AML&quot;</span>))

mutation_dominance_by_dx&lt;-<span class="kw">ggplot</span>(<span class="kw">tally</span>(clone_size_by_gene_Dx<span class="op">%&gt;%</span>
<span class="st">                        </span><span class="kw">filter</span>(Gene<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;TET2&quot;</span>,<span class="st">&quot;ASXL1&quot;</span>,
                                         <span class="st">&quot;DNMT3A.p.R882&quot;</span>,<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;IDH2&quot;</span>))<span class="op">%&gt;%</span><span class="st">                   </span>
<span class="st">                        </span><span class="kw">group_by</span>(Gene,Dx,Clonality)) ,
                        <span class="kw">aes</span>(<span class="dt">x=</span>Dx,<span class="dt">fill=</span>Clonality,<span class="dt">y=</span>n)) <span class="op">+</span>
<span class="st">                        </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">factor</span>(Gene,
                                            <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;TET2&quot;</span>,<span class="st">&quot;ASXL1&quot;</span>,
                                                     <span class="st">&quot;DNMT3A.p.R882&quot;</span>,<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;IDH2&quot;</span>)),<span class="dt">ncol=</span><span class="dv">3</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">geom_col</span>(<span class="dt">position=</span><span class="st">&quot;fill&quot;</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;Dominant&quot;</span>=color_red,
                                                 <span class="st">&quot;Subclone&quot;</span>=<span class="st">&quot;grey80&quot;</span>))<span class="op">+</span>
<span class="st">                        </span><span class="kw">ylab</span>(<span class="st">&quot;Number of samples&quot;</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">theme_bw</span>(<span class="dt">base_size=</span><span class="dv">10</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>,
                            <span class="dt">axis.text.x =</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">30</span>,<span class="dt">hjust=</span><span class="dv">1</span>))
mutation_dominance_by_dx</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-28"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-28-1.png" alt="Clonality by Diagnosis" width="672" />
<p class="caption">
Figure 3.7: Clonality by Diagnosis
</p>
</div>
</div>
<div id="clonality-in-co-mutational-groups" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Clonality in co-mutational groups</h3>
<p>We can also look at it across the DTAI /RAS/FLT3 groups we defined in the paper. Here includes a little code from earlier in the tutorial to set up those groups again</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mutants_in_each_sample&lt;-<span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(<span class="kw">names</span>(final_sample_summary),<span class="cf">function</span>(x){
  y&lt;-<span class="kw">colnames</span>(final_sample_summary[[x]]<span class="op">$</span>NGT)
  z &lt;-<span class="st"> </span><span class="kw">list</span>()
  z<span class="op">$</span>Sample &lt;-<span class="st"> </span>x
  z<span class="op">$</span>DNMT3A &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;DNMT3A&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)
  z<span class="op">$</span>TET2 &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;TET2&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)
  z<span class="op">$</span>ASXL1 &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;ASXL1&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)
  z<span class="op">$</span>IDH &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;IDH&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)
  z<span class="op">$</span>FLT3 &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;FLT3&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)
  z<span class="op">$</span>KIT &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;KIT&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>) <span class="co"># n=1 sample, we put it in the &quot;signalling category&quot;</span>
  z<span class="op">$</span>RAS &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;RAS&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)
  z<span class="op">$</span>JAK2 &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;JAK2&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)
  z<span class="op">$</span>PTPN11 &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;PTPN11&quot;</span>,y)),<span class="dv">1</span>,<span class="dv">0</span>)
  <span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">do.call</span>(rbind,z)))
}))

<span class="co"># Bin into groups based on mutations and disease type</span>
mutants_in_each_sample<span class="op">%&lt;&gt;%</span><span class="kw">mutate</span>(<span class="dt">Group=</span><span class="kw">case_when</span>(
                          (TET2<span class="op">==</span><span class="dv">1</span><span class="op">|</span>DNMT3A<span class="op">==</span><span class="dv">1</span><span class="op">|</span>IDH<span class="op">==</span><span class="dv">1</span><span class="op">|</span>ASXL1<span class="op">==</span><span class="dv">1</span>)<span class="op">&amp;</span>(RAS<span class="op">==</span><span class="dv">0</span><span class="op">&amp;</span>FLT3<span class="op">==</span><span class="dv">0</span>)<span class="op">~</span><span class="st">&#39;DTAI&#39;</span>,
                          (TET2<span class="op">==</span><span class="dv">1</span><span class="op">|</span>DNMT3A<span class="op">==</span><span class="dv">1</span><span class="op">|</span>IDH<span class="op">==</span><span class="dv">1</span><span class="op">|</span>ASXL1<span class="op">==</span><span class="dv">1</span>)<span class="op">&amp;</span>((RAS<span class="op">==</span><span class="dv">1</span><span class="op">&amp;</span>FLT3<span class="op">==</span><span class="dv">0</span>)<span class="op">|</span>
<span class="st">                                                                   </span>(PTPN11<span class="op">==</span><span class="dv">1</span><span class="op">&amp;</span>FLT3<span class="op">==</span><span class="dv">0</span>))<span class="op">~</span><span class="st">&#39;DTAI-RAS&#39;</span>,
                          (TET2<span class="op">==</span><span class="dv">1</span><span class="op">|</span>DNMT3A<span class="op">==</span><span class="dv">1</span><span class="op">|</span>IDH<span class="op">==</span><span class="dv">1</span><span class="op">|</span>ASXL1<span class="op">==</span><span class="dv">1</span>)<span class="op">&amp;</span>(RAS<span class="op">==</span><span class="dv">0</span><span class="op">&amp;</span>FLT3<span class="op">==</span><span class="dv">1</span>)<span class="op">~</span><span class="st">&#39;DTAI-FLT3&#39;</span>,
                          (TET2<span class="op">==</span><span class="dv">1</span><span class="op">|</span>DNMT3A<span class="op">==</span><span class="dv">1</span><span class="op">|</span>IDH<span class="op">==</span><span class="dv">1</span><span class="op">|</span>ASXL1<span class="op">==</span><span class="dv">1</span>)<span class="op">&amp;</span>((RAS<span class="op">==</span><span class="dv">1</span><span class="op">&amp;</span>FLT3<span class="op">==</span><span class="dv">1</span>)<span class="op">|</span>
<span class="st">                                                               </span>(PTPN11<span class="op">==</span><span class="dv">1</span><span class="op">&amp;</span>FLT3<span class="op">==</span><span class="dv">1</span>))<span class="op">~</span><span class="st">&#39;DTAI-FLT3-RAS&#39;</span>,
                          (TET2<span class="op">==</span><span class="dv">0</span><span class="op">&amp;</span>DNMT3A<span class="op">==</span><span class="dv">0</span><span class="op">&amp;</span>IDH<span class="op">==</span><span class="dv">0</span><span class="op">&amp;</span>ASXL1<span class="op">==</span><span class="dv">0</span>)<span class="op">&amp;</span>(RAS<span class="op">==</span><span class="dv">1</span><span class="op">|</span>FLT3<span class="op">==</span><span class="dv">1</span><span class="op">|</span>JAK2<span class="op">==</span><span class="dv">1</span><span class="op">|</span>KIT<span class="op">==</span><span class="dv">1</span>)<span class="op">~</span><span class="st">&#39;Signaling&#39;</span>))<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">left_join</span>(pheno,<span class="dt">by=</span><span class="st">&quot;Sample&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">mutate</span>(<span class="dt">Final_group=</span><span class="kw">case_when</span>(
                                          <span class="kw">grepl</span>(<span class="st">&quot;AML|Other&quot;</span>,Dx)<span class="op">~</span>Group,
                                          <span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;AML|Other&quot;</span>,Dx)<span class="op">~</span>Dx
                                        ))

<span class="co"># Order the groups to match how we have them in the paper</span>
mutants_in_each_sample<span class="op">$</span>Final_group &lt;-<span class="st"> </span><span class="kw">factor</span>(mutants_in_each_sample<span class="op">$</span>Final_group,
                                              <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;CH&quot;</span>,<span class="st">&quot;MPN&quot;</span>,<span class="st">&quot;Signaling&quot;</span>,<span class="st">&quot;DTAI&quot;</span>,
                                                       <span class="st">&quot;DTAI-RAS&quot;</span>,<span class="st">&quot;DTAI-FLT3&quot;</span>,<span class="st">&quot;DTAI-FLT3-RAS&quot;</span>))

<span class="co"># Now merge this with our new data frame defined above</span>

clone_size_by_gene_DTAI&lt;-<span class="kw">left_join</span>(clone_size_by_gene_Dx,mutants_in_each_sample,<span class="dt">by=</span><span class="st">&quot;Sample&quot;</span>)

mutation_dominance_by_DTAI&lt;-<span class="kw">ggplot</span>(<span class="kw">tally</span>(clone_size_by_gene_DTAI<span class="op">%&gt;%</span>
<span class="st">                                    </span><span class="kw">filter</span>(Gene<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;TET2&quot;</span>,<span class="st">&quot;ASXL1&quot;</span>,
                                                     <span class="st">&quot;DNMT3A.p.R882&quot;</span>,<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;IDH2&quot;</span>))<span class="op">%&gt;%</span>
<span class="st">                                    </span><span class="kw">group_by</span>(Gene,Final_group,Clonality)) ,
                        <span class="kw">aes</span>(<span class="dt">x=</span>Final_group,<span class="dt">fill=</span>Clonality,<span class="dt">y=</span>n)) <span class="op">+</span>
<span class="st">                        </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">factor</span>(Gene,
                                            <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;TET2&quot;</span>,<span class="st">&quot;ASXL1&quot;</span>,
                                                     <span class="st">&quot;DNMT3A.p.R882&quot;</span>,<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;IDH2&quot;</span>)),<span class="dt">ncol=</span><span class="dv">3</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">geom_col</span>()<span class="op">+</span>
<span class="st">                        </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;Dominant&quot;</span>=color_red,
                                                 <span class="st">&quot;Subclone&quot;</span>=<span class="st">&quot;grey80&quot;</span>))<span class="op">+</span>
<span class="st">                        </span><span class="kw">ylab</span>(<span class="st">&quot;Number of samples&quot;</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">theme_bw</span>(<span class="dt">base_size=</span><span class="dv">10</span>)<span class="op">+</span>
<span class="st">                        </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>,
                            <span class="dt">axis.text.x =</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">30</span>,<span class="dt">hjust=</span><span class="dv">1</span>))
mutation_dominance_by_DTAI</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-29"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-29-1.png" alt="Miles et al Extended Figure 3C" width="672" />
<p class="caption">
Figure 3.8: Miles et al Extended Figure 3C
</p>
</div>
</div>
<div id="clonality-assocation-with-vaf" class="section level3">
<h3><span class="header-section-number">3.2.3</span> Clonality assocation with VAF</h3>
<p>We can also look at the VAF of each gene and see if that is associated with dominant vs subclone status for each state of disease, or the grouping we setup in the main figure</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggbeeswarm)
pheno&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;./data/pheno.rds&quot;</span>)

data_to_plot&lt;-<span class="kw">inner_join</span>(clone_size_by_gene,pheno)<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">filter</span>(<span class="kw">as.character</span>(Gene)<span class="op">%in%</span>genes_of_interest <span class="op">&amp;</span>
<span class="st">                                </span><span class="op">!</span>Dx<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;CH&quot;</span>))<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">group_by</span>(Gene,Clonality)

summarized_data &lt;-data_to_plot<span class="op">%&gt;%</span><span class="kw">summarise</span>(<span class="dt">mean=</span><span class="kw">mean</span>(VAF),            
                                           <span class="dt">sd =</span> <span class="kw">sd</span>(VAF),
                                           <span class="dt">sem =</span> <span class="kw">sd</span>(VAF)<span class="op">/</span><span class="kw">sqrt</span>(<span class="kw">length</span>(VAF)))


clonality_VAF&lt;-<span class="kw">ggplot</span>(data_to_plot,<span class="kw">aes</span>(<span class="dt">x=</span>Clonality,<span class="dt">y=</span>VAF,<span class="dt">color=</span>Clonality))<span class="op">+</span>
<span class="st">                    </span><span class="kw">facet_wrap</span>(<span class="op">~</span><span class="kw">factor</span>(Gene,<span class="dt">levels=</span>genes_of_interest),
                                <span class="dt">scale=</span><span class="st">&quot;free_x&quot;</span>,<span class="dt">ncol=</span><span class="dv">5</span>)<span class="op">+</span>
<span class="st">                    </span>ggbeeswarm<span class="op">::</span><span class="kw">geom_beeswarm</span>()<span class="op">+</span>
<span class="st">                    </span><span class="kw">geom_errorbar</span>(<span class="dt">data=</span>summarized_data,<span class="kw">aes</span>(<span class="dt">x=</span>Clonality,
                                                           <span class="dt">y=</span>mean,
                                                           <span class="dt">ymin=</span>mean<span class="op">-</span>sem,
                                                           <span class="dt">ymax=</span>mean<span class="op">+</span>sem),
                                                           <span class="dt">color=</span><span class="st">&quot;black&quot;</span>)<span class="op">+</span>
<span class="st">                    </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;Dominant&quot;</span>=color_red,
                                                <span class="st">&quot;Subclone&quot;</span>=<span class="st">&quot;grey50&quot;</span>))<span class="op">+</span>
<span class="st">                    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span><span class="st">        </span><span class="kw">ylab</span>(<span class="st">&quot;Computed VAF&quot;</span>)<span class="op">+</span>
<span class="st">                    </span><span class="kw">theme_classic</span>()<span class="op">+</span><span class="kw">guides</span>(<span class="dt">fill=</span><span class="ot">FALSE</span>)<span class="op">+</span>
<span class="st">                    </span><span class="kw">theme</span>(<span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(),
                          <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>())<span class="op">+</span>
<span class="st">                    </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="fl">1.1</span>),
                                       <span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">0</span>,.<span class="dv">25</span>,.<span class="dv">5</span>,.<span class="dv">75</span>,<span class="dv">1</span>),
                                   <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;0&quot;</span>,<span class="st">&quot;0.25&quot;</span>,<span class="st">&quot;0.50&quot;</span>,<span class="st">&quot;0.75&quot;</span>,<span class="st">&quot;1.0&quot;</span>))

clonality_VAF</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-30"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-30-1.png" alt="Miles et al Extended Figure 3D" width="672" />
<p class="caption">
Figure 3.9: Miles et al Extended Figure 3D
</p>
</div>
<p>Statistics below</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(broom)
clonality_VAF_pvalues&lt;-<span class="kw">data.frame</span>(data_to_plot)<span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">                                </span><span class="kw">filter</span>(<span class="kw">as.character</span>(Clonality)<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Dominant&quot;</span>,<span class="st">&quot;Subclone&quot;</span>)<span class="op">&amp;</span>
<span class="st">                                         </span>Gene<span class="op">!=</span><span class="st">&quot;IDH2&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                                </span><span class="kw">group_by</span>(Gene)<span class="op">%&gt;%</span>
<span class="st">                                </span><span class="kw">select</span>(VAF,Clonality)<span class="op">%&gt;%</span>
<span class="st">                                </span><span class="kw">do</span>(<span class="kw">tidy</span>(<span class="kw">t.test</span>(VAF <span class="op">~</span><span class="st"> </span>Clonality, <span class="dt">data =</span> .)))<span class="op">%&gt;%</span>
<span class="st">                                </span><span class="kw">select</span>(Gene,<span class="dt">Dominant_VAF=</span>estimate2,<span class="dt">Subclone_VAF=</span>estimate1,
                                       p.value)<span class="op">%&gt;%</span>
<span class="st">                                </span><span class="kw">mutate_if</span>(is.numeric, <span class="kw">funs</span>(<span class="kw">as.character</span>(<span class="kw">signif</span>(., <span class="dv">3</span>))))</code></pre></div>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Gene
</th>
<th style="text-align:left;">
Dominant_VAF
</th>
<th style="text-align:left;">
Subclone_VAF
</th>
<th style="text-align:left;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
KRAS
</td>
<td style="text-align:left;">
0.387
</td>
<td style="text-align:left;">
0.0601
</td>
<td style="text-align:left;">
0.00353
</td>
</tr>
<tr>
<td style="text-align:left;">
FLT3
</td>
<td style="text-align:left;">
0.313
</td>
<td style="text-align:left;">
0.0567
</td>
<td style="text-align:left;">
7.9e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
NRAS
</td>
<td style="text-align:left;">
0.41
</td>
<td style="text-align:left;">
0.0507
</td>
<td style="text-align:left;">
1.2e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
TET2
</td>
<td style="text-align:left;">
0.546
</td>
<td style="text-align:left;">
0.0756
</td>
<td style="text-align:left;">
9.37e-17
</td>
</tr>
<tr>
<td style="text-align:left;">
ASXL1
</td>
<td style="text-align:left;">
0.445
</td>
<td style="text-align:left;">
0.109
</td>
<td style="text-align:left;">
3.43e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
DNMT3A
</td>
<td style="text-align:left;">
0.481
</td>
<td style="text-align:left;">
0.124
</td>
<td style="text-align:left;">
1.06e-09
</td>
</tr>
<tr>
<td style="text-align:left;">
JAK2
</td>
<td style="text-align:left;">
0.725
</td>
<td style="text-align:left;">
0.0936
</td>
<td style="text-align:left;">
2.05e-07
</td>
</tr>
<tr>
<td style="text-align:left;">
IDH1
</td>
<td style="text-align:left;">
0.471
</td>
<td style="text-align:left;">
0.0673
</td>
<td style="text-align:left;">
0.000209
</td>
</tr>
<tr>
<td style="text-align:left;">
NPM1
</td>
<td style="text-align:left;">
0.465
</td>
<td style="text-align:left;">
0.278
</td>
<td style="text-align:left;">
0.0907
</td>
</tr>
</tbody>
</table>

</div>
<div id="co-mutation-and-clonality" class="section level3">
<h3><span class="header-section-number">3.2.4</span> Co-mutation and clonality</h3>
<!--  
I need to break this section up into cleaner chunks and comment each line now that it is working.
Next I need to finalize the display of the graph structures
Lastly, I need to determine how many pages I want this to be, and how nested I want this part to be.
-->
<p>Some of our motivations for this section were: 1) Are epigenetic mutations found in the same cell, if so, are these cells the dominant clone? 2) Are there Tet2 IDH in the same sample, and if so same cell? 3) Are there signaling in the same sample, and if so same cell?</p>
<p>We start off here by laoding the packages of interest, including two new ones “UpSetR” and “tibble”. UpSetR is necessary for some of the barplots we make, while “tibble” is useful for the add_column function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
<span class="kw">library</span>(UpSetR)
<span class="kw">library</span>(tidyr)
<span class="kw">library</span>(dplyr)
<span class="kw">library</span>(RColorBrewer)
<span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(pals)
<span class="kw">library</span>(cowplot)
<span class="kw">library</span>(tibble)

<span class="kw">setwd</span>(<span class="st">&quot;/Users/bowmanr/Projects/scDNA&quot;</span>)
final_sample_summary&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;./data/final_sample_summary.rds&quot;</span>)
pheno&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;./data/pheno.rds&quot;</span>)</code></pre></div>
<p>Our eventual goal is make an UpSet plot looking at the overlap of the different epigenetic events. We would like to demarcate on this plot whether the different epigenetic hits are in the same clone, with a specific focus on the dominant clone. We’ll start off by determing for each sample whether there is a mutation for the indicated genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sample_mutations&lt;-<span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(<span class="kw">names</span>(final_sample_summary),<span class="cf">function</span>(sample){
  <span class="kw">data.frame</span>(<span class="st">&quot;Sample&quot;</span>=sample,
            <span class="st">&quot;DNMT3A&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>),
             <span class="st">&quot;TET2&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;TET2&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>),
             <span class="st">&quot;IDH2&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;IDH2&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>),
             <span class="st">&quot;IDH1&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;IDH1&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>),
             <span class="st">&quot;ASXL1&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;ASXL1&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>),
             <span class="st">&quot;FLT3&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;FLT3&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>),
             <span class="st">&quot;JAK2&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;JAK2&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>),
             <span class="st">&quot;NRAS&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;NRAS&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>),
             <span class="st">&quot;KRAS&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;KRAS&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>),
             <span class="st">&quot;PTPN11&quot;</span>=<span class="kw">ifelse</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;PTPN11&quot;</span>,<span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT))),<span class="dv">1</span>,<span class="dv">0</span>)
            )
}))</code></pre></div>
<p>We will next proceed with this same process on a clone level.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">clone_mutations&lt;-<span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(<span class="kw">names</span>(final_sample_summary),<span class="cf">function</span>(sample){
  
  <span class="co"># select the clones</span>
  clones&lt;-final_sample_summary[[sample]]<span class="op">$</span>Clones<span class="op">%&gt;%</span><span class="kw">select</span>(Clone)
  <span class="co"># Extract the mutations</span>
  mutations &lt;-<span class="st"> </span><span class="kw">colnames</span>(final_sample_summary[[sample]]<span class="op">$</span>NGT<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="op">!</span>Clone))
  
  <span class="co"># Separate clones into mutations, identify the mutations in the dominant clone.</span>
  <span class="co"># I&#39;m sure there is a more efficient way to do this, but this is what I have. </span>
  <span class="co"># This might seem a little redundant with the group ungroup and regroup, but the idea </span>
  <span class="co"># is we want to calculate the relative size of each clone in the context of the whole sample</span>
  <span class="co"># then identify and remove the WT clone in the rare event it is the dominant clone, or the largest subclone.</span>
  <span class="co"># In order to then mark the clones as dominant or subclone, we have to ungroup in order to use the context of the whole sample.</span>
  <span class="co"># Lastly, we group again on the Clones and check whether each clone is mutant for the genes of interest.</span>
  <span class="co"># We next remove excess columns representing genes and variants and finally remove duplicate rows. </span>
  <span class="co"># Previosuly each line of this data frame was a variant, and now it is a clone.</span>
  out&lt;-final_sample_summary[[sample]]<span class="op">$</span>Clones<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">mutate</span>(<span class="dt">Clone_size=</span>Count<span class="op">/</span><span class="kw">sum</span>(Count))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">select</span>(Clone,Clone_size)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">separate</span>(<span class="dt">col=</span>Clone,
                                       <span class="dt">into=</span>mutations,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,
                                       <span class="dt">remove=</span><span class="ot">FALSE</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span>mutations,
                                          <span class="dt">names_to=</span><span class="st">&quot;Variant&quot;</span>,
                                          <span class="dt">values_to=</span><span class="st">&quot;Genotype&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">add_column</span>(<span class="dt">Sample=</span><span class="st">`</span><span class="dt">sample</span><span class="st">`</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">group_by</span>(Clone)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">mutate</span>(<span class="dt">WT=</span><span class="kw">ifelse</span>(<span class="kw">all</span>(Genotype<span class="op">==</span><span class="dv">0</span>),<span class="dv">1</span>,<span class="dv">0</span>))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(WT<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(Genotype<span class="op">!=</span><span class="dv">0</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">ungroup</span>()<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">mutate</span>(<span class="dt">Clonality=</span><span class="kw">ifelse</span>(Clone_size<span class="op">==</span><span class="kw">max</span>(Clone_size),
                                                      <span class="st">&quot;Dominant&quot;</span>,<span class="st">&quot;Subclone&quot;</span>))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">group_by</span>(Clone)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">mutate</span>(<span class="dt">Gene=</span><span class="kw">do.call</span>(rbind,<span class="kw">strsplit</span>(Variant,<span class="st">&quot;[</span><span class="ch">\\</span><span class="st">._]&quot;</span>))[,<span class="dv">1</span>])<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">mutate</span>(<span class="dt">DNMT3A=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;DNMT3A&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>),
                                     <span class="dt">TET2=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;TET2&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>),
                                     <span class="dt">ASXL1=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;ASXL1&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>),
                                     <span class="dt">IDH1=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;IDH1&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>),
                                     <span class="dt">IDH2=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;IDH2&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>),
                                     <span class="dt">FLT3=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;FLT3&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>),
                                     <span class="dt">NRAS=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;NRAS&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>),
                                     <span class="dt">KRAS=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;KRAS&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>),
                                     <span class="dt">PTPN11=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;PTPN11&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>),
                                     <span class="dt">JAK2=</span><span class="kw">ifelse</span>(<span class="kw">any</span>(Gene<span class="op">%in%</span><span class="st">&quot;JAK2&quot;</span>),<span class="dv">1</span>,<span class="dv">0</span>))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">ungroup</span>()<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">select</span>(<span class="op">!</span><span class="kw">c</span>(Variant,Genotype,Gene))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">distinct</span>()
  }))</code></pre></div>
<p>Now to check to see if the mutations found in the sample are in the dominant clone. The UpSetR package is great, the queries part was a little non intuitive for me at first glance, however <a href="https://cran.r-project.org/web/packages/UpSetR/vignettes/queries.html">Vignette</a> cleared it up and is worth a read.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Identify genes of interest</span>
DTAI_genes&lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;TET2&quot;</span>,<span class="st">&quot;IDH2&quot;</span>,<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;ASXL1&quot;</span>)

<span class="co"># Subset data.frame above to only the dominant clones</span>
dominant_clone_mutations &lt;-<span class="st"> </span>clone_mutations<span class="op">%&gt;%</span><span class="kw">filter</span>(Clonality<span class="op">==</span><span class="st">&quot;Dominant&quot;</span>)

<span class="co"># For each sample, determine if the dominant clone</span>
sample_mutations<span class="op">$</span>Match&lt;-<span class="kw">ifelse</span>(<span class="kw">sapply</span>(sample_mutations<span class="op">$</span>Sample,<span class="cf">function</span>(sample) {
                                                <span class="kw">all</span>(sample_mutations<span class="op">%&gt;%</span>
<span class="st">                                                        </span><span class="kw">filter</span>(Sample<span class="op">==</span>sample)<span class="op">%&gt;%</span>
<span class="st">                                                        </span><span class="kw">select</span>(<span class="kw">all_of</span>(DTAI_genes))<span class="op">==</span>
<span class="st">                                                    </span>dominant_clone_mutations<span class="op">%&gt;%</span>
<span class="st">                                                        </span><span class="kw">filter</span>(Sample<span class="op">==</span>sample)<span class="op">%&gt;%</span>
<span class="st">                                                        </span><span class="kw">select</span>(<span class="kw">all_of</span>(DTAI_genes)))
                          }) ,<span class="st">&quot;Match&quot;</span>,<span class="st">&quot;Absent&quot;</span>)

<span class="co"># Join this match with the phenotype data to pick a disease state of interest</span>
test_set&lt;-<span class="kw">left_join</span>(sample_mutations,pheno,<span class="dt">by=</span><span class="st">&quot;Sample&quot;</span>)

<span class="co"># Necessary little function to </span>
Myfunc &lt;-<span class="st"> </span><span class="cf">function</span>(row,feature) {
  data &lt;-<span class="st"> </span>(row[feature]<span class="op">==</span><span class="st">&quot;Match&quot;</span>)
}

<span class="co"># Slight difference from BioRxiv paper due to a misclassification of one TET2 mutant CMML sample as AML.</span>
AML&lt;-<span class="kw">upset</span>(test_set<span class="op">%&gt;%</span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;AML&quot;</span>,Dx)), 
           <span class="dt">sets=</span>DTAI_genes,
           <span class="dt">order.by =</span> <span class="kw">c</span>(<span class="st">&quot;degree&quot;</span>),
           <span class="dt">main.bar.color =</span> <span class="st">&quot;grey60&quot;</span>,<span class="dt">decreasing=</span><span class="ot">FALSE</span>,
           <span class="dt">mainbar.y.label =</span> <span class="st">&quot;Number of samples&quot;</span>,
           <span class="dt">sets.x.label =</span> <span class="st">&quot;Number of </span><span class="ch">\n</span><span class="st"> mutant samples&quot;</span>,
           <span class="dt">text.scale=</span><span class="fl">1.25</span>,
           <span class="dt">shade.alpha =</span> <span class="fl">0.75</span>,
           <span class="dt">show.numbers=</span><span class="ot">FALSE</span>,
           <span class="dt">mb.ratio =</span> <span class="kw">c</span>(<span class="fl">0.6</span>, <span class="fl">0.4</span>),
           <span class="dt">queries=</span><span class="kw">list</span>(<span class="kw">list</span>(<span class="dt">query =</span> Myfunc, 
                             <span class="dt">params =</span> <span class="kw">list</span>(<span class="st">&quot;Match&quot;</span>), 
                             <span class="dt">color =</span> <span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>], 
                             <span class="dt">active =</span> <span class="ot">TRUE</span> )))

AML</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-36"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-36-1.png" alt="Miles et al Figure 2D" width="384" />
<p class="caption">
Figure 3.10: Miles et al Figure 2D
</p>
</div>
<p>And now for CH only</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CH&lt;-<span class="kw">upset</span>(test_set<span class="op">%&gt;%</span><span class="kw">filter</span>(Dx<span class="op">==</span><span class="st">&quot;CH&quot;</span>), <span class="dt">sets=</span>DTAI_genes,<span class="dt">order.by =</span> <span class="kw">c</span>(<span class="st">&quot;degree&quot;</span>),
           <span class="dt">main.bar.color =</span> <span class="st">&quot;grey60&quot;</span>,<span class="dt">decreasing=</span><span class="ot">FALSE</span>,
           <span class="dt">mainbar.y.label =</span> <span class="st">&quot;Number of samples&quot;</span>,
           <span class="dt">sets.x.label =</span> <span class="st">&quot;Number of </span><span class="ch">\n</span><span class="st"> mutant samples&quot;</span>,
           <span class="dt">text.scale=</span><span class="fl">1.25</span>,
           <span class="dt">shade.alpha =</span> <span class="fl">0.75</span>,
           <span class="dt">show.numbers=</span><span class="ot">FALSE</span>,
           <span class="dt">queries=</span><span class="kw">list</span>(<span class="kw">list</span>(<span class="dt">query =</span> Myfunc, <span class="dt">params =</span> <span class="kw">list</span>(<span class="st">&quot;Match&quot;</span>), <span class="dt">color =</span> <span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>], <span class="dt">active =</span> <span class="ot">TRUE</span> )))

CH</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-37"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-37-1.png" alt="Miles et al, Extended Figure 3F" width="384" />
<p class="caption">
Figure 3.11: Miles et al, Extended Figure 3F
</p>
</div>
<p>The above plots looks nice and are worth looking at for a host of genes. The signalling effectors are quite interesting. In this setting you can see that when a sample harbors multiple signaling effectors, they are rarely in the dominant clone, unlike the epigenetic mutations above. If you want to edit this to evaluate a gene of interest, you need to go back to the beginning here and add the columns back into the “sample_mutations” and “dominant_clone_mutations” variables.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Identify genes of interest</span>
genes&lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;FLT3&quot;</span>,<span class="st">&quot;NRAS&quot;</span>,<span class="st">&quot;JAK2&quot;</span>,<span class="st">&quot;KRAS&quot;</span>,<span class="st">&quot;PTPN11&quot;</span>)

<span class="co"># Subset data.frame above to only the dominant clones</span>
dominant_clone_mutations &lt;-<span class="st"> </span>clone_mutations<span class="op">%&gt;%</span><span class="kw">filter</span>(Clonality<span class="op">==</span><span class="st">&quot;Dominant&quot;</span>)

<span class="co"># For each sample, determine if the dominant clone</span>
sample_mutations<span class="op">$</span>Match&lt;-<span class="kw">ifelse</span>(<span class="kw">sapply</span>(sample_mutations<span class="op">$</span>Sample,<span class="cf">function</span>(sample) {
                                                <span class="kw">all</span>(sample_mutations<span class="op">%&gt;%</span>
<span class="st">                                                        </span><span class="kw">filter</span>(Sample<span class="op">==</span>sample)<span class="op">%&gt;%</span>
<span class="st">                                                        </span><span class="kw">select</span>(<span class="kw">all_of</span>(genes))<span class="op">==</span>
<span class="st">                                                    </span>dominant_clone_mutations<span class="op">%&gt;%</span>
<span class="st">                                                        </span><span class="kw">filter</span>(Sample<span class="op">==</span>sample)<span class="op">%&gt;%</span>
<span class="st">                                                        </span><span class="kw">select</span>(<span class="kw">all_of</span>(genes)))
                          }) ,<span class="st">&quot;Match&quot;</span>,<span class="st">&quot;Absent&quot;</span>)

<span class="co"># Join this match with the phenotype data to pick a disease state of interest</span>
test_set&lt;-<span class="kw">left_join</span>(sample_mutations,pheno,<span class="dt">by=</span><span class="st">&quot;Sample&quot;</span>)

<span class="co"># Necessary little function to </span>
Myfunc &lt;-<span class="st"> </span><span class="cf">function</span>(row,feature) {
  data &lt;-<span class="st"> </span>(row[feature]<span class="op">==</span><span class="st">&quot;Match&quot;</span>)
}

<span class="co"># Slight difference from BioRxiv paper due to a misclassification of one TET2 mutant CMML sample as AML.</span>
AML_signaling_genes&lt;-<span class="kw">upset</span>(test_set<span class="op">%&gt;%</span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;AML&quot;</span>,Dx)), 
           <span class="dt">sets=</span>genes,
           <span class="dt">order.by =</span> <span class="kw">c</span>(<span class="st">&quot;degree&quot;</span>),
           <span class="dt">main.bar.color =</span> <span class="st">&quot;grey60&quot;</span>,<span class="dt">decreasing=</span><span class="ot">FALSE</span>,
           <span class="dt">mainbar.y.label =</span> <span class="st">&quot;Number of samples&quot;</span>,
           <span class="dt">sets.x.label =</span> <span class="st">&quot;Number of </span><span class="ch">\n</span><span class="st"> mutant samples&quot;</span>,
           <span class="dt">text.scale=</span><span class="fl">1.25</span>,
           <span class="dt">shade.alpha =</span> <span class="fl">0.75</span>,
           <span class="dt">show.numbers=</span><span class="ot">FALSE</span>,
           <span class="dt">mb.ratio =</span> <span class="kw">c</span>(<span class="fl">0.6</span>, <span class="fl">0.4</span>),
           <span class="dt">queries=</span><span class="kw">list</span>(<span class="kw">list</span>(<span class="dt">query =</span> Myfunc, 
                             <span class="dt">params =</span> <span class="kw">list</span>(<span class="st">&quot;Match&quot;</span>), 
                             <span class="dt">color =</span> <span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>], 
                             <span class="dt">active =</span> <span class="ot">TRUE</span> )))

AML_signaling_genes</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-38"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-38-1.png" alt="Not shown in Miles et" width="384" />
<p class="caption">
Figure 3.12: Not shown in Miles et
</p>
</div>
<p>We wanted to visualize this data another way, and gain more resolution into multi-mutant subclones. In order to do this we generated graph structures using the igraph package. We decided to focus specifically on mutli mutant DTAI or multi mutant signaling AMLs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(igraph)

<span class="co"># identify sample with at least 2 DTAI mutationss</span>
multi_DTAI&lt;-test_set<span class="op">%&gt;%</span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;AML&quot;</span>,Dx))<span class="op">%&gt;%</span>
<span class="st">                </span><span class="kw">filter</span>((ASXL1<span class="op">+</span>DNMT3A<span class="op">+</span>TET2<span class="op">+</span>IDH1<span class="op">+</span>IDH2)<span class="op">&gt;=</span><span class="dv">2</span>)<span class="op">%&gt;%</span>
<span class="st">                </span><span class="kw">distinct</span>(Sample)<span class="op">%&gt;%</span><span class="kw">pull</span>(Sample)

<span class="co"># Identify dominant clones </span>
DTAI_dominant_clones&lt;-clone_mutations<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">%in%</span>multi_DTAI)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(Clonality<span class="op">==</span><span class="st">&quot;Dominant&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">select</span>(Clone,Clone_size,Sample,DNMT3A,TET2,ASXL1,IDH1,IDH2)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="kw">c</span>(DNMT3A,TET2,ASXL1,IDH1,IDH2),
                                           <span class="dt">names_to=</span><span class="st">&quot;Gene&quot;</span>,<span class="dt">values_to=</span><span class="st">&quot;Mutated&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(Mutated<span class="op">==</span><span class="dv">1</span>)

<span class="co"># Now we want to know which variants are in the dominant clone, and the size of that clone. </span>
<span class="co"># I&#39;m sure there is a nice way to do this in dplyr, grouping on sample, but I couldn&#39;t figure it out</span>
<span class="co"># so we will use lapply.</span>
genes_in_each_dominant_clone&lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind,<span class="kw">setNames</span>(<span class="kw">lapply</span>(multi_DTAI,<span class="cf">function</span>(x){
    
    <span class="co"># Extract the genes</span>
    dominant_variants&lt;-<span class="st"> </span>DTAI_dominant_clones<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">==</span>x)<span class="op">%&gt;%</span><span class="kw">pull</span>(Gene)
    
    <span class="co"># Extract the clone size</span>
    dominant_clone_size&lt;-<span class="st"> </span>DTAI_dominant_clones<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">==</span>x)<span class="op">%&gt;%</span><span class="kw">pull</span>(Clone_size)
  
    <span class="co"># if there are more than two DTAI variants in the dominant clone make a combinatorial edgelist</span>
    <span class="cf">if</span>(<span class="kw">length</span>(dominant_variants)<span class="op">&gt;=</span><span class="dv">2</span>){
      <span class="kw">return</span>(<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">combn</span>(dominant_variants,<span class="dv">2</span>)),dominant_clone_size,<span class="st">&quot;Dominant&quot;</span>),<span class="kw">c</span>(<span class="st">&quot;to&quot;</span>,<span class="st">&quot;from&quot;</span>,<span class="st">&quot;size&quot;</span>,<span class="st">&quot;Clonality&quot;</span>)))} 
    <span class="co"># if there is only 1 mutant in the dominant clone, list it for now so we can count the mutation, </span>
    <span class="co"># but we will eventually filter it out</span>
  <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(dominant_variants)<span class="op">==</span><span class="dv">1</span>){
      <span class="kw">return</span>(<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">c</span>(dominant_variants,dominant_variants)),dominant_clone_size,<span class="st">&quot;Subclone&quot;</span>),<span class="kw">c</span>(<span class="st">&quot;to&quot;</span>,<span class="st">&quot;from&quot;</span>,<span class="st">&quot;size&quot;</span>,<span class="st">&quot;Clonality&quot;</span>)))} 
    <span class="co"># if no DTAI mutants in the dominant clone, ignore.</span>
  <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(dominant_variants)<span class="op">==</span><span class="dv">0</span>){
      <span class="ot">NULL</span>
  }
}),multi_DTAI))<span class="op">%&gt;%</span><span class="kw">distinct</span>()

<span class="co"># Now we will go for a similar process with subclones.</span>
DTAI_sub_clones&lt;-clone_mutations<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">%in%</span>multi_DTAI)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(Clonality<span class="op">!=</span><span class="st">&quot;Dominant&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">select</span>(Clone,Clone_size,Sample,DNMT3A,TET2,ASXL1,IDH1,IDH2)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="kw">c</span>(DNMT3A,TET2,ASXL1,IDH1,IDH2),
                                           <span class="dt">names_to=</span><span class="st">&quot;Gene&quot;</span>,<span class="dt">values_to=</span><span class="st">&quot;Mutated&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(Mutated<span class="op">==</span><span class="dv">1</span>)<span class="op">%&gt;%</span>
<span class="co"># This is how we specifically select multi mutant subclone</span>
<span class="st">                              </span><span class="kw">group_by</span>(Clone,Sample)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">add_tally</span>()<span class="op">%&gt;%</span><span class="kw">filter</span>(n<span class="op">&gt;</span><span class="dv">1</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">ungroup</span>()

<span class="co"># Same process as above, but note that we decided to only plot the largest multi mutant clone.</span>
<span class="co"># Try getting rid of this and seeing how it looks.</span>
genes_in_each_subclone &lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind,<span class="kw">setNames</span>(<span class="kw">lapply</span>(multi_DTAI,<span class="cf">function</span>(x){
    subclone_variants &lt;-<span class="st"> </span>DTAI_sub_clones<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">==</span>x)<span class="op">%&gt;%</span>
<span class="st">                                  </span><span class="kw">filter</span>(Clone_size<span class="op">==</span><span class="kw">max</span>(Clone_size))<span class="op">%&gt;%</span>
<span class="st">                                  </span><span class="kw">pull</span>(Gene)
    subclone_size &lt;-<span class="st"> </span>DTAI_sub_clones<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">==</span>x)<span class="op">%&gt;%</span>
<span class="st">                                  </span><span class="kw">filter</span>(Clone_size<span class="op">==</span><span class="kw">max</span>(Clone_size))<span class="op">%&gt;%</span>
<span class="st">                                  </span><span class="kw">pull</span>(Clone_size)
  
    <span class="cf">if</span>(<span class="kw">length</span>(subclone_variants)<span class="op">&gt;=</span><span class="dv">2</span>){
      <span class="kw">return</span>(<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">combn</span>(<span class="kw">rev</span>(subclone_variants),<span class="dv">2</span>)),subclone_size,<span class="st">&quot;Subclone&quot;</span>),<span class="kw">c</span>(<span class="st">&quot;to&quot;</span>,<span class="st">&quot;from&quot;</span>,<span class="st">&quot;size&quot;</span>,<span class="st">&quot;Clonality&quot;</span>)))} 
  <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(subclone_variants)<span class="op">==</span><span class="dv">1</span>){
      <span class="kw">return</span>(<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">c</span>(subclone_variants,subclone_variants)),subclone_size,<span class="st">&quot;Subclone&quot;</span>),<span class="kw">c</span>(<span class="st">&quot;to&quot;</span>,<span class="st">&quot;from&quot;</span>,<span class="st">&quot;size&quot;</span>,<span class="st">&quot;Clonality&quot;</span>)))} 
  <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(subclone_variants)<span class="op">==</span><span class="dv">0</span>){
      <span class="ot">NULL</span>
  }
}),multi_DTAI))<span class="op">%&gt;%</span><span class="kw">distinct</span>()

<span class="co"># Now bind these two dataframe together</span>
final_set&lt;-<span class="st"> </span><span class="kw">rbind</span>(genes_in_each_dominant_clone,genes_in_each_subclone)

<span class="co"># And remove the edges that are self referencing. We preserve the input variable so we can represent</span>
<span class="co"># the vertex size in relation to total mutation burden in this subset of patients.</span>
final_set_filtered &lt;-final_set<span class="op">%&gt;%</span><span class="kw">filter</span>(to<span class="op">!=</span>from)

graph&lt;-<span class="kw">graph_from_data_frame</span>(final_set_filtered,<span class="dt">directed=</span>F)<span class="op">%&gt;%</span>
<span class="st">                    </span><span class="kw">set_edge_attr</span>(<span class="st">&quot;weight&quot;</span>, <span class="dt">value =</span> <span class="kw">as.numeric</span>(final_set_filtered<span class="op">%&gt;%</span><span class="kw">pull</span>(size))<span class="op">*</span><span class="dv">3</span>) <span class="op">%&gt;%</span>
<span class="st">                    </span><span class="kw">set_edge_attr</span>(<span class="st">&quot;color&quot;</span>, <span class="dt">value =</span> <span class="kw">ifelse</span>(final_set_filtered<span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">                                                          </span><span class="kw">pull</span>(Clonality)<span class="op">==</span><span class="st">&quot;Dominant&quot;</span>,
                                                          <span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>],<span class="st">&quot;grey20&quot;</span>))

mutant_counts&lt;-<span class="kw">table</span>(<span class="kw">c</span>(<span class="kw">as.character</span>(final_set<span class="op">$</span>to),<span class="kw">as.character</span>(final_set<span class="op">$</span>from)))[<span class="kw">names</span>(<span class="kw">V</span>(graph))]
scaled_mutant_counts &lt;-mutant_counts<span class="op">/</span><span class="kw">sum</span>(mutant_counts)<span class="op">*</span><span class="dv">50</span>

radian.rescale &lt;-<span class="st"> </span><span class="cf">function</span>(x, <span class="dt">start=</span><span class="dv">0</span>, <span class="dt">direction=</span><span class="dv">1</span>) {
  c.rotate &lt;-<span class="st"> </span><span class="cf">function</span>(x) (x <span class="op">+</span><span class="st"> </span>start) <span class="op">%%</span><span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>pi) <span class="op">*</span><span class="st"> </span>direction
  <span class="kw">c.rotate</span>(scales<span class="op">::</span><span class="kw">rescale</span>(x, <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span> <span class="op">*</span><span class="st"> </span>pi), <span class="kw">range</span>(x)))
}

lab.locs &lt;-<span class="st"> </span><span class="kw">radian.rescale</span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dt">direction=</span><span class="op">-</span><span class="dv">1</span>, <span class="dt">start=</span><span class="dv">5</span>)
lab.locs[<span class="dv">3</span>]&lt;-<span class="st"> </span><span class="op">-</span><span class="fl">2.5</span></code></pre></div>
<p>The permute function below, and the indexing of the vertex.size is only to preserve the order that is present in the manuscript. When I recoded everything, the order of the mutations changed in the data frame and inverted the graph.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">reordered_graph&lt;-igraph<span class="op">::</span><span class="kw">permute</span>(graph,<span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">5</span>))
<span class="kw">plot.igraph</span>(reordered_graph,
            <span class="dt">edge.width =</span> <span class="kw">E</span>(reordered_graph)<span class="op">$</span>weight,
            <span class="dt">vertex.color=</span><span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>],
            <span class="dt">vertex.frame.color=</span><span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>],
            <span class="dt">vertex.size=</span>scaled_mutant_counts[<span class="kw">names</span>(<span class="kw">V</span>(reordered_graph))], 
            <span class="dt">vertex.label.family=</span><span class="st">&quot;Helvetica&quot;</span>,
            <span class="dt">vertex.label.color=</span><span class="st">&quot;black&quot;</span>,
            <span class="dt">vertex.label.degree=</span>lab.locs,
            <span class="dt">vertex.label.dist=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">3</span>,<span class="dv">7</span>,<span class="dv">3</span>),
            <span class="dt">layout=</span>layout_in_circle)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-40"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-40-1.png" alt="Miles et al Figure 2E" width="576" />
<p class="caption">
Figure 3.13: Miles et al Figure 2E
</p>
</div>
<p>Now for the signalling genes. Everything is the exactly the same as above, so I’ve skipped commenting it unless where necessary. The graph looks a little different here than the BioRxiv manuscript due to a sample diagnosis being incorrect previously.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">multi_signaling&lt;-test_set<span class="op">%&gt;%</span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;AML&quot;</span>,Dx))<span class="op">%&gt;%</span>
<span class="st">                </span><span class="kw">filter</span>((FLT3<span class="op">+</span>JAK2<span class="op">+</span>NRAS<span class="op">+</span>KRAS<span class="op">+</span>PTPN11)<span class="op">&gt;=</span><span class="dv">2</span>)<span class="op">%&gt;%</span>
<span class="st">                </span><span class="kw">distinct</span>(Sample)<span class="op">%&gt;%</span><span class="kw">pull</span>(Sample)

signaling_dominant_clones&lt;-clone_mutations<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">%in%</span>multi_signaling)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(Clonality<span class="op">==</span><span class="st">&quot;Dominant&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">select</span>(Clone_size,Sample,FLT3,JAK2,NRAS,KRAS,PTPN11)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="kw">c</span>(FLT3,JAK2,NRAS,KRAS,PTPN11),
                                           <span class="dt">names_to=</span><span class="st">&quot;Gene&quot;</span>,<span class="dt">values_to=</span><span class="st">&quot;Mutated&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(Mutated<span class="op">==</span><span class="dv">1</span>)

genes_in_each_dominant_clone&lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind,<span class="kw">setNames</span>(<span class="kw">lapply</span>(multi_signaling,<span class="cf">function</span>(x){
    dominant_variants&lt;-<span class="st"> </span>signaling_dominant_clones<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">==</span>x)<span class="op">%&gt;%</span><span class="kw">pull</span>(Gene)
    dominant_clone_size&lt;-<span class="st"> </span>signaling_dominant_clones<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">==</span>x)<span class="op">%&gt;%</span><span class="kw">pull</span>(Clone_size)
  
    <span class="cf">if</span>(<span class="kw">length</span>(dominant_variants)<span class="op">&gt;=</span><span class="dv">2</span>){
      <span class="kw">return</span>(<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">combn</span>((dominant_variants),<span class="dv">2</span>)),dominant_clone_size,<span class="st">&quot;Dominant&quot;</span>),<span class="kw">c</span>(<span class="st">&quot;to&quot;</span>,<span class="st">&quot;from&quot;</span>,<span class="st">&quot;size&quot;</span>,<span class="st">&quot;Clonality&quot;</span>)))} 
  <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(dominant_variants)<span class="op">==</span><span class="dv">1</span>){
      <span class="kw">return</span>(<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">c</span>(dominant_variants,dominant_variants)),dominant_clone_size,<span class="st">&quot;Subclone&quot;</span>),<span class="kw">c</span>(<span class="st">&quot;to&quot;</span>,<span class="st">&quot;from&quot;</span>,<span class="st">&quot;size&quot;</span>,<span class="st">&quot;Clonality&quot;</span>)))} 
  <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(dominant_variants)<span class="op">==</span><span class="dv">0</span>){
      <span class="ot">NULL</span>
  }
}),multi_signaling))<span class="op">%&gt;%</span><span class="kw">distinct</span>()

signaling_sub_clones&lt;-clone_mutations<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">%in%</span>multi_signaling)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(Clonality<span class="op">!=</span><span class="st">&quot;Dominant&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">select</span>(Clone,Clone_size,Sample,FLT3,JAK2,NRAS,KRAS,PTPN11)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="kw">c</span>(FLT3,JAK2,NRAS,KRAS,PTPN11),
                                           <span class="dt">names_to=</span><span class="st">&quot;Gene&quot;</span>,<span class="dt">values_to=</span><span class="st">&quot;Mutated&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">filter</span>(Mutated<span class="op">==</span><span class="dv">1</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">group_by</span>(Clone,Sample)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">add_tally</span>()<span class="op">%&gt;%</span><span class="kw">filter</span>(n<span class="op">&gt;</span><span class="dv">1</span>)<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">ungroup</span>()

genes_in_each_subclone&lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind,<span class="kw">setNames</span>(<span class="kw">lapply</span>(multi_signaling,<span class="cf">function</span>(x){
    subclone_variants&lt;-<span class="st"> </span>signaling_sub_clones<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">==</span>x)<span class="op">%&gt;%</span>
<span class="st">                                  </span><span class="kw">filter</span>(Clone_size<span class="op">==</span><span class="kw">max</span>(Clone_size))<span class="op">%&gt;%</span><span class="kw">pull</span>(Gene)
    subclone_size&lt;-<span class="st"> </span>signaling_sub_clones<span class="op">%&gt;%</span><span class="kw">filter</span>(Sample<span class="op">==</span>x)<span class="op">%&gt;%</span>
<span class="st">                                  </span><span class="kw">filter</span>(Clone_size<span class="op">==</span><span class="kw">max</span>(Clone_size))<span class="op">%&gt;%</span><span class="kw">pull</span>(Clone_size)
  
    <span class="cf">if</span>(<span class="kw">length</span>(subclone_variants)<span class="op">&gt;=</span><span class="dv">2</span>){
      <span class="kw">return</span>(<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">combn</span>((subclone_variants),<span class="dv">2</span>)),subclone_size,<span class="st">&quot;Subclone&quot;</span>),<span class="kw">c</span>(<span class="st">&quot;to&quot;</span>,<span class="st">&quot;from&quot;</span>,<span class="st">&quot;size&quot;</span>,<span class="st">&quot;Clonality&quot;</span>)))} 
  <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(subclone_variants)<span class="op">==</span><span class="dv">1</span>){
      <span class="kw">return</span>(<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">t</span>(<span class="kw">c</span>(subclone_variants,subclone_variants)),subclone_size,<span class="st">&quot;Subclone&quot;</span>),<span class="kw">c</span>(<span class="st">&quot;to&quot;</span>,<span class="st">&quot;from&quot;</span>,<span class="st">&quot;size&quot;</span>,<span class="st">&quot;Clonality&quot;</span>)))} 
  <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(subclone_variants)<span class="op">==</span><span class="dv">0</span>){
      <span class="ot">NULL</span>
  }
}),multi_signaling))<span class="op">%&gt;%</span><span class="kw">distinct</span>()


final_set&lt;-<span class="st"> </span><span class="kw">rbind</span>(genes_in_each_dominant_clone,genes_in_each_subclone)

final_set_filtered &lt;-final_set<span class="op">%&gt;%</span><span class="kw">filter</span>(to<span class="op">!=</span>from)

graph&lt;-<span class="kw">graph_from_data_frame</span>(final_set_filtered,<span class="dt">directed=</span>F)<span class="op">%&gt;%</span>
<span class="st">                    </span><span class="kw">set_edge_attr</span>(<span class="st">&quot;weight&quot;</span>, <span class="dt">value =</span> <span class="kw">as.numeric</span>(final_set_filtered<span class="op">%&gt;%</span><span class="kw">pull</span>(size))<span class="op">*</span><span class="dv">3</span>) <span class="op">%&gt;%</span>
<span class="st">                    </span><span class="kw">set_edge_attr</span>(<span class="st">&quot;color&quot;</span>, <span class="dt">value =</span> <span class="kw">ifelse</span>(final_set_filtered<span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">                                                          </span><span class="kw">pull</span>(Clonality)<span class="op">==</span><span class="st">&quot;Dominant&quot;</span>,
                                                          <span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>],<span class="st">&quot;grey20&quot;</span>))

mutant_counts&lt;-<span class="kw">table</span>(<span class="kw">c</span>(<span class="kw">as.character</span>(final_set<span class="op">$</span>to),<span class="kw">as.character</span>(final_set<span class="op">$</span>from)))[<span class="kw">names</span>(<span class="kw">V</span>(graph))]
scaled_mutant_counts &lt;-mutant_counts<span class="op">/</span><span class="kw">sum</span>(mutant_counts)<span class="op">*</span><span class="dv">50</span>
radian.rescale &lt;-<span class="st"> </span><span class="cf">function</span>(x, <span class="dt">start=</span><span class="dv">0</span>, <span class="dt">direction=</span><span class="dv">1</span>) {
  c.rotate &lt;-<span class="st"> </span><span class="cf">function</span>(x) (x <span class="op">+</span><span class="st"> </span>start) <span class="op">%%</span><span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>pi) <span class="op">*</span><span class="st"> </span>direction
  <span class="kw">c.rotate</span>(scales<span class="op">::</span><span class="kw">rescale</span>(x, <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span> <span class="op">*</span><span class="st"> </span>pi), <span class="kw">range</span>(x)))
}

lab.locs &lt;-<span class="st"> </span><span class="kw">radian.rescale</span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dt">direction=</span><span class="op">-</span><span class="dv">1</span>, <span class="dt">start=</span><span class="dv">5</span>)
lab.locs[<span class="dv">3</span>]&lt;-<span class="st"> </span><span class="op">-</span><span class="fl">2.5</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">reordered_graph&lt;-igraph<span class="op">::</span><span class="kw">permute</span>(graph,<span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">3</span>))
<span class="kw">plot.igraph</span>(reordered_graph,
            <span class="dt">edge.width =</span> <span class="kw">E</span>(reordered_graph)<span class="op">$</span>weight,
            <span class="dt">vertex.color=</span><span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>],
            <span class="dt">vertex.frame.color=</span><span class="kw">brewer.pal</span>(<span class="dv">5</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">5</span>],
            <span class="dt">vertex.size=</span>scaled_mutant_counts[<span class="kw">names</span>(<span class="kw">V</span>(reordered_graph))], 
            <span class="dt">vertex.label.family=</span><span class="st">&quot;Helvetica&quot;</span>,
            <span class="dt">vertex.label.color=</span><span class="st">&quot;black&quot;</span>,
            <span class="dt">vertex.label.degree=</span>lab.locs,
            <span class="dt">vertex.label.dist=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">3</span>,<span class="dv">7</span>,<span class="dv">3</span>),
            <span class="dt">layout=</span>layout_in_circle)</code></pre></div>
<p><img src="scDNA_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
<p>For the last part, we wanted to dial in on DNMT3A and IDH mutant leukemias to see if there were distinct co mutants between DNMT3A, IDH1, IDH2 and combinatorial mutant samples. This plot is interesting, and we looked at it many different ways including on the sample level as depicted in the preprint, and on the completely clonal level below. I’ll show the sample level as well to demonstrate the difference in data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Add a column for whether a clone contains 2 signaling mutations (as demonstrated above)</span>
clone_mutations_added&lt;-clone_mutations<span class="op">%&gt;%</span><span class="kw">inner_join</span>(pheno,<span class="dt">by=</span><span class="st">&quot;Sample&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">          </span><span class="kw">rowwise</span>()<span class="op">%&gt;%</span>
<span class="st">          </span><span class="kw">add_column</span>(<span class="dt">signal2=</span><span class="kw">ifelse</span>(<span class="kw">c</span>(.<span class="op">$</span>FLT3<span class="op">+</span>.<span class="op">$</span>JAK2<span class="op">+</span>.<span class="op">$</span>NRAS<span class="op">+</span>.<span class="op">$</span>KRAS<span class="op">+</span>.<span class="op">$</span>PTPN11)<span class="op">&gt;=</span><span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">0</span>))<span class="op">%&gt;%</span>
<span class="st">          </span><span class="kw">ungroup</span>()

gene_sets&lt;-<span class="kw">list</span>(<span class="st">&quot;DNMT3A&quot;</span>=<span class="st">&quot;DNMT3A&quot;</span>,
                <span class="st">&quot;DNMT3A IDH1&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;IDH1&quot;</span>),
                <span class="st">&quot;DNMT3A IDH2&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;IDH2&quot;</span>),
                <span class="st">&quot;IDH1&quot;</span>=<span class="st">&quot;IDH1&quot;</span>,
                <span class="st">&quot;IDH2&quot;</span>=<span class="st">&quot;IDH2&quot;</span>)

comutant_status&lt;-<span class="kw">do.call</span>(rbind,<span class="kw">setNames</span>(<span class="kw">lapply</span>(gene_sets, <span class="cf">function</span>(genes){
  epi_to_exclude &lt;-<span class="st"> </span><span class="kw">setdiff</span>(<span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;TET2&quot;</span>,<span class="st">&quot;ASXL1&quot;</span>,<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;IDH2&quot;</span>),genes)
  clone_subset&lt;-<span class="st"> </span>clone_mutations_added<span class="op">%&gt;%</span><span class="kw">filter</span>(Dx<span class="op">==</span><span class="st">&quot;AML&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">filter_at</span>(<span class="kw">vars</span>(<span class="kw">all_of</span>(epi_to_exclude)),<span class="kw">all_vars</span>(.<span class="op">==</span><span class="dv">0</span>))<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">filter_at</span>(<span class="kw">vars</span>(<span class="kw">all_of</span>(genes)), <span class="kw">all_vars</span>(.<span class="op">==</span><span class="dv">1</span>))<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">select</span>(<span class="kw">c</span>(Sample,<span class="kw">all_of</span>(genes),FLT3,JAK2,NRAS,KRAS,PTPN11,signal2))
  
  <span class="kw">data.frame</span>(<span class="st">&quot;Group&quot;</span>  =<span class="kw">paste</span>(genes,<span class="dt">sep=</span><span class="st">&quot; &quot;</span>,<span class="dt">collapse =</span> <span class="st">&quot; &quot;</span>),
             <span class="st">&quot;Total&quot;</span>  =clone_subset<span class="op">%&gt;%</span><span class="kw">summarise</span>(<span class="dt">Count=</span><span class="kw">n</span>())<span class="op">%&gt;%</span><span class="kw">pull</span>(Count),
             <span class="st">&quot;FLT3&quot;</span>   =clone_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(FLT3)  <span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;PTPN11&quot;</span> =clone_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(PTPN11)<span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;JAK2&quot;</span>   =clone_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(JAK2)  <span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;KRAS&quot;</span>   =clone_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(KRAS)  <span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;NRAS&quot;</span>   =clone_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(NRAS)  <span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;Multiple mutants&quot;</span>=clone_subset<span class="op">%&gt;%</span><span class="kw">tally</span>(signal2)<span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;None&quot;</span>   =clone_subset<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">add_column</span>(<span class="dt">None=</span><span class="kw">ifelse</span>(<span class="kw">c</span>(.<span class="op">$</span>FLT3<span class="op">+</span>.<span class="op">$</span>JAK2<span class="op">+</span>
<span class="st">                                                            </span>.<span class="op">$</span>NRAS<span class="op">+</span>.<span class="op">$</span>KRAS<span class="op">+</span>
<span class="st">                                                            </span>.<span class="op">$</span>PTPN11)<span class="op">==</span><span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>))<span class="op">%&gt;%</span>
<span class="st">                                                      </span><span class="kw">tally</span>(None)<span class="op">%&gt;%</span><span class="kw">pull</span>(n))

}),<span class="kw">names</span>(gene_sets)))

final&lt;-comutant_status<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="op">!</span>Total)<span class="op">%&gt;%</span>
<span class="st">                </span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="kw">c</span>(FLT3,PTPN11,JAK2,KRAS,NRAS,Multiple.mutants,None),
                             <span class="dt">names_to=</span><span class="st">&quot;Mutation&quot;</span>,
                             <span class="dt">values_to=</span><span class="st">&quot;Count&quot;</span>)
      
final<span class="op">$</span>Mutation &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;Multiple.mutants&quot;</span>,<span class="st">&quot;Multiple mutants&quot;</span>,final<span class="op">$</span>Mutation)
final<span class="op">$</span>Mutation &lt;-<span class="st"> </span><span class="kw">factor</span>(final<span class="op">$</span>Mutation, 
                         <span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;JAK2&quot;</span>,<span class="st">&quot;PTPN11&quot;</span>,<span class="st">&quot;NRAS&quot;</span>,<span class="st">&quot;KRAS&quot;</span>,<span class="st">&quot;FLT3&quot;</span>,<span class="st">&quot;Multiple mutants&quot;</span>,<span class="st">&quot;None&quot;</span>)))
final<span class="op">$</span>Group &lt;-<span class="st"> </span><span class="kw">factor</span>(final<span class="op">$</span>Group, 
                      <span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;DNMT3A IDH1&quot;</span>,<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;DNMT3A IDH2&quot;</span>,<span class="st">&quot;IDH2&quot;</span>)))

color_set&lt;-<span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">9</span>,<span class="st">&quot;Set1&quot;</span>)[<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">8</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">9</span>)])

gg_co_mutants&lt;-<span class="kw">ggplot</span>(final, <span class="kw">aes</span>(<span class="dt">x=</span>Group,<span class="dt">fill=</span>Mutation,<span class="dt">y=</span>Count))<span class="op">+</span>
<span class="st">                </span><span class="kw">geom_col</span>(<span class="dt">position=</span><span class="st">&quot;fill&quot;</span>)<span class="op">+</span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">12</span>)<span class="op">+</span>
<span class="st">                </span><span class="kw">ylab</span>(<span class="st">&quot;Fraction of clones&quot;</span>)<span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span><span class="kw">coord_flip</span>()<span class="op">+</span>
<span class="st">                </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="op">+</span>
<span class="st">                </span><span class="kw">ggtitle</span>(<span class="st">&quot;Clone level data&quot;</span>)<span class="op">+</span>
<span class="st">                </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust=</span><span class="fl">0.5</span>))<span class="op">+</span>
<span class="st">                </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>color_set,<span class="st">&quot;Co mutation&quot;</span>,
                                  <span class="dt">guide =</span> <span class="kw">guide_legend</span>(<span class="dt">reverse =</span> <span class="ot">TRUE</span>))</code></pre></div>
<p>Now the same thing below but starting with the sample based mutation matrix we used for the UpSet plots.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Add a column for whether a clone contains 2 signaling mutations (as demonstrated above)</span>
sample_mutations_added&lt;-sample_mutations<span class="op">%&gt;%</span><span class="kw">inner_join</span>(pheno,<span class="dt">by=</span><span class="st">&quot;Sample&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">          </span><span class="kw">rowwise</span>()<span class="op">%&gt;%</span>
<span class="st">          </span><span class="kw">add_column</span>(<span class="dt">signal2=</span><span class="kw">ifelse</span>(<span class="kw">c</span>(.<span class="op">$</span>FLT3<span class="op">+</span>.<span class="op">$</span>JAK2<span class="op">+</span>.<span class="op">$</span>NRAS<span class="op">+</span>.<span class="op">$</span>KRAS<span class="op">+</span>.<span class="op">$</span>PTPN11)<span class="op">&gt;=</span><span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">0</span>))<span class="op">%&gt;%</span>
<span class="st">          </span><span class="kw">ungroup</span>()

gene_sets&lt;-<span class="kw">list</span>(<span class="st">&quot;DNMT3A&quot;</span>=<span class="st">&quot;DNMT3A&quot;</span>,
                <span class="st">&quot;DNMT3A IDH1&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;IDH1&quot;</span>),
                <span class="st">&quot;DNMT3A IDH2&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;IDH2&quot;</span>),
                <span class="st">&quot;IDH1&quot;</span>=<span class="st">&quot;IDH1&quot;</span>,
                <span class="st">&quot;IDH2&quot;</span>=<span class="st">&quot;IDH2&quot;</span>)

comutant_sample_status&lt;-<span class="kw">do.call</span>(rbind,<span class="kw">setNames</span>(<span class="kw">lapply</span>(gene_sets, <span class="cf">function</span>(genes){
  epi_to_exclude &lt;-<span class="st"> </span><span class="kw">setdiff</span>(<span class="kw">c</span>(<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;TET2&quot;</span>,<span class="st">&quot;ASXL1&quot;</span>,<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;IDH2&quot;</span>),genes)
  sample_subset&lt;-<span class="st"> </span>sample_mutations_added<span class="op">%&gt;%</span><span class="kw">filter</span>(Dx<span class="op">==</span><span class="st">&quot;AML&quot;</span>)<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">filter_at</span>(<span class="kw">vars</span>(<span class="kw">all_of</span>(epi_to_exclude)),<span class="kw">all_vars</span>(.<span class="op">==</span><span class="dv">0</span>))<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">filter_at</span>(<span class="kw">vars</span>(<span class="kw">all_of</span>(genes)), <span class="kw">all_vars</span>(.<span class="op">==</span><span class="dv">1</span>))<span class="op">%&gt;%</span>
<span class="st">                          </span><span class="kw">select</span>(<span class="kw">c</span>(Sample,<span class="kw">all_of</span>(genes),FLT3,JAK2,NRAS,KRAS,PTPN11,signal2))
  
  <span class="kw">data.frame</span>(<span class="st">&quot;Group&quot;</span>  =<span class="kw">paste</span>(genes,<span class="dt">sep=</span><span class="st">&quot; &quot;</span>,<span class="dt">collapse =</span> <span class="st">&quot; &quot;</span>),
             <span class="st">&quot;Total&quot;</span>  =sample_subset<span class="op">%&gt;%</span><span class="kw">summarise</span>(<span class="dt">Count=</span><span class="kw">n</span>())<span class="op">%&gt;%</span><span class="kw">pull</span>(Count),
             <span class="st">&quot;FLT3&quot;</span>   =sample_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(FLT3)  <span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;PTPN11&quot;</span> =sample_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(PTPN11)<span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;JAK2&quot;</span>   =sample_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(JAK2)  <span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;KRAS&quot;</span>   =sample_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(KRAS)  <span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;NRAS&quot;</span>   =sample_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(signal2<span class="op">==</span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">tally</span>(NRAS)  <span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;Multiple mutants&quot;</span>=sample_subset<span class="op">%&gt;%</span><span class="kw">tally</span>(signal2)<span class="op">%&gt;%</span><span class="kw">pull</span>(n),
             <span class="st">&quot;None&quot;</span>   =sample_subset<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">add_column</span>(<span class="dt">None=</span><span class="kw">ifelse</span>(<span class="kw">c</span>(.<span class="op">$</span>FLT3<span class="op">+</span>.<span class="op">$</span>JAK2<span class="op">+</span>
<span class="st">                                                            </span>.<span class="op">$</span>NRAS<span class="op">+</span>.<span class="op">$</span>KRAS<span class="op">+</span>
<span class="st">                                                            </span>.<span class="op">$</span>PTPN11)<span class="op">==</span><span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>))<span class="op">%&gt;%</span>
<span class="st">                                                      </span><span class="kw">tally</span>(None)<span class="op">%&gt;%</span><span class="kw">pull</span>(n))

}),<span class="kw">names</span>(gene_sets)))

final&lt;-comutant_sample_status<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="op">!</span>Total)<span class="op">%&gt;%</span>
<span class="st">                </span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="kw">c</span>(FLT3,PTPN11,JAK2,KRAS,NRAS,Multiple.mutants,None),
                             <span class="dt">names_to=</span><span class="st">&quot;Mutation&quot;</span>,
                             <span class="dt">values_to=</span><span class="st">&quot;Count&quot;</span>)
      
final<span class="op">$</span>Mutation &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;Multiple.mutants&quot;</span>,<span class="st">&quot;Multiple mutants&quot;</span>,final<span class="op">$</span>Mutation)
final<span class="op">$</span>Mutation &lt;-<span class="st"> </span><span class="kw">factor</span>(final<span class="op">$</span>Mutation, 
                         <span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;JAK2&quot;</span>,<span class="st">&quot;PTPN11&quot;</span>,<span class="st">&quot;NRAS&quot;</span>,<span class="st">&quot;KRAS&quot;</span>,<span class="st">&quot;FLT3&quot;</span>,<span class="st">&quot;Multiple mutants&quot;</span>,<span class="st">&quot;None&quot;</span>)))
final<span class="op">$</span>Group &lt;-<span class="st"> </span><span class="kw">factor</span>(final<span class="op">$</span>Group, 
                      <span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;IDH1&quot;</span>,<span class="st">&quot;DNMT3A IDH1&quot;</span>,<span class="st">&quot;DNMT3A&quot;</span>,<span class="st">&quot;DNMT3A IDH2&quot;</span>,<span class="st">&quot;IDH2&quot;</span>)))

color_set&lt;-<span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">9</span>,<span class="st">&quot;Set1&quot;</span>)[<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">8</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">9</span>)])

gg_co_mutants_sample&lt;-<span class="kw">ggplot</span>(final, <span class="kw">aes</span>(<span class="dt">x=</span>Group,<span class="dt">fill=</span>Mutation,<span class="dt">y=</span>Count))<span class="op">+</span>
<span class="st">                </span><span class="kw">geom_col</span>(<span class="dt">position=</span><span class="st">&quot;fill&quot;</span>)<span class="op">+</span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">12</span>)<span class="op">+</span>
<span class="st">                </span><span class="kw">ylab</span>(<span class="st">&quot;Fraction of Samples&quot;</span>)<span class="op">+</span><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span><span class="kw">coord_flip</span>()<span class="op">+</span>
<span class="st">                </span><span class="kw">scale_y_continuous</span>(<span class="dt">expand=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="op">+</span>
<span class="st">                </span><span class="kw">ggtitle</span>(<span class="st">&quot;Sample level data&quot;</span>)<span class="op">+</span>
<span class="st">                </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">hjust=</span><span class="fl">0.5</span>))<span class="op">+</span>
<span class="st">                </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>color_set,<span class="st">&quot;Co mutation&quot;</span>,
                                  <span class="dt">guide =</span> <span class="kw">guide_legend</span>(<span class="dt">reverse =</span> <span class="ot">TRUE</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot_grid</span>(gg_co_mutants,gg_co_mutants_sample,<span class="dt">ncol=</span><span class="dv">1</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-45"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-45-1.png" alt="Miles et al Figure 2G" width="480" />
<p class="caption">
Figure 3.14: Miles et al Figure 2G
</p>
</div>
<p>This is one of the very last pieces we wanted to address. As you can see, we observe an ability of epigenetic modifications to synergize and persist in the dominant clone, while signaling mutations appear to be mututally exclusive. We wanted to determine clones that harbor 2+ epigenetic modifiers were larger than clones that harbored 2+ siganling mutations. To investigate the clone size directly, we focused on a subset of samples that were mutated for both DNMT3A and IDH1/2 as well as harboring multiple mutations in RAS/FLT3/PTPN11.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Run back the patients we looked at in the network plots</span>
patients_of_interest &lt;-<span class="st"> </span>sample_mutations_added<span class="op">%&gt;%</span><span class="kw">filter</span>(DNMT3A<span class="op">==</span><span class="dv">1</span><span class="op">&amp;</span>signal2<span class="op">==</span><span class="dv">1</span><span class="op">&amp;</span>
<span class="st">                                                          </span>(IDH1<span class="op">==</span><span class="dv">1</span><span class="op">|</span>IDH2<span class="op">==</span><span class="dv">1</span>))<span class="op">%&gt;%</span>
<span class="st">                                                  </span><span class="kw">pull</span>(Sample)

dual_mutation_fractions&lt;-<span class="kw">do.call</span>(rbind,<span class="kw">setNames</span>(<span class="kw">lapply</span>(patients_of_interest,<span class="cf">function</span>(sample){
  <span class="co"># Extract the genotype matrix for each patient</span>
    NGT&lt;-final_sample_summary[[sample]]<span class="op">$</span>NGT<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="op">!</span>Clone)
  <span class="co"># Simplfy the matrix to mutant vs non mutant cells.</span>
    NGT[NGT<span class="op">&gt;</span><span class="dv">0</span>] &lt;-<span class="dv">1</span>
  <span class="co"># Extract the mutations of interest</span>
    <span class="kw">data.frame</span>(<span class="st">&quot;Sample&quot;</span>=sample,
               <span class="st">&quot;Epigenetic&quot;</span>=NGT<span class="op">%&gt;%</span><span class="kw">select_if</span>(<span class="kw">grepl</span>(<span class="st">&quot;DNMT3A|IDH&quot;</span>,<span class="kw">names</span>(.)))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">mutate</span>(<span class="dt">dual_epi=</span><span class="kw">rowSums</span>(.))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">summarise</span>(<span class="dt">Epigenetic=</span><span class="kw">sum</span>(dual_epi<span class="op">&gt;=</span><span class="dv">2</span>)<span class="op">/</span><span class="kw">n</span>()),
               <span class="st">&quot;Signaling&quot;</span>=NGT<span class="op">%&gt;%</span><span class="kw">select_if</span>(<span class="kw">grepl</span>(<span class="st">&quot;FLT3|RAS|JAK2|PTPN11&quot;</span>,<span class="kw">names</span>(.)))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">mutate</span>(<span class="dt">dual_signaling=</span><span class="kw">rowSums</span>(.))<span class="op">%&gt;%</span>
<span class="st">                              </span><span class="kw">summarise</span>(<span class="dt">Signaling=</span><span class="kw">sum</span>(dual_signaling<span class="op">&gt;=</span><span class="dv">2</span>)<span class="op">/</span><span class="kw">n</span>()))
}),patients_of_interest))

<span class="co"># Plot the data</span>
gg_fraction_comutated_cells&lt;-<span class="kw">ggplot</span>(dual_mutation_fractions<span class="op">%&gt;%</span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="op">!</span>Sample,
                                                                      <span class="dt">names_to=</span><span class="st">&quot;Group&quot;</span>,
                                                                      <span class="dt">values_to =</span> <span class="st">&quot;Fraction&quot;</span>),
                                    <span class="kw">aes</span>(<span class="dt">x=</span>Group,<span class="dt">y=</span>Fraction,<span class="dt">fill=</span>Group))<span class="op">+</span>
<span class="st">                                        </span><span class="kw">geom_boxplot</span>()<span class="op">+</span>
<span class="st">                                        </span><span class="kw">geom_jitter</span>(<span class="dt">width=</span><span class="fl">0.1</span>)<span class="op">+</span>
<span class="st">                                        </span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">12</span>)<span class="op">+</span>
<span class="st">                                        </span><span class="kw">scale_fill_brewer</span>(<span class="dt">type=</span><span class="st">&quot;qual&quot;</span>,<span class="dt">palette =</span> <span class="st">&quot;Set1&quot;</span>,<span class="st">&quot;Mutation pairs&quot;</span>)<span class="op">+</span>
<span class="st">                                        </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;Fraction of co-mutated cells&quot;</span>)

gg_fraction_comutated_cells</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-46"></span>
<img src="scDNA_files/figure-html/unnamed-chunk-46-1.png" alt="Miles et al Extended Figure 3G" width="384" />
<p class="caption">
Figure 3.15: Miles et al Extended Figure 3G
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">### need to clean this up with tidy at some point
dual_mutation_fractions_melted&lt;-dual_mutation_fractions<span class="op">%&gt;%</span><span class="kw">pivot_longer</span>(<span class="dt">cols=</span><span class="op">!</span>Sample,
                          <span class="dt">names_to=</span><span class="st">&quot;Group&quot;</span>,
                          <span class="dt">values_to =</span> <span class="st">&quot;Fraction&quot;</span>)
<span class="kw">t.test</span>(dual_mutation_fractions_melted<span class="op">$</span>Fraction<span class="op">~</span><span class="kw">factor</span>(dual_mutation_fractions_melted<span class="op">$</span>Group))</code></pre></div>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  dual_mutation_fractions_melted$Fraction by factor(dual_mutation_fractions_melted$Group)
## t = 4.665, df = 6.132, p-value = 0.003259
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.3011350 0.9583285
## sample estimates:
## mean in group Epigenetic  mean in group Signaling 
##                0.9016239                0.2718921</code></pre>

</div>
</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="figure-1-cohort-characterization.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="final-words.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scDNA.pdf", "scDNA.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
