<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.2 Functions | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</title>
  <meta name="description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  <meta name="generator" content="bookdown 0.18.3 and GitBook 2.6.7" />

  <meta property="og:title" content="6.2 Functions | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.2 Functions | Single cell mutational profiling delineates clonal trajectories in myeloid malignancies" />
  
  <meta name="twitter:description" content="This is a code base for all of the analyses performed in Miles, Bowman et al." />
  

<meta name="author" content="Robert Bowman" />


<meta name="date" content="2020-05-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="AppendixA.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Foreward</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="data-extraction-and-setup.html"><a href="data-extraction-and-setup.html"><i class="fa fa-check"></i><b>3</b> Data Extraction and Setup</a><ul>
<li class="chapter" data-level="3.1" data-path="tapestri-library.html"><a href="tapestri-library.html"><i class="fa fa-check"></i><b>3.1</b> Tapestri Package</a></li>
<li class="chapter" data-level="3.2" data-path="post-processing.html"><a href="post-processing.html"><i class="fa fa-check"></i><b>3.2</b> Post processing</a></li>
<li class="chapter" data-level="3.3" data-path="assessing-clonal-abundance.html"><a href="assessing-clonal-abundance.html"><i class="fa fa-check"></i><b>3.3</b> Assessing clonal abundance</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="cohort-description.html"><a href="cohort-description.html"><i class="fa fa-check"></i><b>4</b> Cohort Description</a></li>
<li class="chapter" data-level="5" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>5</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="Appendix.html"><a href="Appendix.html"><i class="fa fa-check"></i><b>6</b> Appendix</a><ul>
<li class="chapter" data-level="6.1" data-path="AppendixA.html"><a href="AppendixA.html"><i class="fa fa-check"></i><b>6.1</b> Import from Tapestri Insights</a></li>
<li class="chapter" data-level="6.2" data-path="functions.html"><a href="functions.html"><i class="fa fa-check"></i><b>6.2</b> Functions</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Single cell mutational profiling delineates clonal trajectories in myeloid malignancies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="functions" class="section level2">
<h2><span class="header-section-number">6.2</span> Functions</h2>
<p>I’m likely going to wrap all of this into a package soon, but here we will start with a set of packages and functions used throughout the study. I need to go through and annotate with package was used with with function, and that will take a little time. So for the sake of making the code available as soon as possible, I’ll present the intermediate product here.</p>
<p>Packages The following are a list of packages that are used during the study. I will update this list with the minial set and begin to put together a package soon.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ape)
<span class="kw">library</span>(circlize)
<span class="kw">library</span>(cooccur)
<span class="kw">library</span>(corrplot)
<span class="kw">library</span>(cowplot)
<span class="kw">library</span>(data.table)
<span class="kw">library</span>(dplyr)
<span class="kw">library</span>(forcats)
<span class="kw">library</span>(gdata)
<span class="kw">library</span>(ggbeeswarm)
<span class="kw">library</span>(ggcorrplot)
<span class="kw">library</span>(ggnet)
<span class="kw">library</span>(ggplotify)
<span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(ggrepel)
<span class="kw">library</span>(ggridges)
<span class="kw">library</span>(grid)
<span class="kw">library</span>(gridExtra)
<span class="kw">library</span>(ggthemes)
<span class="kw">library</span>(ggtree)
<span class="kw">library</span>(igraph)
<span class="kw">library</span>(knitr)
<span class="kw">library</span>(miscTools)
<span class="kw">library</span>(network)
<span class="kw">library</span>(pals)
<span class="kw">library</span>(parallel)
<span class="kw">library</span>(pheatmap)
<span class="kw">library</span>(phylobase)
<span class="kw">library</span>(purrr)
<span class="kw">library</span>(qgraph)
<span class="kw">library</span>(RColorBrewer)
<span class="kw">library</span>(rcompanion)
<span class="kw">library</span>(ReinforcementLearning)
<span class="kw">library</span>(reshape2)
<span class="kw">library</span>(sna)
<span class="kw">library</span>(tidygraph)
<span class="kw">library</span>(tidyr)
<span class="kw">library</span>(tidyverse)
<span class="kw">library</span>(TRONCO)
<span class="kw">library</span>(vegan)
<span class="kw">library</span>(widyr)</code></pre></div>
<p>Custom functions</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)

random_distribution_NGT_CI &lt;-<span class="st"> </span><span class="cf">function</span>(NGT_to_clone_list,clonal_abundance,sample_to_test,replicates,clone_cutoff){
                        mat_of_interest &lt;-<span class="st"> </span>NGT_to_clone[[sample_to_test]][,<span class="dv">1</span><span class="op">:</span>(<span class="kw">ncol</span>(NGT_to_clone[[sample_to_test]])<span class="op">-</span><span class="dv">1</span>)]
                        bulk_VAF_order &lt;-<span class="kw">names</span>(<span class="kw">sort</span>(<span class="kw">colSums</span>(mat_of_interest),<span class="dt">decreasing=</span><span class="ot">TRUE</span>))
                        test&lt;-<span class="kw">replicate</span>(<span class="dt">n=</span>replicates,<span class="kw">apply</span>(mat_of_interest,<span class="dt">MARGIN=</span><span class="kw">c</span>(<span class="dv">2</span>),<span class="cf">function</span>(x){
                          z&lt;-<span class="kw">sample</span>(x,<span class="dt">size=</span><span class="kw">length</span>(x),<span class="dt">replace=</span><span class="ot">FALSE</span>)
                          <span class="kw">return</span>(<span class="kw">as.numeric</span>(z))}),<span class="dt">simplify =</span> <span class="st">&quot;array&quot;</span>)
                        test2&lt;-<span class="kw">apply</span>(test,<span class="dv">3</span>,<span class="cf">function</span>(q){
                          x&lt;-<span class="kw">data.frame</span>(q[,bulk_VAF_order],
                                        <span class="st">&quot;Clone&quot;</span>=<span class="kw">apply</span>(q[,bulk_VAF_order],<span class="dv">1</span>, <span class="cf">function</span>(p){<span class="kw">paste</span>(p,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)}))
                          y&lt;-<span class="kw">data.table</span>(<span class="kw">data.frame</span>(<span class="st">&quot;Count&quot;</span>=<span class="kw">as.matrix</span>(<span class="kw">table</span>(x[,<span class="st">&quot;Clone&quot;</span>])),
                                                   <span class="st">&quot;Clone&quot;</span>=<span class="kw">names</span>(<span class="kw">table</span>(x[,<span class="st">&quot;Clone&quot;</span>]))),
                                        <span class="dt">key=</span><span class="st">&quot;Count&quot;</span>)
                          y<span class="op">$</span>Clone&lt;-<span class="kw">factor</span>(y<span class="op">$</span>Clone,<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">as.character</span>(y<span class="op">$</span>Clone)))
                          <span class="kw">return</span>(<span class="kw">data.frame</span>(y))})
                        <span class="cf">if</span>(<span class="kw">class</span>(test2)<span class="op">==</span><span class="st">&quot;list&quot;</span>){y &lt;-<span class="st"> </span><span class="kw">lapply</span>(test2,data.frame) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">reduce</span>(full_join, <span class="dt">by =</span> <span class="st">&quot;Clone&quot;</span>)}
                        <span class="cf">if</span>(<span class="kw">class</span>(test2)<span class="op">==</span><span class="st">&quot;array&quot;</span>){y &lt;-<span class="st"> </span><span class="kw">apply</span>(test2,<span class="dv">3</span>,data.frame) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">reduce</span>(full_join, <span class="dt">by =</span> <span class="st">&quot;Clone&quot;</span>)}
                        y[<span class="kw">is.na</span>(y)]&lt;-<span class="dv">0</span>
                        <span class="kw">rownames</span>(y)&lt;-y[,<span class="st">&quot;Clone&quot;</span>]
                        y&lt;-<span class="kw">as.matrix</span>(y[,<span class="op">-</span><span class="dv">2</span>])
                        <span class="kw">mode</span>(y)&lt;-<span class="st">&#39;numeric&#39;</span>
                        z&lt;-<span class="kw">t</span>(<span class="kw">apply</span>(y,<span class="dv">1</span>,<span class="cf">function</span>(p){<span class="kw">quantile</span>(p,<span class="dt">probs=</span><span class="kw">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>))}))
                        z_mat&lt;-<span class="kw">data.frame</span>(z,<span class="st">&quot;Clone&quot;</span>=<span class="kw">rownames</span>(z))
                        set&lt;-<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">full_join</span>(<span class="kw">data.frame</span>(clonal_abundance[[sample_to_test]]),z_mat,<span class="dt">by=</span><span class="st">&quot;Clone&quot;</span>)),<span class="kw">c</span>(<span class="st">&quot;Count&quot;</span>,<span class="st">&quot;Clone&quot;</span>,<span class="st">&quot;LCI&quot;</span>,<span class="st">&quot;UCI&quot;</span>))<span class="op">%&gt;%</span><span class="kw">filter</span>(LCI<span class="op">&gt;=</span>clone_cutoff<span class="op">&amp;!</span><span class="kw">is.na</span>(Count))
                        <span class="kw">return</span>(set)
}

bootstrap_allele_dropout_NGT_mat &lt;-<span class="st"> </span><span class="cf">function</span>(NGT_to_clone_list,clonal_abundance,sample_to_test,replicates,clone_cutoff){
                  mat_of_interest &lt;-<span class="st"> </span>NGT_to_clone[[sample_to_test]][,<span class="dv">1</span><span class="op">:</span>(<span class="kw">ncol</span>(NGT_to_clone[[sample_to_test]])<span class="op">-</span><span class="dv">1</span>)]
                  bulk_VAF_order &lt;-<span class="kw">names</span>(<span class="kw">sort</span>(<span class="kw">colSums</span>(mat_of_interest),<span class="dt">decreasing=</span><span class="ot">TRUE</span>))
                  test&lt;-<span class="kw">replicate</span>(<span class="dt">n=</span>replicates,<span class="kw">apply</span>(mat_of_interest,<span class="dt">MARGIN=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),<span class="cf">function</span>(x){
                    <span class="cf">if</span>(x<span class="op">==</span><span class="dv">1</span>){z&lt;-x} 
                    <span class="cf">if</span>(x<span class="op">==</span><span class="dv">0</span>){z&lt;-<span class="kw">sample</span>(<span class="dt">x=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="dt">size=</span><span class="dv">1</span>,<span class="dt">prob=</span><span class="kw">c</span>(<span class="fl">0.9</span>,<span class="fl">0.1</span>))}
                    <span class="cf">if</span>(x<span class="op">==</span><span class="dv">2</span>){z&lt;-<span class="kw">sample</span>(<span class="dt">x=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">1</span>),<span class="dt">size=</span><span class="dv">1</span>,<span class="dt">prob=</span><span class="kw">c</span>(<span class="fl">0.9</span>,<span class="fl">0.1</span>))}
                    <span class="kw">return</span>(<span class="kw">as.numeric</span>(z))}),<span class="dt">simplify =</span> <span class="st">&quot;array&quot;</span>)
                  test2&lt;-<span class="kw">apply</span>(test,<span class="dv">3</span>,<span class="cf">function</span>(q){
                    x&lt;-<span class="kw">data.frame</span>(q[,bulk_VAF_order],
                                  <span class="st">&quot;Clone&quot;</span>=<span class="kw">apply</span>(q[,bulk_VAF_order],<span class="dv">1</span>, <span class="cf">function</span>(p){<span class="kw">paste</span>(p,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)}))
                    y&lt;-<span class="kw">data.table</span>(<span class="kw">data.frame</span>(<span class="st">&quot;Count&quot;</span>=<span class="kw">as.matrix</span>(<span class="kw">table</span>(x[,<span class="st">&quot;Clone&quot;</span>])),
                                             <span class="st">&quot;Clone&quot;</span>=<span class="kw">names</span>(<span class="kw">table</span>(x[,<span class="st">&quot;Clone&quot;</span>]))),
                                  <span class="dt">key=</span><span class="st">&quot;Count&quot;</span>)
                    y<span class="op">$</span>Clone&lt;-<span class="kw">factor</span>(y<span class="op">$</span>Clone,<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">as.character</span>(y<span class="op">$</span>Clone)))
                    <span class="kw">return</span>(<span class="kw">data.frame</span>(y))})
                  
                  <span class="cf">if</span>(<span class="kw">class</span>(test2)<span class="op">==</span><span class="st">&quot;list&quot;</span>){y &lt;-<span class="st"> </span><span class="kw">lapply</span>(test2,data.frame) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">reduce</span>(full_join, <span class="dt">by =</span> <span class="st">&quot;Clone&quot;</span>)}
                  <span class="cf">if</span>(<span class="kw">class</span>(test2)<span class="op">==</span><span class="st">&quot;array&quot;</span>){y &lt;-<span class="st"> </span><span class="kw">apply</span>(test2,<span class="dv">3</span>,data.frame) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">reduce</span>(full_join, <span class="dt">by =</span> <span class="st">&quot;Clone&quot;</span>)}
                  y[<span class="kw">is.na</span>(y)]&lt;-<span class="dv">0</span>
                  <span class="kw">rownames</span>(y)&lt;-y[,<span class="st">&quot;Clone&quot;</span>]
                  y&lt;-<span class="kw">as.matrix</span>(y[,<span class="op">-</span><span class="dv">2</span>])
                  <span class="kw">mode</span>(y)&lt;-<span class="st">&#39;numeric&#39;</span>
                  z&lt;-<span class="kw">t</span>(<span class="kw">apply</span>(y,<span class="dv">1</span>,<span class="cf">function</span>(p){<span class="kw">quantile</span>(p,<span class="dt">probs=</span><span class="kw">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>))}))
                  z_mat&lt;-<span class="kw">data.frame</span>(z,<span class="st">&quot;Clone&quot;</span>=<span class="kw">rownames</span>(z))
                  set&lt;-<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">full_join</span>(<span class="kw">data.frame</span>(clonal_abundance[[sample_to_test]]),z_mat,<span class="dt">by=</span><span class="st">&quot;Clone&quot;</span>)),<span class="kw">c</span>(<span class="st">&quot;Count&quot;</span>,<span class="st">&quot;Clone&quot;</span>,<span class="st">&quot;LCI&quot;</span>,<span class="st">&quot;UCI&quot;</span>))<span class="op">%&gt;%</span><span class="kw">filter</span>(LCI<span class="op">&gt;=</span>clone_cutoff<span class="op">&amp;!</span><span class="kw">is.na</span>(Count))
                  <span class="kw">return</span>(set)
} 

bootstrap_NGT_mat &lt;-<span class="st"> </span><span class="cf">function</span>(NGT_to_clone_list,clonal_abundance,sample_to_test,replicates,clone_cutoff){
  test&lt;-<span class="kw">replicate</span>(<span class="dt">n=</span>replicates,<span class="kw">resample_fun</span>(NGT_to_clone[[sample_to_test]]),<span class="dt">simplify =</span> <span class="st">&quot;array&quot;</span>)
  <span class="cf">if</span>(<span class="kw">class</span>(test)<span class="op">==</span><span class="st">&quot;list&quot;</span>){
    y &lt;-<span class="st"> </span><span class="kw">lapply</span>(test,data.frame) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">reduce</span>(full_join, <span class="dt">by =</span> <span class="st">&quot;Clone&quot;</span>)
  }
  <span class="cf">if</span>(<span class="kw">class</span>(test)<span class="op">==</span><span class="st">&quot;array&quot;</span>){
    y &lt;-<span class="st"> </span><span class="kw">apply</span>(test,<span class="dv">3</span>,data.frame) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">reduce</span>(full_join, <span class="dt">by =</span> <span class="st">&quot;Clone&quot;</span>)
  }
  y[<span class="kw">is.na</span>(y)]&lt;-<span class="dv">0</span>
  <span class="kw">rownames</span>(y)&lt;-y[,<span class="st">&quot;Clone&quot;</span>]
  y&lt;-<span class="kw">as.matrix</span>(y[,<span class="op">-</span><span class="dv">2</span>])
  <span class="kw">mode</span>(y)&lt;-<span class="st">&#39;numeric&#39;</span>
  z&lt;-<span class="kw">t</span>(<span class="kw">apply</span>(y,<span class="dv">1</span>,<span class="cf">function</span>(p){
    <span class="kw">quantile</span>(p,<span class="dt">probs=</span><span class="kw">c</span>(<span class="fl">0.025</span>,<span class="fl">0.975</span>))
  }))
  z_mat&lt;-<span class="kw">data.frame</span>(z,<span class="st">&quot;Clone&quot;</span>=<span class="kw">rownames</span>(z))
  set&lt;-<span class="kw">setNames</span>(<span class="kw">data.frame</span>(<span class="kw">inner_join</span>(<span class="kw">data.frame</span>(clonal_abundance[[sample_to_test]]),z_mat,<span class="dt">by=</span><span class="st">&quot;Clone&quot;</span>)),<span class="kw">c</span>(<span class="st">&quot;Count&quot;</span>,<span class="st">&quot;Clone&quot;</span>,<span class="st">&quot;LCI&quot;</span>,<span class="st">&quot;UCI&quot;</span>))<span class="op">%&gt;%</span><span class="kw">filter</span>(LCI<span class="op">&gt;=</span>clone_cutoff)
  <span class="kw">return</span>(set)
}

resample_fun&lt;-<span class="cf">function</span>(data){
  x &lt;-<span class="st"> </span>data[<span class="kw">sample</span>(<span class="dt">x=</span><span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(data),<span class="dt">replace=</span><span class="ot">TRUE</span>),]
  <span class="kw">return</span>(<span class="kw">as.matrix</span>(<span class="kw">data.table</span>(<span class="kw">data.frame</span>(<span class="st">&quot;Count&quot;</span>=<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">table</span>(x[,<span class="st">&quot;Clone&quot;</span>])),
                                         <span class="st">&quot;Clone&quot;</span>=<span class="kw">names</span>(<span class="kw">table</span>(x[,<span class="st">&quot;Clone&quot;</span>]))),
                              <span class="dt">key=</span><span class="st">&quot;Count&quot;</span>)))
}

extract_NGT_files &lt;-<span class="st"> </span><span class="cf">function</span>(sample,variants){
  NGT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;NGT.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
  <span class="kw">return</span>(NGT)
}

count_unknown_cells&lt;-<span class="st"> </span><span class="cf">function</span>(sample,variants){
    NGT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;NGT.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
    cells_with_unknown&lt;-NGT[<span class="kw">apply</span>(NGT[,variants],<span class="dt">MARGIN=</span><span class="dv">1</span>,<span class="dt">FUN=</span><span class="cf">function</span>(x){<span class="kw">sum</span>(x<span class="op">==</span><span class="dv">3</span>)<span class="op">&gt;</span><span class="dv">0</span>}),<span class="st">&quot;Cell&quot;</span>]
    <span class="kw">return</span>(<span class="kw">c</span>(<span class="kw">length</span>(cells_with_unknown)<span class="op">/</span><span class="kw">nrow</span>(NGT)))
  }

extract_DP_files &lt;-<span class="st"> </span><span class="cf">function</span>(sample,variants){
  DP &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;DP.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
  cells_with_unknown&lt;-DP[<span class="kw">apply</span>(DP[,variants],<span class="dt">MARGIN=</span><span class="dv">1</span>,<span class="dt">FUN=</span><span class="cf">function</span>(x){<span class="kw">sum</span>(x<span class="op">==</span><span class="dv">3</span>)<span class="op">&gt;</span><span class="dv">0</span>}),<span class="st">&quot;Cell&quot;</span>]
  matrix_of_interest&lt;-DP[<span class="op">!</span>DP<span class="op">$</span>Cell<span class="op">%in%</span>cells_with_unknown,variants]
  <span class="kw">return</span>(matrix_of_interest)
}

extract_SNP_info &lt;-<span class="st"> </span><span class="cf">function</span>(sample){
  AF &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;AF.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cdna&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;gene&quot;</span>],x[<span class="st">&quot;cdna&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(x[<span class="st">&quot;amplicon&quot;</span>]<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;MSK_RL_AMP54&quot;</span>, <span class="st">&quot;MSK_RL_AMP55&quot;</span>, <span class="st">&quot;MSK_RL_AMP56&quot;</span>,<span class="st">&quot;MSK_RL_AMP57&quot;</span>,<span class="st">&quot;MSK_RL_AMP58&quot;</span>),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;POS&quot;</span>],x[<span class="st">&quot;ALT&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })
  } <span class="cf">else</span> {
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;Variants.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    <span class="kw">colnames</span>(SNP_INFO)[<span class="dv">3</span>] &lt;-<span class="st"> &quot;protein&quot;</span>
    SNP_INFO<span class="op">$</span>cDNA &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>cDNA)
    SNP_INFO<span class="op">$</span>protein &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>protein)
    SNP_INFO<span class="op">$</span>Variant &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Variant)
    SNP_INFO<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Gene)
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cDNA&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;Gene&quot;</span>],x[<span class="st">&quot;cDNA&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^chr13&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>]),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
     }) 
  }
  <span class="kw">return</span>(SNP_INFO)
}

<span class="co"># run the following just once to load the function</span>
plot_variants_and_selection &lt;-<span class="st"> </span><span class="cf">function</span>(sample){
  <span class="co">#Load in the data</span>
  AF &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;AF.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - variant allele frequency</span>
  DP &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;DP.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                     <span class="co"># - read depth</span>
  GQ &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;GQ.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - quality scores</span>
  NGT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;NGT.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) <span class="co"># - genotype call converted to categorical (0-reference, 1-heterozygous mutation, 2-homozygous mutation, 3-unknown)</span>
  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cdna&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;gene&quot;</span>],x[<span class="st">&quot;cdna&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(x[<span class="st">&quot;amplicon&quot;</span>]<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;MSK_RL_AMP54&quot;</span>, <span class="st">&quot;MSK_RL_AMP55&quot;</span>, <span class="st">&quot;MSK_RL_AMP56&quot;</span>,<span class="st">&quot;MSK_RL_AMP57&quot;</span>,<span class="st">&quot;MSK_RL_AMP58&quot;</span>),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;POS&quot;</span>],x[<span class="st">&quot;ALT&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  } <span class="cf">else</span> {
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;Variants.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    <span class="kw">colnames</span>(SNP_INFO)[<span class="dv">3</span>] &lt;-<span class="st"> &quot;protein&quot;</span>
    SNP_INFO<span class="op">$</span>cDNA &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>cDNA)
    SNP_INFO<span class="op">$</span>protein &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>protein)
    SNP_INFO<span class="op">$</span>Variant &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Variant)
    SNP_INFO<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Gene)
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cDNA&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;Gene&quot;</span>],x[<span class="st">&quot;cDNA&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^chr13&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>]),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  }
  
  protein_changes_of_interest &lt;-<span class="st"> </span>(SNP_INFO <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(protein<span class="op">!=</span><span class="st">&quot;&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(protein))[,<span class="dv">1</span>] <span class="co"># select column of variants</span>
  SNP_changes_of_interest &lt;-<span class="st"> </span><span class="kw">rownames</span>(SNP_INFO)[SNP_INFO[,<span class="st">&quot;protein&quot;</span>]<span class="op">%in%</span>protein_changes_of_interest]
  SNP_protein_key &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Mutant&quot;</span>=SNP_changes_of_interest,<span class="st">&quot;Protein&quot;</span>=protein_changes_of_interest)
  
  <span class="co"># Cell and clone selection and Average depth per cell per allele</span>
  distribution_of_unknowns_by_variant &lt;-<span class="st"> </span><span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">2</span>,table)
  NGT_subset&lt;-NGT[,<span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>,<span class="st">&quot;Cell&quot;</span>,SNP_changes_of_interest)]
  <span class="co">#NGT_interest$Clone &lt;- apply(NGT_interest[,-c(1,2)],1,function(x){paste(x,sep=&quot;_&quot;,collapse=&quot;_&quot;)})</span>
  <span class="co">#clonal_abundance &lt;- sort(table(NGT_interest$Clone))</span>
  
  DP_subset&lt;-<span class="st"> </span>DP[,<span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>,<span class="st">&quot;Cell&quot;</span>,SNP_changes_of_interest)]
  DP_NGT_melt&lt;-<span class="kw">inner_join</span>(<span class="kw">melt</span>(DP_subset[,<span class="op">-</span><span class="dv">1</span>],<span class="dt">id.var=</span><span class="st">&quot;Cell&quot;</span>),<span class="kw">melt</span>(NGT_subset[,<span class="op">-</span><span class="dv">1</span>],<span class="dt">id.var=</span><span class="st">&quot;Cell&quot;</span>),<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;Cell&quot;</span>,<span class="st">&quot;variable&quot;</span>))
  <span class="kw">colnames</span>(DP_NGT_melt) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Cell&quot;</span>,<span class="st">&quot;Mutant&quot;</span>,<span class="st">&quot;Depth&quot;</span>,<span class="st">&quot;Genotype&quot;</span>)
  DP_NGT_melt<span class="op">$</span>Genotype &lt;-<span class="st"> </span><span class="kw">ifelse</span>(DP_NGT_melt<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">3</span>,<span class="st">&quot;Unknown&quot;</span>,
                                 <span class="kw">ifelse</span>(DP_NGT_melt<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">0</span>,<span class="st">&quot;WT&quot;</span>,
                                        <span class="kw">ifelse</span>(DP_NGT_melt<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">1</span>,<span class="st">&quot;Heterozygous&quot;</span>,
                                               <span class="kw">ifelse</span>(DP_NGT_melt<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">2</span>,<span class="st">&quot;Homozygous&quot;</span>,DP_NGT_melt<span class="op">$</span>Genotype))))
  DP_NGT_melt<span class="op">$</span>Genotype  &lt;-<span class="st"> </span><span class="kw">factor</span>(DP_NGT_melt<span class="op">$</span>Genotype ,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>,<span class="st">&quot;Unknown&quot;</span>))
  
  summarized&lt;-<span class="kw">data.frame</span>(DP_NGT_melt<span class="op">%&gt;%</span><span class="kw">group_by</span>(Mutant,Genotype)<span class="op">%&gt;%</span><span class="kw">summarise</span>(<span class="st">&quot;Depth&quot;</span>=<span class="kw">mean</span>(Depth)))
  summarized_no_unknown &lt;-<span class="st"> </span>summarized <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Genotype<span class="op">!=</span><span class="st">&quot;Unknown&quot;</span>)
  grouped &lt;-<span class="st"> </span><span class="kw">group_by</span>(summarized_no_unknown, Mutant)
  output &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">summarise</span>(grouped, <span class="dt">mean=</span><span class="kw">mean</span>(Depth), <span class="dt">sd=</span><span class="kw">sd</span>(Depth))) 
  output<span class="op">$</span>Include &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;Include&quot;</span>,<span class="kw">length</span>(<span class="kw">rownames</span>(output)))
  output_final &lt;-<span class="st"> </span><span class="kw">inner_join</span>(SNP_protein_key,output)
  
  depth_plot&lt;-<span class="st"> </span><span class="kw">ggplot</span>(summarized_no_unknown,<span class="kw">aes</span>(<span class="dt">x=</span>Mutant,<span class="dt">y=</span>Depth,<span class="dt">color=</span>Genotype))<span class="op">+</span><span class="kw">geom_point</span>()<span class="op">+</span><span class="kw">coord_flip</span>()<span class="op">+</span>
<span class="st">    </span><span class="kw">scale_x_discrete</span>(<span class="dt">limits =</span> <span class="kw">rev</span>(<span class="kw">levels</span>(summarized_no_unknown<span class="op">$</span>Mutant)))<span class="op">+</span><span class="kw">theme_bw</span>()<span class="co">#+</span>
  
  <span class="kw">write.csv</span>(output_final,<span class="kw">paste0</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,sample,<span class="st">&quot;_VARIANT_SELECTION.csv&quot;</span>),<span class="dt">row.names=</span><span class="ot">FALSE</span>)
  <span class="kw">ggsave</span>(<span class="kw">paste0</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,sample,<span class="st">&quot;_VARIANT_SELECTION.pdf&quot;</span>),depth_plot,<span class="dt">width=</span><span class="dv">10</span>,<span class="dt">height=</span><span class="dv">20</span>)
}

run_mbio_analysis&lt;-<span class="cf">function</span>(sample,diversity){
  <span class="co">#Load in the data</span>
  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;VARIANT_SELECTION_LAM&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    LAM_VARIANTS_MAT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,sample,<span class="st">&quot;_VARIANT_SELECTION_LAM.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    LAM_VARIANTS_SNP &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Mutant))[,<span class="dv">1</span>])
    LAM_VARIANTS_Protein &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Protein))[,<span class="dv">1</span>])
  } <span class="cf">else</span> {
    <span class="kw">print</span>(<span class="st">&quot;No variant selection performed!&quot;</span>) 
    <span class="kw">return</span>(<span class="st">&quot;next&quot;</span>)
  }
  AF &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;AF.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - variant allele frequency</span>
  DP &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;DP.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                     <span class="co"># - read depth</span>
  GQ &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;GQ.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - quality scores</span>
  NGT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;NGT.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) <span class="co"># - genotype call converted to categorical (0-reference, 1-heterozygous mutation, 2-homozygous mutation, 3-unknown)</span>
  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cdna&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;gene&quot;</span>],x[<span class="st">&quot;cdna&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(x[<span class="st">&quot;amplicon&quot;</span>]<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;MSK_RL_AMP54&quot;</span>, <span class="st">&quot;MSK_RL_AMP55&quot;</span>, <span class="st">&quot;MSK_RL_AMP56&quot;</span>,<span class="st">&quot;MSK_RL_AMP57&quot;</span>,<span class="st">&quot;MSK_RL_AMP58&quot;</span>),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;POS&quot;</span>],x[<span class="st">&quot;ALT&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  } <span class="cf">else</span> {
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;Variants.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    <span class="kw">colnames</span>(SNP_INFO)[<span class="dv">3</span>] &lt;-<span class="st"> &quot;protein&quot;</span>
    SNP_INFO<span class="op">$</span>cDNA &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>cDNA)
    SNP_INFO<span class="op">$</span>protein &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>protein)
    SNP_INFO<span class="op">$</span>Variant &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Variant)
    SNP_INFO<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Gene)
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cDNA&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;Gene&quot;</span>],x[<span class="st">&quot;cDNA&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^chr13&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>]),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  }
  
  SNP_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_SNP
  protein_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_Protein
  SNP_protein_key &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Mutant&quot;</span>=SNP_changes_of_interest,<span class="st">&quot;Protein&quot;</span>=protein_changes_of_interest)
  
  <span class="co"># Cell and clone selection</span>
  distribution_of_unknowns_by_variant &lt;-<span class="st"> </span><span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">2</span>,table)
  <span class="co">#print(distribution_of_unknowns_by_variant)</span>
  cells_with_unknown&lt;-NGT[<span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">1</span>,<span class="dt">FUN=</span><span class="cf">function</span>(x){<span class="kw">sum</span>(x<span class="op">==</span><span class="dv">3</span>)<span class="op">&gt;</span><span class="dv">0</span>}),<span class="st">&quot;Cell&quot;</span>]
  matrix_of_interest&lt;-NGT[<span class="op">!</span>NGT<span class="op">$</span>Cell<span class="op">%in%</span>cells_with_unknown,SNP_changes_of_interest]
  bulk_VAF_order&lt;-SNP_INFO[<span class="kw">colnames</span>(matrix_of_interest)[<span class="kw">order</span>(<span class="kw">colSums</span>(matrix_of_interest),<span class="dt">decreasing=</span><span class="ot">TRUE</span>)],<span class="st">&quot;protein&quot;</span>]
  bulk_VAF_order &lt;-<span class="st"> </span>bulk_VAF_order[<span class="op">!</span><span class="kw">duplicated</span>(bulk_VAF_order)]
  matrix_of_interest<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">apply</span>(matrix_of_interest,<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)})
  clonal_abundance &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">table</span>(matrix_of_interest<span class="op">$</span>Clone))
  clones_to_include&lt;-<span class="kw">names</span>(clonal_abundance)[clonal_abundance<span class="op">&gt;</span><span class="kw">length</span>(<span class="kw">rownames</span>(matrix_of_interest))<span class="op">*</span><span class="dv">0</span>]
  matrix_subset_clones &lt;-<span class="st"> </span>matrix_of_interest[matrix_of_interest<span class="op">$</span>Clone<span class="op">%in%</span>clones_to_include,]
  
  <span class="co">#Average depth per cell per allele</span>
  DP_subset&lt;-<span class="st"> </span>DP[<span class="op">-</span>cells_with_unknown,<span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>,<span class="st">&quot;Cell&quot;</span>,<span class="kw">rownames</span>(SNP_INFO)[SNP_INFO<span class="op">$</span>protein<span class="op">%in%</span>protein_changes_of_interest])]
  NGT_subset &lt;-<span class="st"> </span>NGT[NGT<span class="op">$</span>Cell<span class="op">%in%</span>DP_subset<span class="op">$</span>Cell,<span class="kw">colnames</span>(DP_subset)]
  DP_NGT_melt&lt;-<span class="kw">inner_join</span>(<span class="kw">melt</span>(DP_subset[,<span class="op">-</span><span class="dv">1</span>],<span class="dt">id.var=</span><span class="st">&quot;Cell&quot;</span>),<span class="kw">melt</span>(NGT[,<span class="op">-</span><span class="dv">1</span>],<span class="dt">id.var=</span><span class="st">&quot;Cell&quot;</span>),<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;Cell&quot;</span>,<span class="st">&quot;variable&quot;</span>))
  <span class="kw">colnames</span>(DP_NGT_melt) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Cell&quot;</span>,<span class="st">&quot;Mutant&quot;</span>,<span class="st">&quot;Depth&quot;</span>,<span class="st">&quot;Genotype&quot;</span>)
  DP_NGT_melt<span class="op">$</span>Genotype &lt;-<span class="st"> </span><span class="kw">ifelse</span>(DP_NGT_melt<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">3</span>,<span class="st">&quot;Unknown&quot;</span>,
                                 <span class="kw">ifelse</span>(DP_NGT_melt<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">0</span>,<span class="st">&quot;WT&quot;</span>,
                                        <span class="kw">ifelse</span>(DP_NGT_melt<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">1</span>,<span class="st">&quot;Heterozygous&quot;</span>,
                                               <span class="kw">ifelse</span>(DP_NGT_melt<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">2</span>,<span class="st">&quot;Homozygous&quot;</span>,DP_NGT_melt<span class="op">$</span>Genotype))))
  DP_NGT_melt<span class="op">$</span>Genotype  &lt;-<span class="st"> </span><span class="kw">factor</span>(DP_NGT_melt<span class="op">$</span>Genotype ,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>,<span class="st">&quot;Unknown&quot;</span>))
  
  summarized&lt;-<span class="kw">data.frame</span>(DP_NGT_melt<span class="op">%&gt;%</span><span class="kw">group_by</span>(Mutant,Genotype)<span class="op">%&gt;%</span><span class="kw">summarise</span>(<span class="st">&quot;Depth&quot;</span>=<span class="kw">mean</span>(Depth)))
  
  
  <span class="co"># Verify variants selected are unique and encode protein changes</span>
  <span class="op">!</span><span class="kw">any</span>(<span class="kw">duplicated</span>(SNP_INFO[<span class="kw">colnames</span>(matrix_subset_clones),<span class="st">&quot;protein&quot;</span>])) <span class="co">#Should return TRUE</span>
  <span class="kw">colnames</span>(matrix_subset_clones)&lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">as.character</span>(SNP_INFO[<span class="kw">colnames</span>(matrix_subset_clones)[<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(SNP_changes_of_interest)],<span class="st">&quot;protein&quot;</span>]),<span class="st">&quot;Clone&quot;</span>)
  
  <span class="co"># Aggregate data to get clone counts and architecture.</span>
  dedup&lt;-matrix_subset_clones[<span class="op">!</span><span class="kw">duplicated</span>(matrix_subset_clones),]
  clonal_architecture &lt;-<span class="st"> </span><span class="kw">melt</span>(dedup)
  <span class="kw">colnames</span>(clonal_architecture) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Clone&quot;</span>,<span class="st">&quot;Mutant&quot;</span>,<span class="st">&quot;Genotype&quot;</span>)
  clonal_architecture<span class="op">$</span>Genotype &lt;-<span class="st"> </span><span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">3</span>,<span class="ot">NA</span>,
                                         <span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">0</span>,<span class="st">&quot;WT&quot;</span>,
                                                <span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">1</span>,<span class="st">&quot;Heterozygous&quot;</span>,
                                                       <span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">2</span>,<span class="st">&quot;Homozygous&quot;</span>,clonal_architecture<span class="op">$</span>Genotype))))
  clonal_architecture<span class="op">$</span>Genotype  &lt;-<span class="st"> </span><span class="kw">factor</span>(clonal_architecture<span class="op">$</span>Genotype ,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>))
  clonal_abundance &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Count&quot;</span>=<span class="kw">as.matrix</span>(<span class="kw">table</span>(matrix_subset_clones<span class="op">$</span>Clone)))
  clonal_abundance<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">rownames</span>(clonal_abundance)
  clonal_abundance &lt;-<span class="st"> </span><span class="kw">data.table</span>(clonal_abundance,<span class="dt">key=</span><span class="st">&quot;Count&quot;</span>)
  clonal_abundance<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">factor</span>(clonal_abundance<span class="op">$</span>Clone,<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(clonal_abundance<span class="op">$</span>Clone)))
  clonal_architecture<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">as.character</span>(clonal_architecture<span class="op">$</span>Clone),<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">as.character</span>(clonal_abundance<span class="op">$</span>Clone)))
  clonal_architecture<span class="op">$</span>Mutant &lt;-<span class="kw">factor</span>(clonal_architecture<span class="op">$</span>Mutant , <span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">as.character</span>(bulk_VAF_order)))
  
  <span class="co"># Generate clonal architecture heatmap</span>
  gg_heatmap &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> clonal_architecture, <span class="kw">aes</span>(<span class="dt">x =</span> Clone, <span class="dt">y =</span> <span class="kw">factor</span>(Mutant), <span class="dt">fill =</span> Genotype)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">1</span>],
                               <span class="st">&quot;Heterozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">3</span>],
                               <span class="st">&quot;Homozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">6</span>],
                               <span class="st">&quot;Unknown&quot;</span>=<span class="st">&quot;grey50&quot;</span>),<span class="st">&quot;Genotype&quot;</span>)  <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Mutation&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>,
          <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), 
          <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
          <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),
          <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(),
          <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>),<span class="st">&quot;cm&quot;</span>))
  
  <span class="co"># Generate clonal abundance barplot</span>
  gg_clonal_barplot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> <span class="kw">data.frame</span>(clonal_abundance), <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(Clone), <span class="dt">y =</span> Count)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill =</span> Count)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_gray</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>()<span class="op">+</span>
<span class="st">    </span><span class="kw">ylim</span>(<span class="dv">0</span>,<span class="kw">max</span>(clonal_abundance<span class="op">$</span>Count)<span class="op">*</span><span class="fl">1.3</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Cell Count&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>Count), <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="dt">width=</span><span class="fl">0.9</span>), <span class="dt">vjust=</span><span class="op">-</span><span class="fl">0.25</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_distiller</span>(<span class="dt">name =</span> <span class="st">&quot;Value&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="dt">direction =</span> <span class="dv">1</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(),  <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
          <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,
          <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="op">-</span><span class="fl">0.2</span>,<span class="dv">1</span>),<span class="st">&quot;cm&quot;</span>))
  
  <span class="co"># Generate genotype calls by allele</span>
  distribution_of_unknowns_by_variant &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">2</span>,<span class="cf">function</span>(x){<span class="kw">table</span>(<span class="kw">factor</span>(x,<span class="dt">levels=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)))}))
  <span class="kw">rownames</span>(distribution_of_unknowns_by_variant) &lt;-<span class="st"> </span>SNP_INFO[<span class="kw">rownames</span>(distribution_of_unknowns_by_variant),<span class="st">&quot;protein&quot;</span>]
  <span class="kw">colnames</span>(distribution_of_unknowns_by_variant) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>,<span class="st">&quot;Unknown&quot;</span>)
  allele_melted &lt;-<span class="kw">melt</span>(distribution_of_unknowns_by_variant) 
  <span class="kw">colnames</span>(allele_melted) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Mutant&quot;</span>,<span class="st">&quot;Genotype&quot;</span>,<span class="st">&quot;Cells&quot;</span>)
  allele_melted<span class="op">$</span>Genotype &lt;-<span class="st"> </span><span class="kw">factor</span>(allele_melted<span class="op">$</span>Genotype,<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>,<span class="st">&quot;Unknown&quot;</span>)))
  allele_melted<span class="op">$</span>Mutant &lt;-<span class="kw">factor</span>(allele_melted<span class="op">$</span>Mutant , <span class="dt">levels=</span><span class="kw">as.character</span>(bulk_VAF_order))
  
  <span class="co">#Generate allele barplot</span>
  gg_alleles &lt;-<span class="st">  </span><span class="kw">ggplot</span>(<span class="dt">data =</span> <span class="kw">data.frame</span>(allele_melted), <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(Mutant), <span class="dt">y =</span> Cells)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill =</span> Genotype),<span class="dt">position=</span><span class="st">&quot;stack&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_gray</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_minimal</span>()<span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Cell Count&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">1</span>],
                               <span class="st">&quot;Heterozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">3</span>],
                               <span class="st">&quot;Homozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">6</span>],
                               <span class="st">&quot;Unknown&quot;</span>=<span class="st">&quot;grey50&quot;</span>),<span class="st">&quot;Genotype&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">30</span>,<span class="dt">hjust=</span><span class="dv">1</span>,<span class="dt">vjust=</span><span class="dv">1</span>))
  
  
  <span class="co">#Generate depth by alelle by mutant                       </span>
  gg_depth &lt;-<span class="st"> </span><span class="kw">ggplot</span>(summarized,<span class="kw">aes</span>(<span class="dt">x=</span>Mutant,<span class="dt">color=</span>Genotype,<span class="dt">y=</span>Depth))<span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="dt">width=</span><span class="fl">0.4</span>))<span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">1</span>],
                                <span class="st">&quot;Heterozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">3</span>],
                                <span class="st">&quot;Homozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">6</span>],
                                <span class="st">&quot;Unknown&quot;</span>=<span class="st">&quot;grey50&quot;</span>),<span class="st">&quot;Genotype&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">30</span>,<span class="dt">hjust=</span><span class="dv">1</span>,<span class="dt">vjust=</span><span class="dv">1</span>),
          <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>),<span class="st">&quot;cm&quot;</span>))
  
  <span class="co"># Compute spacing of graphs</span>
  plots &lt;-<span class="st"> </span><span class="kw">list</span>(gg_depth,gg_clonal_barplot,gg_alleles,gg_heatmap)
  grobs &lt;-<span class="st"> </span><span class="kw">list</span>()
  widths &lt;-<span class="st"> </span><span class="kw">list</span>()
  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(plots)){
    grobs[[i]] &lt;-<span class="st"> </span><span class="kw">ggplotGrob</span>(plots[[i]])
    widths[[i]] &lt;-<span class="st"> </span>grobs[[i]]<span class="op">$</span>widths[<span class="dv">2</span><span class="op">:</span><span class="dv">5</span>]
  }
  maxwidth &lt;-<span class="st"> </span><span class="kw">do.call</span>(grid<span class="op">::</span>unit.pmax, widths)
  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(grobs)){
    grobs[[i]]<span class="op">$</span>widths[<span class="dv">2</span><span class="op">:</span><span class="dv">5</span>] &lt;-<span class="st"> </span><span class="kw">as.list</span>(maxwidth)
  }
  
  <span class="co"># Generate plot</span>
  grob.title &lt;-<span class="st"> </span><span class="kw">textGrob</span>(sample, <span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">20</span>))
  <span class="kw">pdf</span>(<span class="kw">paste</span>(output_folder,<span class="st">&quot;Graphs/&quot;</span>,sample,<span class="st">&quot;.pdf&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="dt">width =</span> <span class="dv">16</span>, <span class="dt">height =</span> <span class="dv">8</span>) <span class="co"># Open a new pdf file</span>
  <span class="kw">grid.arrange</span>(<span class="dt">grobs=</span>grobs, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">top =</span> grob.title)
  <span class="kw">graphics.off</span>()
  
  
  <span class="cf">if</span>(diversity<span class="op">==</span><span class="ot">TRUE</span>){
    clonal_abundance_mat &lt;-<span class="st"> </span>clonal_abundance
    clonal_abundance<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">rownames</span>(clonal_abundance)
    clonal_abundance_mat<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">rownames</span>(clonal_abundance_mat)
    clonal_abundance_count &lt;-<span class="st"> </span><span class="kw">data.table</span>(clonal_abundance,<span class="dt">key=</span><span class="st">&quot;Count&quot;</span>)
    shannon &lt;-<span class="st"> </span><span class="kw">diversity</span>(clonal_abundance_mat[,<span class="st">&quot;Count&quot;</span>])
    <span class="kw">write.csv2</span>(shannon,<span class="kw">paste</span>(output_folder,<span class="st">&quot;Diversity_score/&quot;</span>,sample,<span class="st">&quot;.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))
  }
}


stats_output&lt;-<span class="cf">function</span>(sample){
  <span class="co">#Load in the data</span>
  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;VARIANT_SELECTION_LAM&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    LAM_VARIANTS_MAT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,sample,<span class="st">&quot;_VARIANT_SELECTION_LAM.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    LAM_VARIANTS_SNP &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Mutant))[,<span class="dv">1</span>])
    LAM_VARIANTS_Protein &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Protein))[,<span class="dv">1</span>])
  } <span class="cf">else</span> {
    <span class="kw">print</span>(<span class="st">&quot;No variant selection performed!&quot;</span>) 
    <span class="kw">return</span>(<span class="st">&quot;next&quot;</span>)
  }
  AF &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;AF.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - variant allele frequency</span>
  DP &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;DP.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                     <span class="co"># - read depth</span>
  GQ &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;GQ.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - quality scores</span>
  NGT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;NGT.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) <span class="co"># - genotype call converted to categorical (0-reference, 1-heterozygous mutation, 2-homozygous mutation, 3-unknown)</span>
  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cdna&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;gene&quot;</span>],x[<span class="st">&quot;cdna&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(x[<span class="st">&quot;amplicon&quot;</span>]<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;MSK_RL_AMP54&quot;</span>, <span class="st">&quot;MSK_RL_AMP55&quot;</span>, <span class="st">&quot;MSK_RL_AMP56&quot;</span>,<span class="st">&quot;MSK_RL_AMP57&quot;</span>,<span class="st">&quot;MSK_RL_AMP58&quot;</span>),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;POS&quot;</span>],x[<span class="st">&quot;ALT&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  } <span class="cf">else</span> {
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;Variants.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    <span class="kw">colnames</span>(SNP_INFO)[<span class="dv">3</span>] &lt;-<span class="st"> &quot;protein&quot;</span>
    SNP_INFO<span class="op">$</span>cDNA &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>cDNA)
    SNP_INFO<span class="op">$</span>protein &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>protein)
    SNP_INFO<span class="op">$</span>Variant &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Variant)
    SNP_INFO<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Gene)
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cDNA&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;Gene&quot;</span>],x[<span class="st">&quot;cDNA&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^chr13&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>]),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  }
  
  SNP_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_SNP
  protein_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_Protein
  SNP_protein_key &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Mutant&quot;</span>=SNP_changes_of_interest,<span class="st">&quot;Protein&quot;</span>=protein_changes_of_interest)
  
  <span class="co"># Cell and clone selection</span>
  distribution_of_unknowns_by_variant &lt;-<span class="st"> </span><span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">2</span>,table)
  <span class="co">#print(distribution_of_unknowns_by_variant)</span>
  cells_with_unknown&lt;-NGT[<span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">1</span>,<span class="dt">FUN=</span><span class="cf">function</span>(x){<span class="kw">sum</span>(x<span class="op">==</span><span class="dv">3</span>)<span class="op">&gt;</span><span class="dv">0</span>}),<span class="st">&quot;Cell&quot;</span>]
  matrix_of_interest&lt;-NGT[<span class="op">!</span>NGT<span class="op">$</span>Cell<span class="op">%in%</span>cells_with_unknown,SNP_changes_of_interest]
  bulk_VAF_order&lt;-SNP_INFO[<span class="kw">colnames</span>(matrix_of_interest)[<span class="kw">order</span>(<span class="kw">colSums</span>(matrix_of_interest),<span class="dt">decreasing=</span><span class="ot">TRUE</span>)],<span class="st">&quot;protein&quot;</span>]
  bulk_VAF_order &lt;-<span class="st"> </span>bulk_VAF_order[<span class="op">!</span><span class="kw">duplicated</span>(bulk_VAF_order)]
  matrix_of_interest<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">apply</span>(matrix_of_interest,<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)})

  <span class="co"># Aggregate data to get clone counts and architecture.</span>
  dedup&lt;-matrix_of_interest[<span class="op">!</span><span class="kw">duplicated</span>(matrix_of_interest),]
  clonal_abundance &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Count&quot;</span>=<span class="kw">as.matrix</span>(<span class="kw">table</span>(matrix_of_interest<span class="op">$</span>Clone)))
  clonal_abundance<span class="op">$</span>Clone &lt;-<span class="st">  </span><span class="kw">factor</span>(<span class="kw">rownames</span>(clonal_abundance),<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="kw">rownames</span>(clonal_abundance))))
  clonal_abundance<span class="op">$</span>Count &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(clonal_abundance<span class="op">$</span>Count)
  clonal_abundance<span class="op">$</span>Dominance &lt;-<span class="st"> </span><span class="kw">apply</span>(clonal_abundance,<span class="dv">1</span>,<span class="cf">function</span>(x){
    (<span class="kw">as.numeric</span>(x[<span class="st">&quot;Count&quot;</span>])<span class="op">/</span><span class="kw">sum</span>(clonal_abundance[<span class="st">&quot;Count&quot;</span>]) ) <span class="op">/</span><span class="st"> </span>(<span class="kw">max</span>(<span class="kw">as.numeric</span>(clonal_abundance[,<span class="st">&quot;Count&quot;</span>]))<span class="op">/</span><span class="kw">sum</span>(clonal_abundance[<span class="st">&quot;Count&quot;</span>]))
  })
  clonal_abundance<span class="op">$</span>Mutation_count &lt;-<span class="st"> </span><span class="kw">apply</span>(clonal_abundance,<span class="dv">1</span>,<span class="cf">function</span>(x){
    <span class="kw">sum</span>(<span class="kw">as.numeric</span>(<span class="kw">strsplit</span>( <span class="kw">as.character</span>(x[<span class="dv">2</span>]),<span class="dt">split=</span><span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]]))
  })
  shannon &lt;-<span class="st"> </span><span class="kw">diversity</span>(clonal_abundance[,<span class="st">&quot;Count&quot;</span>])
  <span class="kw">print</span>(shannon)
  matrix_output &lt;-<span class="st"> </span>dedup<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="op">-</span>Clone)
  matrix_output_t &lt;-<span class="st"> </span><span class="kw">t</span>(matrix_output)
  <span class="kw">rownames</span>(matrix_output_t) &lt;-<span class="st"> </span><span class="dv">0</span><span class="op">:</span>(<span class="kw">dim</span>(matrix_output_t)[[<span class="dv">1</span>]]<span class="op">-</span><span class="dv">1</span>)
  <span class="kw">write.table</span>(shannon,<span class="kw">paste0</span>(output_folder,<span class="st">&quot;Diversity_score/&quot;</span>,sample,<span class="st">&quot;.txt&quot;</span>),<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">row.names=</span><span class="ot">FALSE</span>)
  <span class="kw">write.table</span>(matrix_output_t,<span class="kw">paste0</span>(output_folder,<span class="st">&quot;Genotype_matrix/&quot;</span>,sample,<span class="st">&quot;.txt&quot;</span>),<span class="dt">sep=</span><span class="st">&quot; &quot;</span>,<span class="dt">row.names=</span><span class="ot">TRUE</span>,<span class="dt">quote=</span><span class="ot">FALSE</span>,<span class="dt">col.names =</span> <span class="ot">FALSE</span>)
  <span class="kw">write.table</span>(clonal_abundance,<span class="kw">paste0</span>(output_folder,<span class="st">&quot;Summary_mat/&quot;</span>,sample,<span class="st">&quot;.txt&quot;</span>),<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">row.names=</span><span class="ot">FALSE</span>)
}


permute_data &lt;-<span class="st"> </span><span class="cf">function</span>(sample,nrun){
  <span class="co">#Load in the data</span>
  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;VARIANT_SELECTION_LAM&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    LAM_VARIANTS_MAT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,sample,<span class="st">&quot;_VARIANT_SELECTION_LAM.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    LAM_VARIANTS_SNP &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Mutant))[,<span class="dv">1</span>])
    LAM_VARIANTS_Protein &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Protein))[,<span class="dv">1</span>])
  } <span class="cf">else</span> {
    <span class="kw">print</span>(<span class="st">&quot;No variant selection performed!&quot;</span>) 
    <span class="kw">return</span>(<span class="st">&quot;next&quot;</span>)
  }
  AF &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;AF.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - variant allele frequency</span>
  DP &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;DP.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                     <span class="co"># - read depth</span>
  GQ &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;GQ.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - quality scores</span>
  NGT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;NGT.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) <span class="co"># - genotype call converted to categorical (0-reference, 1-heterozygous mutation, 2-homozygous mutation, 3-unknown)</span>
  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cdna&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;gene&quot;</span>],x[<span class="st">&quot;cdna&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(x[<span class="st">&quot;amplicon&quot;</span>]<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;MSK_RL_AMP54&quot;</span>, <span class="st">&quot;MSK_RL_AMP55&quot;</span>, <span class="st">&quot;MSK_RL_AMP56&quot;</span>,<span class="st">&quot;MSK_RL_AMP57&quot;</span>,<span class="st">&quot;MSK_RL_AMP58&quot;</span>),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;POS&quot;</span>],x[<span class="st">&quot;ALT&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  } <span class="cf">else</span> {
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;Variants.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    <span class="kw">colnames</span>(SNP_INFO)[<span class="dv">3</span>] &lt;-<span class="st"> &quot;protein&quot;</span>
    SNP_INFO<span class="op">$</span>cDNA &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>cDNA)
    SNP_INFO<span class="op">$</span>protein &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>protein)
    SNP_INFO<span class="op">$</span>Variant &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Variant)
    SNP_INFO<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Gene)
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cDNA&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;Gene&quot;</span>],x[<span class="st">&quot;cDNA&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^chr13&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>]),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  }
  
  SNP_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_SNP
  protein_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_Protein
  SNP_protein_key &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Mutant&quot;</span>=SNP_changes_of_interest,<span class="st">&quot;Protein&quot;</span>=protein_changes_of_interest)
  
  <span class="co"># Cell and clone selection</span>
  distribution_of_unknowns_by_variant &lt;-<span class="st"> </span><span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">2</span>,table)
  <span class="co">#print(distribution_of_unknowns_by_variant)</span>
  cells_with_unknown&lt;-NGT[<span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">1</span>,<span class="dt">FUN=</span><span class="cf">function</span>(x){<span class="kw">sum</span>(x<span class="op">==</span><span class="dv">3</span>)<span class="op">&gt;</span><span class="dv">0</span>}),<span class="st">&quot;Cell&quot;</span>]
  matrix_of_interest&lt;-NGT[<span class="op">!</span>NGT<span class="op">$</span>Cell<span class="op">%in%</span>cells_with_unknown,SNP_changes_of_interest]
  bulk_VAF_order&lt;-SNP_INFO[<span class="kw">colnames</span>(matrix_of_interest)[<span class="kw">order</span>(<span class="kw">colSums</span>(matrix_of_interest),<span class="dt">decreasing=</span><span class="ot">TRUE</span>)],<span class="st">&quot;protein&quot;</span>]
  bulk_VAF_order &lt;-<span class="st"> </span>bulk_VAF_order[<span class="op">!</span><span class="kw">duplicated</span>(bulk_VAF_order)]
  matrix_of_interest<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">apply</span>(matrix_of_interest,<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)})
  
  <span class="co"># Aggregate data to get clone counts and architecture.</span>
  dedup&lt;-matrix_of_interest[<span class="op">!</span><span class="kw">duplicated</span>(matrix_of_interest),]
  clonal_abundance &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Count&quot;</span>=<span class="kw">as.matrix</span>(<span class="kw">table</span>(matrix_of_interest<span class="op">$</span>Clone)))
  clonal_abundance<span class="op">$</span>Clone &lt;-<span class="st">  </span><span class="kw">factor</span>(<span class="kw">rownames</span>(clonal_abundance),<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(<span class="kw">rownames</span>(clonal_abundance))))
  clonal_abundance<span class="op">$</span>Count &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(clonal_abundance<span class="op">$</span>Count)
  
  clonal_architecture &lt;-<span class="st"> </span><span class="kw">melt</span>(dedup)
  <span class="kw">colnames</span>(clonal_architecture) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Clone&quot;</span>,<span class="st">&quot;Mutant&quot;</span>,<span class="st">&quot;Genotype&quot;</span>)
  clonal_architecture<span class="op">$</span>Genotype &lt;-<span class="st"> </span><span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">3</span>,<span class="ot">NA</span>,
                                         <span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">0</span>,<span class="st">&quot;WT&quot;</span>,
                                                <span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">1</span>,<span class="st">&quot;Heterozygous&quot;</span>,
                                                       <span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">2</span>,<span class="st">&quot;Homozygous&quot;</span>,clonal_architecture<span class="op">$</span>Genotype))))
  clonal_architecture<span class="op">$</span>Genotype  &lt;-<span class="st"> </span><span class="kw">factor</span>(clonal_architecture<span class="op">$</span>Genotype ,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>))

  results_holder &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">list</span>(<span class="dv">1</span><span class="op">:</span>nrun))
  randomization_list&lt;-<span class="kw">lapply</span>(results_holder,<span class="cf">function</span>(x){
    <span class="kw">apply</span>(matrix_of_interest[,<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">colnames</span>(matrix_of_interest))<span class="op">-</span><span class="dv">1</span>],<span class="dv">2</span>,<span class="cf">function</span>(x){
      y &lt;-<span class="st"> </span><span class="kw">sample</span>(x,<span class="kw">length</span>(x),<span class="dt">replace=</span><span class="ot">FALSE</span>)
    })})
  
  <span class="co">#randomization_list&lt;-plyr::alply(randomization_array,3)</span>
  <span class="co"># names(randomization_list) &lt;- NULL</span>
  
  empirical_distribution&lt;-<span class="kw">suppressMessages</span>(<span class="kw">lapply</span>(randomization_list, <span class="cf">function</span>(randomization){
    randomization<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">apply</span>(randomization,<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)})
    loop_dedup&lt;-matrix_of_interest[<span class="op">!</span><span class="kw">duplicated</span>(randomization),]
    loop_clonal_architecture &lt;-<span class="st"> </span><span class="kw">melt</span>(loop_dedup)
    <span class="kw">colnames</span>(loop_clonal_architecture) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Clone&quot;</span>,<span class="st">&quot;Mutant&quot;</span>,<span class="st">&quot;Genotype&quot;</span>)
    loop_clonal_architecture<span class="op">$</span>Genotype &lt;-<span class="st"> </span><span class="kw">ifelse</span>(loop_clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">3</span>,<span class="ot">NA</span>,
                                                <span class="kw">ifelse</span>(loop_clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">0</span>,<span class="st">&quot;WT&quot;</span>,
                                                       <span class="kw">ifelse</span>(loop_clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">1</span>,<span class="st">&quot;Heterozygous&quot;</span>,
                                                              <span class="kw">ifelse</span>(loop_clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">2</span>,<span class="st">&quot;Homozygous&quot;</span>,clonal_architecture<span class="op">$</span>Genotype))))
    loop_clonal_architecture<span class="op">$</span>Genotype  &lt;-<span class="st"> </span><span class="kw">factor</span>(loop_clonal_architecture<span class="op">$</span>Genotype ,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>))
    loop_clonal_abundance &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Count&quot;</span>=<span class="kw">as.matrix</span>(<span class="kw">table</span>(randomization<span class="op">$</span>Clone)))
    loop_clonal_abundance<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">rownames</span>(loop_clonal_abundance)
    <span class="kw">return</span>(<span class="kw">data.frame</span>(loop_clonal_abundance))
  }))
  
  
  empirical_distribution_mat&lt;-empirical_distribution<span class="op">%&gt;%</span>purrr<span class="op">::</span><span class="kw">reduce</span>(dplyr<span class="op">::</span>full_join,<span class="dt">by=</span><span class="st">&quot;Clone&quot;</span>)
  <span class="kw">colnames</span>(empirical_distribution_mat)&lt;-<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">colnames</span>(empirical_distribution_mat))
  <span class="kw">rownames</span>(empirical_distribution_mat) &lt;-<span class="st"> </span>empirical_distribution_mat[,<span class="dv">2</span>]
  empirical_distribution_mat &lt;-<span class="st"> </span>empirical_distribution_mat[,<span class="op">-</span><span class="dv">2</span>]
  empirical_distribution_mat[<span class="kw">is.na</span>(empirical_distribution_mat)]&lt;-<span class="dv">0</span>
  
  clonal_abundance<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">as.character</span>(clonal_abundance<span class="op">$</span>Clone)
  clonal_abundance<span class="op">$</span>Count &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(clonal_abundance<span class="op">$</span>Count)
  clonal_abundance<span class="op">$</span>pvalue &lt;-<span class="kw">apply</span>(clonal_abundance,<span class="dv">1</span>,<span class="cf">function</span>(x){
  <span class="dv">1</span><span class="op">-</span><span class="st">  </span><span class="kw">sum</span>(<span class="kw">as.numeric</span>(x[[<span class="st">&quot;Count&quot;</span>]])<span class="op">&gt;</span>empirical_distribution_mat[x[[<span class="st">&quot;Clone&quot;</span>]],])<span class="op">/</span>nrun
    })
  
  
  <span class="co">#print(dedup)</span>
  <span class="kw">colnames</span>(clonal_abundance)[<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="kw">colnames</span>(dedup)[<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(SNP_changes_of_interest)],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;__&quot;</span>)
  clonal_abundance &lt;-<span class="st"> </span>clonal_abundance[<span class="kw">order</span>(clonal_abundance[,<span class="st">&quot;Count&quot;</span>],<span class="dt">decreasing =</span> <span class="ot">TRUE</span>),]
  clonal_abundance<span class="op">$</span>Dominance &lt;-<span class="st"> </span><span class="kw">apply</span>(clonal_abundance,<span class="dv">1</span>,<span class="cf">function</span>(x){
    (<span class="kw">as.numeric</span>(x[<span class="st">&quot;Count&quot;</span>])<span class="op">/</span><span class="kw">sum</span>(clonal_abundance[<span class="st">&quot;Count&quot;</span>]) ) <span class="op">/</span><span class="st"> </span>(<span class="kw">max</span>(<span class="kw">as.numeric</span>(clonal_abundance[,<span class="st">&quot;Count&quot;</span>]))<span class="op">/</span><span class="kw">sum</span>(clonal_abundance[<span class="st">&quot;Count&quot;</span>]))
      })
  clonal_abundance<span class="op">$</span>Poisson &lt;-<span class="st"> </span><span class="kw">apply</span>(clonal_abundance,<span class="dv">1</span>,<span class="cf">function</span>(x){
        <span class="kw">poisson.test</span>(<span class="kw">as.numeric</span>(x[<span class="st">&quot;Count&quot;</span>]),<span class="kw">mean</span>(<span class="kw">as.numeric</span>(clonal_abundance<span class="op">$</span>Count)),<span class="dt">alternative=</span><span class="st">&quot;greater&quot;</span>)<span class="op">$</span>p.value
      })
  clonal_abundance<span class="op">$</span>Mutation_count &lt;-<span class="st"> </span><span class="kw">apply</span>(clonal_abundance,<span class="dv">1</span>,<span class="cf">function</span>(x){
    <span class="kw">sum</span>(<span class="kw">as.numeric</span>(<span class="kw">strsplit</span>( <span class="kw">as.character</span>(x[<span class="dv">2</span>]),<span class="dt">split=</span><span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]]))
  })
  
  <span class="kw">write.table</span>(clonal_abundance,<span class="kw">paste0</span>(output_folder,<span class="st">&quot;Summary_mat/&quot;</span>,sample,<span class="st">&quot;.txt&quot;</span>),<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">row.names=</span><span class="ot">FALSE</span>)
  
}



replot_clonal_selection &lt;-<span class="st"> </span><span class="cf">function</span>(sample){
  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;VARIANT_SELECTION_LAM&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    LAM_VARIANTS_MAT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,sample,<span class="st">&quot;_VARIANT_SELECTION_LAM.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    LAM_VARIANTS_SNP &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Mutant))[,<span class="dv">1</span>])
    LAM_VARIANTS_Protein &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Protein))[,<span class="dv">1</span>])
  } <span class="cf">else</span> {
    <span class="kw">print</span>(<span class="st">&quot;No variant selection performed!&quot;</span>) 
    <span class="kw">return</span>(<span class="st">&quot;next&quot;</span>)
  }
  DP &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;DP.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                     <span class="co"># - read depth</span>
  NGT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;NGT.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) <span class="co"># - genotype call converted to categorical (0-reference, 1-heterozygous mutation, 2-homozygous mutation, 3-unknown)</span>
  
  <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(NGT)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cdna&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;gene&quot;</span>],x[<span class="st">&quot;cdna&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(x[<span class="st">&quot;amplicon&quot;</span>]<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;MSK_RL_AMP54&quot;</span>, <span class="st">&quot;MSK_RL_AMP55&quot;</span>, <span class="st">&quot;MSK_RL_AMP56&quot;</span>,<span class="st">&quot;MSK_RL_AMP57&quot;</span>,<span class="st">&quot;MSK_RL_AMP58&quot;</span>),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;POS&quot;</span>],x[<span class="st">&quot;ALT&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  } <span class="cf">else</span> {
    SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;Variants.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
    <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(NGT)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
    <span class="kw">colnames</span>(SNP_INFO)[<span class="dv">3</span>] &lt;-<span class="st"> &quot;protein&quot;</span>
    SNP_INFO<span class="op">$</span>cDNA &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>cDNA)
    SNP_INFO<span class="op">$</span>protein &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>protein)
    SNP_INFO<span class="op">$</span>Variant &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Variant)
    SNP_INFO<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Gene)
    SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
      <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
             <span class="kw">ifelse</span>(x[<span class="st">&quot;cDNA&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;Gene&quot;</span>],x[<span class="st">&quot;cDNA&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                    <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^chr13&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>]),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
    })<span class="co"># - variant metadata</span>
  }
  
  SNP_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_SNP
  protein_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_Protein
  SNP_protein_key &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Mutant&quot;</span>=SNP_changes_of_interest,<span class="st">&quot;Protein&quot;</span>=protein_changes_of_interest)
  <span class="kw">rownames</span>(SNP_protein_key) &lt;-<span class="st"> </span>SNP_protein_key<span class="op">$</span>Mutant
  <span class="co"># Cell and clone selection</span>
  distribution_of_unknowns_by_variant &lt;-<span class="st"> </span><span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">2</span>,table)
  <span class="co">#print(distribution_of_unknowns_by_variant)</span>
  cells_with_unknown&lt;-NGT[<span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">1</span>,<span class="dt">FUN=</span><span class="cf">function</span>(x){<span class="kw">sum</span>(x<span class="op">==</span><span class="dv">3</span>)<span class="op">&gt;</span><span class="dv">0</span>}),<span class="st">&quot;Cell&quot;</span>]
  matrix_of_interest&lt;-NGT[<span class="op">!</span>NGT<span class="op">$</span>Cell<span class="op">%in%</span>cells_with_unknown,SNP_changes_of_interest]
  bulk_VAF_order&lt;-SNP_INFO[<span class="kw">colnames</span>(matrix_of_interest)[<span class="kw">order</span>(<span class="kw">colSums</span>(matrix_of_interest),<span class="dt">decreasing=</span><span class="ot">TRUE</span>)],<span class="st">&quot;protein&quot;</span>]
  bulk_VAF_order &lt;-<span class="st"> </span>bulk_VAF_order[<span class="op">!</span><span class="kw">duplicated</span>(bulk_VAF_order)]
  matrix_of_interest<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">apply</span>(matrix_of_interest,<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)})
  
  dedup&lt;-matrix_of_interest[<span class="op">!</span><span class="kw">duplicated</span>(matrix_of_interest),]
  <span class="kw">colnames</span>(dedup) &lt;-<span class="st"> </span>SNP_protein_key[<span class="kw">colnames</span>(dedup),<span class="st">&quot;Protein&quot;</span>]
  clonal_architecture &lt;-<span class="st"> </span><span class="kw">melt</span>(dedup)
  <span class="kw">colnames</span>(clonal_architecture) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Clone&quot;</span>,<span class="st">&quot;Mutant&quot;</span>,<span class="st">&quot;Genotype&quot;</span>)
  clonal_architecture<span class="op">$</span>Genotype &lt;-<span class="st"> </span><span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">3</span>,<span class="ot">NA</span>,
                                         <span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">0</span>,<span class="st">&quot;WT&quot;</span>,
                                                <span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">1</span>,<span class="st">&quot;Heterozygous&quot;</span>,
                                                       <span class="kw">ifelse</span>(clonal_architecture<span class="op">$</span>Genotype<span class="op">==</span><span class="dv">2</span>,<span class="st">&quot;Homozygous&quot;</span>,clonal_architecture<span class="op">$</span>Genotype))))
  clonal_architecture<span class="op">$</span>Genotype  &lt;-<span class="st"> </span><span class="kw">factor</span>(clonal_architecture<span class="op">$</span>Genotype ,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>,<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>))
  clonal_abundance &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Count&quot;</span>=<span class="kw">as.matrix</span>(<span class="kw">table</span>(matrix_of_interest<span class="op">$</span>Clone)))
  clonal_abundance<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">rownames</span>(clonal_abundance)
  clonal_abundance &lt;-<span class="st"> </span><span class="kw">data.table</span>(clonal_abundance,<span class="dt">key=</span><span class="st">&quot;Count&quot;</span>)
  clonal_abundance<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">factor</span>(clonal_abundance<span class="op">$</span>Clone,<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(clonal_abundance<span class="op">$</span>Clone)))
  clonal_architecture<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">factor</span>(<span class="kw">as.character</span>(clonal_architecture<span class="op">$</span>Clone),<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">as.character</span>(clonal_abundance<span class="op">$</span>Clone)))
  clonal_architecture<span class="op">$</span>Mutant &lt;-<span class="kw">factor</span>(clonal_architecture<span class="op">$</span>Mutant , <span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">as.character</span>(bulk_VAF_order)))
  
  clones_import &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="kw">paste0</span>(<span class="st">&quot;/Volumes/LevineLab/Levine Lab/Bobby/Collaborations/MissionBio/08_08_19/Summary_mat/&quot;</span>,sample,<span class="st">&quot;.txt&quot;</span>),<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)
  <span class="kw">colnames</span>(clones_import)[<span class="dv">2</span>] &lt;-<span class="st"> &quot;Clone&quot;</span>
  select_clones_full &lt;-<span class="st"> </span>clones_import <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(pvalue<span class="op">&lt;</span><span class="fl">0.1</span><span class="op">|</span>Poisson<span class="op">&lt;</span><span class="fl">0.05</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Clone)
  select_clones_dominant &lt;-<span class="st"> </span>clones_import <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(pvalue<span class="op">&lt;</span><span class="fl">0.1</span><span class="op">|</span>Poisson<span class="op">&lt;</span><span class="fl">0.05</span>)<span class="op">%&gt;%</span><span class="kw">filter</span>(Dominance<span class="op">&gt;</span><span class="fl">0.05</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Clone)

  
  
  <span class="cf">if</span>(<span class="kw">length</span>(select_clones_dominant[,<span class="dv">1</span>])<span class="op">&gt;</span><span class="dv">3</span>){
    clonal_architecture_subset &lt;-<span class="st"> </span>clonal_architecture<span class="op">%&gt;%</span><span class="kw">filter</span>(Clone<span class="op">%in%</span>select_clones_dominant[,<span class="dv">1</span>])
    clonal_abundance_subset &lt;-<span class="st"> </span>clonal_abundance<span class="op">%&gt;%</span><span class="kw">filter</span>(Clone<span class="op">%in%</span>select_clones_dominant[,<span class="dv">1</span>])
    name_var &lt;-<span class="st"> &quot;VAF Subset&quot;</span>
  } <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(select_clones_full[,<span class="dv">1</span>])<span class="op">&gt;</span><span class="dv">20</span>){
    <span class="kw">print</span>(<span class="st">&quot;Complex clonal picture, skipping&quot;</span>)
    <span class="kw">return</span>(<span class="cf">next</span>)
  } <span class="cf">else</span> <span class="cf">if</span>(<span class="kw">length</span>(select_clones_full[,<span class="dv">1</span>])<span class="op">&lt;</span><span class="dv">2</span>){
    <span class="kw">print</span>(<span class="st">&quot;Simple clonal picture, plotting all clones&quot;</span>)
    clonal_architecture_subset &lt;-<span class="st"> </span>clonal_architecture
    clonal_abundance_subset &lt;-<span class="st"> </span>clonal_abundance
    name_var &lt;-<span class="st"> &quot;All Clones&quot;</span>
  } <span class="cf">else</span> {
    clonal_architecture_subset &lt;-<span class="st"> </span>clonal_architecture<span class="op">%&gt;%</span><span class="kw">filter</span>(Clone<span class="op">%in%</span>select_clones_full[,<span class="dv">1</span>])
    clonal_abundance_subset &lt;-<span class="st"> </span>clonal_abundance<span class="op">%&gt;%</span><span class="kw">filter</span>(Clone<span class="op">%in%</span>select_clones_full[,<span class="dv">1</span>])
    name_var &lt;-<span class="st"> &quot;Full&quot;</span>
  }
  <span class="co"># Generate clonal architecture heatmap</span>
  gg_heatmap &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> clonal_architecture_subset, <span class="kw">aes</span>(<span class="dt">x =</span> Clone, <span class="dt">y =</span> <span class="kw">factor</span>(Mutant), <span class="dt">fill =</span> Genotype)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">1</span>],
                               <span class="st">&quot;Heterozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">3</span>],
                               <span class="st">&quot;Homozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">6</span>],
                               <span class="st">&quot;Unknown&quot;</span>=<span class="st">&quot;grey50&quot;</span>),<span class="st">&quot;Genotype&quot;</span>)  <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Mutation&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>,
          <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), 
          <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
          <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),
          <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(),
          <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>),<span class="st">&quot;cm&quot;</span>))
  
  <span class="co"># Generate clonal abundance barplot</span>
  gg_clonal_barplot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> <span class="kw">data.frame</span>(clonal_abundance_subset), <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(Clone), <span class="dt">y =</span> Count)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill =</span> Count)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_gray</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>()<span class="op">+</span>
<span class="st">    </span><span class="kw">ylim</span>(<span class="dv">0</span>,<span class="kw">max</span>(clonal_abundance<span class="op">$</span>Count)<span class="op">*</span><span class="fl">1.3</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Cell Count&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>Count), <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="dt">width=</span><span class="fl">0.9</span>), <span class="dt">vjust=</span><span class="op">-</span><span class="fl">0.25</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_distiller</span>(<span class="dt">name =</span> <span class="st">&quot;Value&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="dt">direction =</span> <span class="dv">1</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(),  <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
          <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,
          <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="op">-</span><span class="fl">0.2</span>,<span class="dv">1</span>),<span class="st">&quot;cm&quot;</span>))

  <span class="co"># Compute spacing of graphs</span>
  plots &lt;-<span class="st"> </span><span class="kw">list</span>(gg_clonal_barplot,gg_heatmap)
  grobs &lt;-<span class="st"> </span><span class="kw">list</span>()
  widths &lt;-<span class="st"> </span><span class="kw">list</span>()
  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(plots)){
    grobs[[i]] &lt;-<span class="st"> </span><span class="kw">ggplotGrob</span>(plots[[i]])
    widths[[i]] &lt;-<span class="st"> </span>grobs[[i]]<span class="op">$</span>widths[<span class="dv">2</span><span class="op">:</span><span class="dv">5</span>]
  }
  maxwidth &lt;-<span class="st"> </span><span class="kw">do.call</span>(grid<span class="op">::</span>unit.pmax, widths)
  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(grobs)){
    grobs[[i]]<span class="op">$</span>widths[<span class="dv">2</span><span class="op">:</span><span class="dv">5</span>] &lt;-<span class="st"> </span><span class="kw">as.list</span>(maxwidth)
  }
  
  <span class="co"># Generate plot</span>
  grob.title &lt;-<span class="st"> </span><span class="kw">textGrob</span>(<span class="kw">paste</span>(sample,<span class="st">&quot;-&quot;</span>,name_var), <span class="dt">hjust =</span> <span class="fl">0.5</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">20</span>))
  <span class="kw">pdf</span>(<span class="kw">paste</span>(output_folder,<span class="st">&quot;Graphs_significant/&quot;</span>,sample,<span class="st">&quot;.pdf&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>), <span class="dt">width =</span> <span class="dv">16</span>, <span class="dt">height =</span> <span class="dv">8</span>) <span class="co"># Open a new pdf file</span>
  <span class="kw">grid.arrange</span>(<span class="dt">grobs=</span>grobs, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">top =</span> grob.title)
  <span class="kw">graphics.off</span>()

}


plot_clonal_heatmap_and_barplot &lt;-<span class="cf">function</span>(sample,output_folder,clonal_abundance,clonal_architecture,clone_cell_count,shrink)
{
  <span class="co">#clonal_abundance_subset &lt;-data.frame( clonal_abundance[[sample]])%&gt;%filter(Count&gt;=5)</span>
  <span class="co">#clonal_architecture_subset &lt;- data.frame(clonal_architecture[[sample]])%&gt;%</span>
   <span class="co">#                                         filter(Clone%in%as.character(clonal_abundance_subset$Clone))</span>
  clonal_abundance_subset&lt;-final_sample_summary[[sample]]<span class="op">$</span>Clones<span class="co">#clonal_abundance</span>
  clonal_abundance_subset<span class="op">$</span>Clone&lt;-<span class="kw">factor</span>(final_sample_summary[[sample]]<span class="op">$</span>Clones[,<span class="st">&quot;Clone&quot;</span>],<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">c</span>(final_sample_summary[[sample]]<span class="op">$</span>Clones[,<span class="st">&quot;Clone&quot;</span>])))<span class="co">#clonal_abundance</span>
  clonal_architecture_subset&lt;-final_sample_summary[[sample]]<span class="op">$</span>Architecture<span class="op">%&gt;%</span>
<span class="st">                                      </span><span class="kw">filter</span>(Clone<span class="op">%in%</span><span class="kw">as.character</span>(clonal_abundance_subset<span class="op">$</span>Clone))
  
  clonal_architecture_subset<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">factor</span>(clonal_architecture_subset<span class="op">$</span>Clone, <span class="dt">levels=</span><span class="kw">levels</span>(clonal_abundance_subset<span class="op">$</span>Clone))
  gg_heatmap &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> clonal_architecture_subset, <span class="kw">aes</span>(<span class="dt">x =</span> Clone, <span class="dt">y =</span> <span class="kw">factor</span>(Mutant,<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(Mutant)))), <span class="dt">fill =</span> Genotype)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span><span class="co"># scale_y_discrete(limits = rev(levels(Mutant)))+</span>

<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">1</span>],
                               <span class="st">&quot;Heterozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">3</span>],
                               <span class="st">&quot;Homozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">6</span>],
                               <span class="st">&quot;Unknown&quot;</span>=<span class="st">&quot;grey50&quot;</span>),<span class="st">&quot;Genotype&quot;</span>)  <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">6</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Mutation&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>,
          <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), 
          <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
          <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),
          <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(),
          <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>),<span class="st">&quot;cm&quot;</span>))
  
  <span class="co"># Generate clonal abundance barplot</span>
  gg_clonal_barplot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> <span class="kw">data.frame</span>(clonal_abundance_subset), <span class="kw">aes</span>(<span class="dt">x =</span> Clone, <span class="dt">y =</span> Count)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill =</span> Count)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_gray</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">6</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">ylim</span>(<span class="dv">0</span>,<span class="kw">max</span>(clonal_abundance_subset<span class="op">$</span>Count)<span class="op">*</span><span class="fl">1.3</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Cell Count&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>Count), <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="dt">width=</span><span class="fl">0.9</span>), <span class="dt">vjust=</span><span class="op">-</span><span class="fl">0.25</span>,<span class="dt">size=</span><span class="dv">1</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_distiller</span>(<span class="dt">name =</span> <span class="st">&quot;Value&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="dt">direction =</span> <span class="dv">1</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(),  <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
          <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,
          <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="op">-</span><span class="fl">0.2</span>,<span class="dv">1</span>),<span class="st">&quot;cm&quot;</span>))
  
  grob.title &lt;-<span class="st"> </span><span class="kw">textGrob</span>(<span class="kw">paste</span>(sample,<span class="st">&quot;-&quot;</span>), <span class="dt">hjust=</span> <span class="fl">0.5</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">12</span>))
  final_plot&lt;-<span class="kw">plot_grid</span>(grob.title,gg_clonal_barplot,gg_heatmap,<span class="dt">ncol=</span><span class="dv">1</span>,<span class="dt">align=</span><span class="st">&quot;hv&quot;</span>,<span class="dt">axis=</span><span class="st">&quot;l&quot;</span>,<span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.1</span>,<span class="dv">1</span>,<span class="fl">0.75</span>))
 
   <span class="kw">save_plot</span>(<span class="kw">paste</span>(output_folder,sample,<span class="st">&quot;.pdf&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>),final_plot, <span class="dt">ncol=</span><span class="dv">1</span>) <span class="co"># Open a new pdf file</span>
}

generate_and_plot_cooccurence &lt;-<span class="st"> </span><span class="cf">function</span>(mut_mat){
  cooccur_mat &lt;-<span class="st"> </span><span class="kw">cooccur</span>(<span class="dt">mat=</span><span class="kw">t</span>(mut_mat),<span class="dt">type=</span><span class="st">&quot;spp_site&quot;</span>,<span class="dt">only_effects =</span> <span class="ot">FALSE</span>,<span class="dt">eff_matrix=</span><span class="ot">TRUE</span>,<span class="dt">thresh=</span><span class="ot">FALSE</span>,<span class="dt">eff_standard=</span><span class="ot">FALSE</span>,<span class="dt">spp_names=</span><span class="ot">TRUE</span>)<span class="op">$</span>results
  cooccur_mat<span class="op">$</span>score&lt;-<span class="st">  </span><span class="kw">ifelse</span>(cooccur_mat[,<span class="st">&quot;p_lt&quot;</span>]<span class="op">&lt;=</span><span class="fl">0.05</span>,<span class="op">-</span><span class="dv">1</span>,<span class="kw">ifelse</span>(cooccur_mat[,<span class="st">&quot;p_gt&quot;</span>]<span class="op">&lt;=</span><span class="fl">0.05</span>,<span class="dv">1</span>,<span class="dv">0</span>))
  cooccur_mat_subset &lt;-<span class="st"> </span><span class="kw">rbind</span>(cooccur_mat[,<span class="kw">c</span>(<span class="st">&quot;sp1_name&quot;</span>,<span class="st">&quot;sp2_name&quot;</span>,<span class="st">&quot;score&quot;</span>)],<span class="kw">c</span>(<span class="kw">setdiff</span>(cooccur_mat<span class="op">$</span>sp2_name,cooccur_mat<span class="op">$</span>sp1_name),<span class="kw">setdiff</span>(cooccur_mat<span class="op">$</span>sp1_name,cooccur_mat<span class="op">$</span>sp2_name),<span class="dv">0</span>))
  cooccur_data_mat &lt;-<span class="st"> </span><span class="kw">dcast</span>(cooccur_mat_subset, sp1_name  <span class="op">~</span><span class="st"> </span>sp2_name, <span class="dt">value.var=</span><span class="st">&quot;score&quot;</span>)
  <span class="kw">rownames</span>(cooccur_data_mat) &lt;-<span class="st"> </span>cooccur_data_mat[,<span class="dv">1</span>]
  cooccur_data_mat2&lt;-cooccur_data_mat[,<span class="op">-</span><span class="dv">1</span>]
  cooccur_data_mat2[<span class="kw">is.na</span>(cooccur_data_mat2)]&lt;-<span class="dv">0</span>
  cooccur_data_mat2[<span class="kw">lower.tri</span>(cooccur_data_mat2)]&lt;-<span class="ot">NA</span>
  cooccur_data_mat2<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">rownames</span>(cooccur_data_mat2)
  cooccur_data_mat_melt &lt;-<span class="st"> </span><span class="kw">na.omit</span>(<span class="kw">melt</span>(cooccur_data_mat2, <span class="st">&#39;Gene&#39;</span>, <span class="dt">variable_name=</span><span class="st">&#39;Gene2&#39;</span>))
  cooccur_data_mat_melt<span class="op">$</span>variable &lt;-<span class="st"> </span><span class="kw">factor</span>(cooccur_data_mat_melt<span class="op">$</span>variable, <span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">levels</span>(cooccur_data_mat_melt<span class="op">$</span>variable)))
  
  
  <span class="co"># Triangle heatmap to compare cohorts</span>
  grob_corrplot&lt;-<span class="kw">ggplot</span>(cooccur_data_mat_melt, <span class="kw">aes</span>(Gene, variable))<span class="op">+</span>
<span class="st">    </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> <span class="kw">factor</span>(value)), <span class="dt">color=</span><span class="st">&#39;grey90&#39;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;-1&quot;</span>=<span class="st">&quot;firebrick3&quot;</span>,<span class="st">&quot;0&quot;</span>=<span class="st">&quot;white&quot;</span>,<span class="st">&quot;1&quot;</span>=<span class="st">&quot;steelblue2&quot;</span>),<span class="st">&quot;Correlation&quot;</span>,
                      <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Mutually Exclusive&quot;</span>,<span class="st">&quot;Not Significant&quot;</span>,<span class="st">&quot;Mutually Inclusive&quot;</span>))<span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">10</span>)<span class="op">+</span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">45</span>,<span class="dt">hjust=</span><span class="dv">1</span>,<span class="dt">vjust=</span><span class="dv">1</span>),
          <span class="dt">axis.line =</span> <span class="kw">element_blank</span>(),
          <span class="dt">legend.position =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>), 
          <span class="dt">legend.justification =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>),
          <span class="dt">legend.direction =</span> <span class="st">&quot;vertical&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.key.size =</span> <span class="kw">unit</span>(<span class="fl">0.5</span>,<span class="st">&quot;line&quot;</span>))
  <span class="kw">return</span>(<span class="kw">list</span>(<span class="st">&quot;plot&quot;</span>=grob_corrplot,
         <span class="st">&quot;data&quot;</span>=cooccur_mat))
}

diversity_from_bulk_VAF &lt;-<span class="st"> </span><span class="cf">function</span>(sample){
    <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;VARIANT_SELECTION_LAM&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
      LAM_VARIANTS_MAT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,sample,<span class="st">&quot;_VARIANT_SELECTION_LAM.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
      LAM_VARIANTS_SNP &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Mutant))[,<span class="dv">1</span>])
      LAM_VARIANTS_Protein &lt;-<span class="st"> </span><span class="kw">as.character</span>((LAM_VARIANTS_MAT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Include<span class="op">==</span><span class="st">&quot;Include&quot;</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(Protein))[,<span class="dv">1</span>])
    } <span class="cf">else</span> {
      <span class="kw">print</span>(<span class="st">&quot;No variant selection performed!&quot;</span>) 
      <span class="kw">return</span>(<span class="st">&quot;next&quot;</span>)
    }
    <span class="co">#AF &lt;- read.csv(paste(&quot;./&quot;,sample,&quot;/&quot;,&quot;AF.csv&quot;,sep=&quot;&quot;))                    # - variant allele frequency</span>
    <span class="co">#DP &lt;- read.csv(paste(&quot;./&quot;,sample,&quot;/&quot;,&quot;DP.csv&quot;,sep=&quot;&quot;))                     # - read depth</span>
    <span class="co">#GQ &lt;- read.csv(paste(&quot;./&quot;,sample,&quot;/&quot;,&quot;GQ.csv&quot;,sep=&quot;&quot;))                    # - quality scores</span>
    NGT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;NGT.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) <span class="co"># - genotype call converted to categorical (0-reference, 1-heterozygous mutation, 2-homozygous mutation, 3-unknown)</span>
    
    <span class="cf">if</span>(<span class="kw">any</span>(<span class="kw">grepl</span>(<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="kw">list.files</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))))){
      SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)) 
      <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(NGT)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
      SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
        <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
               <span class="kw">ifelse</span>(x[<span class="st">&quot;cdna&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;gene&quot;</span>],x[<span class="st">&quot;cdna&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                      <span class="kw">ifelse</span>(x[<span class="st">&quot;amplicon&quot;</span>]<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;MSK_RL_AMP54&quot;</span>, <span class="st">&quot;MSK_RL_AMP55&quot;</span>, <span class="st">&quot;MSK_RL_AMP56&quot;</span>,<span class="st">&quot;MSK_RL_AMP57&quot;</span>,<span class="st">&quot;MSK_RL_AMP58&quot;</span>),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;POS&quot;</span>],x[<span class="st">&quot;ALT&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
      })<span class="co"># - variant metadata</span>
    } <span class="cf">else</span> {
      SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;Variants.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
      <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(NGT)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
      <span class="kw">colnames</span>(SNP_INFO)[<span class="dv">3</span>] &lt;-<span class="st"> &quot;protein&quot;</span>
      SNP_INFO<span class="op">$</span>cDNA &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>cDNA)
      SNP_INFO<span class="op">$</span>protein &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>protein)
      SNP_INFO<span class="op">$</span>Variant &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Variant)
      SNP_INFO<span class="op">$</span>Gene &lt;-<span class="st"> </span><span class="kw">as.character</span>(SNP_INFO<span class="op">$</span>Gene)
      SNP_INFO[,<span class="kw">c</span>(<span class="st">&quot;protein&quot;</span>)] &lt;-<span class="st"> </span><span class="kw">apply</span>(SNP_INFO,<span class="dv">1</span>,<span class="cf">function</span>(x){
        <span class="kw">ifelse</span>(x[<span class="st">&quot;protein&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,x[<span class="st">&quot;protein&quot;</span>],
               <span class="kw">ifelse</span>(x[<span class="st">&quot;cDNA&quot;</span>]<span class="op">!=</span><span class="st">&quot;&quot;</span>,<span class="kw">paste</span>(x[<span class="st">&quot;Gene&quot;</span>],x[<span class="st">&quot;cDNA&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),
                      <span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^chr13&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>]),<span class="kw">paste</span>(<span class="st">&quot;FLT3_INS&quot;</span>,x[<span class="st">&quot;Variant&quot;</span>],<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>),<span class="st">&quot;&quot;</span>)))
      })<span class="co"># - variant metadata</span>
    }
    
    SNP_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_SNP
    protein_changes_of_interest &lt;-<span class="st"> </span>LAM_VARIANTS_Protein
    SNP_protein_key &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="st">&quot;Mutant&quot;</span>=SNP_changes_of_interest,<span class="st">&quot;Protein&quot;</span>=protein_changes_of_interest)
    
    <span class="co"># Cell and clone selection</span>
    distribution_of_unknowns_by_variant &lt;-<span class="st"> </span><span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">2</span>,table)
    <span class="co">#print(distribution_of_unknowns_by_variant)</span>
    cells_with_unknown&lt;-NGT[<span class="kw">apply</span>(NGT[,SNP_changes_of_interest],<span class="dt">MARGIN=</span><span class="dv">1</span>,<span class="dt">FUN=</span><span class="cf">function</span>(x){<span class="kw">sum</span>(x<span class="op">==</span><span class="dv">3</span>)<span class="op">&gt;</span><span class="dv">0</span>}),<span class="st">&quot;Cell&quot;</span>]
    matrix_of_interest&lt;-NGT[<span class="op">!</span>NGT<span class="op">$</span>Cell<span class="op">%in%</span>cells_with_unknown,SNP_changes_of_interest]
    bulk_VAF_order&lt;-SNP_INFO[<span class="kw">colnames</span>(matrix_of_interest)[<span class="kw">order</span>(<span class="kw">colSums</span>(matrix_of_interest),<span class="dt">decreasing=</span><span class="ot">TRUE</span>)],<span class="st">&quot;protein&quot;</span>]
    bulk_VAF_order &lt;-<span class="st"> </span>bulk_VAF_order[<span class="op">!</span><span class="kw">duplicated</span>(bulk_VAF_order)]
    matrix_of_interest<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">apply</span>(matrix_of_interest,<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)})
    
    output&lt;-<span class="kw">diversity</span>(<span class="kw">colSums</span>(matrix_of_interest[<span class="kw">colnames</span>(matrix_of_interest)<span class="op">!=</span><span class="st">&quot;Clone&quot;</span>])<span class="op">/</span><span class="kw">length</span>(<span class="kw">rownames</span>(matrix_of_interest)))
    <span class="kw">write.table</span>(output,<span class="kw">paste0</span>(output_folder,<span class="st">&quot;Diversity_score_faux_VAF/&quot;</span>,sample,<span class="st">&quot;.txt&quot;</span>),<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)
  
}

## Old Functions
plot_variants_AF_and_DP &lt;-<span class="st"> </span><span class="cf">function</span>(sample){
  
  <span class="co">#Load in the data</span>
  AF &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;AF.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - variant allele frequency</span>
  DP &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;DP.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                     <span class="co"># - read depth</span>
  GQ &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;GQ.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                    <span class="co"># - quality scores</span>
  NGT &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;NGT.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))                 <span class="co"># - genotype call converted to categorical (0-reference, 1-heterozygous mutation, 2-homozygous mutation, 3-unknown)</span>
  SNP_INFO &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="kw">paste</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,<span class="st">&quot;SNP_INFO.csv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>))         <span class="co"># - variant metadata</span>
  <span class="kw">rownames</span>(SNP_INFO) &lt;-<span class="kw">colnames</span>(AF)[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]  <span class="co"># - renames SNP info to have the variants as rownames</span>
  
  AF_NGT_Melt&lt;-<span class="kw">inner_join</span>(AF <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unite</span>(<span class="st">&quot;Sample_cell&quot;</span>,<span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>,<span class="st">&quot;Cell&quot;</span>)) <span class="op">%&gt;%</span><span class="kw">melt</span>(<span class="st">&quot;Sample_cell&quot;</span>),
                          NGT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unite</span>(<span class="st">&quot;Sample_cell&quot;</span>,<span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>,<span class="st">&quot;Cell&quot;</span>)) <span class="op">%&gt;%</span><span class="kw">melt</span>(<span class="st">&quot;Sample_cell&quot;</span>),
                          <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;Sample_cell&quot;</span>,<span class="st">&quot;variable&quot;</span>))
  <span class="kw">colnames</span>(AF_NGT_Melt) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Cell&quot;</span>,<span class="st">&quot;Variant&quot;</span>,<span class="st">&quot;AF&quot;</span>,<span class="st">&quot;NGT&quot;</span>)
  gg_AF_NGT&lt;-<span class="st">  </span><span class="kw">ggplot</span>(AF_NGT_Melt,<span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">as.factor</span>(NGT),<span class="dt">y=</span>AF,<span class="dt">color=</span>NGT))<span class="op">+</span><span class="st"> </span><span class="kw">geom_quasirandom</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>Variant,<span class="dt">ncol=</span><span class="dv">3</span>)
  
  DP_NGT_Melt&lt;-<span class="kw">inner_join</span>(DP <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unite</span>(<span class="st">&quot;Sample_cell&quot;</span>,<span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>,<span class="st">&quot;Cell&quot;</span>)) <span class="op">%&gt;%</span><span class="kw">melt</span>(<span class="st">&quot;Sample_cell&quot;</span>),
                          NGT <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unite</span>(<span class="st">&quot;Sample_cell&quot;</span>,<span class="kw">c</span>(<span class="st">&quot;Sample&quot;</span>,<span class="st">&quot;Cell&quot;</span>)) <span class="op">%&gt;%</span><span class="kw">melt</span>(<span class="st">&quot;Sample_cell&quot;</span>),
                          <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;Sample_cell&quot;</span>,<span class="st">&quot;variable&quot;</span>))
  <span class="kw">colnames</span>(DP_NGT_Melt) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Cell&quot;</span>,<span class="st">&quot;Variant&quot;</span>,<span class="st">&quot;DP&quot;</span>,<span class="st">&quot;NGT&quot;</span>)
  gg_DP_NGT&lt;-<span class="st"> </span><span class="kw">ggplot</span>(DP_NGT_Melt,<span class="kw">aes</span>(<span class="dt">x=</span><span class="kw">as.factor</span>(NGT),<span class="dt">y=</span>DP,<span class="dt">fill=</span>NGT))<span class="op">+</span><span class="kw">geom_boxplot</span>()<span class="op">+</span><span class="kw">facet_wrap</span>(<span class="op">~</span>Variant,<span class="dt">ncol=</span><span class="dv">3</span>,<span class="dt">scale=</span><span class="st">&quot;free_y&quot;</span>)
  
  
  <span class="kw">ggsave</span>(<span class="kw">paste0</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,sample,<span class="st">&quot;_DP_NGT.pdf&quot;</span>),gg_DP_NGT,<span class="dt">width=</span><span class="dv">10</span>,<span class="dt">height=</span><span class="dv">40</span>)
  <span class="kw">ggsave</span>(<span class="kw">paste0</span>(<span class="st">&quot;./&quot;</span>,sample,<span class="st">&quot;/&quot;</span>,sample,<span class="st">&quot;_AF_NGT.pdf&quot;</span>),gg_AF_NGT,<span class="dt">width=</span><span class="dv">10</span>,<span class="dt">height=</span><span class="dv">40</span>)
  
}


plot_COMPASS_data &lt;-<span class="cf">function</span>(sample,output_folder,melted_protein_mat,clonal_abundance,clonal_architecture,clone_cell_count,shrink)
  {
  melted_protein_mat &lt;-melted_protein_mat[[sample]]
    clonal_abundance_subset &lt;-<span class="kw">data.frame</span>( clonal_abundance[[sample]])<span class="co">#%&gt;%filter(Count&gt;=5)</span>
    clonal_architecture_subset &lt;-<span class="st"> </span><span class="kw">data.frame</span>(clonal_architecture[[sample]])<span class="co">#%&gt;%</span>
    <span class="co">#  filter(Clone%in%as.character(clonal_abundance_subset$Clone))</span>
    genes_to_display &lt;-<span class="kw">unique</span>(<span class="kw">as.character</span>(clonal_architecture_subset[clonal_architecture_subset<span class="op">$</span>Genotype<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>),<span class="st">&quot;Mutant&quot;</span>]))
    
    <span class="cf">if</span>(shrink<span class="op">==</span><span class="ot">TRUE</span>){
      clonal_architecture_subset &lt;-clonal_architecture_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(Mutant<span class="op">%in%</span>genes_to_display)
    }
    clonal_architecture_subset<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">factor</span>(clonal_architecture_subset<span class="op">$</span>Clone, <span class="dt">levels=</span><span class="kw">levels</span>(clonal_abundance_subset<span class="op">$</span>Clone))
    melted_protein_mat_subset&lt;-melted_protein_mat<span class="op">%&gt;%</span><span class="kw">filter</span>(Clone<span class="op">%in%</span><span class="kw">as.character</span>(clonal_architecture_subset<span class="op">$</span>Clone))
    melted_protein_mat_subset<span class="op">$</span>Clone &lt;-<span class="kw">factor</span>(melted_protein_mat_subset<span class="op">$</span>Clone, <span class="dt">levels=</span><span class="kw">levels</span>(clonal_abundance_subset<span class="op">$</span>Clone))
    gg_protein_heatmap&lt;-<span class="kw">ggplot</span>(melted_protein_mat_subset, <span class="kw">aes</span>(<span class="dt">y=</span>variable, <span class="dt">x=</span>(Clone))) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tile</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> value),<span class="dt">colour =</span> <span class="st">&quot;white&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">      </span><span class="kw">scale_fill_distiller</span>(<span class="dt">palette =</span> <span class="st">&quot;PRGn&quot;</span>)<span class="op">+</span>
<span class="st">      </span><span class="kw">theme_minimal</span>(<span class="dt">base_size=</span><span class="dv">6</span>) <span class="op">+</span>
<span class="st">      </span><span class="kw">theme</span>( <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), 
             <span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>,<span class="dt">legend.direction =</span> <span class="st">&quot;vertical&quot;</span>,
             <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(),
             <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),<span class="st">&quot;cm&quot;</span>))
    
     gg_heatmap &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> clonal_architecture_subset, <span class="kw">aes</span>(<span class="dt">x =</span> Clone, <span class="dt">y =</span> <span class="kw">factor</span>(Mutant,<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(Mutant)))), <span class="dt">fill =</span> Genotype)) <span class="op">+</span><span class="st"> </span>
<span class="st">      </span><span class="kw">geom_tile</span>() <span class="op">+</span><span class="co"># scale_y_discrete(limits = rev(levels(Mutant)))+</span>
<span class="st">      </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">1</span>],
                                 <span class="st">&quot;Heterozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">3</span>],
                                 <span class="st">&quot;Homozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">6</span>],
                                 <span class="st">&quot;Unknown&quot;</span>=<span class="st">&quot;grey50&quot;</span>),<span class="st">&quot;Genotype&quot;</span>)  <span class="op">+</span>
<span class="st">      </span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">6</span>) <span class="op">+</span>
<span class="st">      </span><span class="kw">ylab</span>(<span class="st">&quot;Mutation&quot;</span>)<span class="op">+</span>
<span class="st">      </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;vertical&quot;</span>,
            <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), 
            <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
            <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),
            <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(),
            <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),<span class="st">&quot;cm&quot;</span>))
    
    clone_colors &lt;-<span class="st"> </span><span class="kw">alphabet</span>()[<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">unique</span>(clonal_abundance_subset<span class="op">$</span>Clone))]
    <span class="kw">names</span>(clone_colors) &lt;-<span class="kw">levels</span>(clonal_abundance_subset<span class="op">$</span>Clone)
    <span class="co"># Generate clonal abundance barplot</span>
    gg_clonal_barplot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> <span class="kw">data.frame</span>(clonal_abundance_subset), <span class="kw">aes</span>(<span class="dt">x =</span> Clone, <span class="dt">y =</span> Count)) <span class="op">+</span><span class="st"> </span>
<span class="st">      </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill =</span> Clone)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_gray</span>() <span class="op">+</span>
<span class="st">      </span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">6</span>)<span class="op">+</span>
<span class="st">      </span><span class="kw">ylim</span>(<span class="dv">0</span>,<span class="kw">max</span>(clonal_abundance_subset<span class="op">$</span>Count)<span class="op">*</span><span class="fl">1.3</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">      </span><span class="kw">ylab</span>(<span class="st">&quot;Cell Count&quot;</span>)<span class="op">+</span>
<span class="st">      </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>Count), <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="dt">width=</span><span class="fl">0.9</span>), <span class="dt">vjust=</span><span class="op">-</span><span class="fl">0.25</span>,<span class="dt">size=</span><span class="dv">1</span>)<span class="op">+</span>
<span class="st">      </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>clone_colors)<span class="op">+</span>
<span class="st">      </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(),  <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
            <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,
            <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>),<span class="st">&quot;cm&quot;</span>))
    
    grob.title &lt;-<span class="st"> </span><span class="kw">textGrob</span>(<span class="kw">paste</span>(sample,<span class="st">&quot;-&quot;</span>), <span class="dt">hjust=</span> <span class="fl">0.5</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">12</span>))
    final_plot&lt;-<span class="kw">plot_grid</span>(gg_clonal_barplot,<span class="kw">addSmallLegend</span>(gg_heatmap),<span class="kw">addSmallLegend</span>(gg_protein_heatmap),<span class="dt">ncol=</span><span class="dv">1</span>,<span class="dt">align=</span><span class="st">&quot;hv&quot;</span>,<span class="dt">axis=</span><span class="st">&quot;lrtb&quot;</span>,<span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="dv">1</span>,(<span class="kw">length</span>(genes_to_display)<span class="op">*</span><span class="fl">0.15</span>),<span class="dv">1</span>))
    
    <span class="kw">save_plot</span>(<span class="kw">paste</span>(output_folder,sample,<span class="st">&quot;.pdf&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>),final_plot, <span class="dt">ncol=</span><span class="dv">1</span>) <span class="co"># Open a new pdf file</span>
  }

addSmallLegend &lt;-<span class="st"> </span><span class="cf">function</span>(myPlot, <span class="dt">pointSize =</span> <span class="dv">3</span>, <span class="dt">textSize =</span> <span class="dv">8</span>, <span class="dt">spaceLegend =</span> <span class="fl">0.5</span>) {
  myPlot <span class="op">+</span>
<span class="st">    </span><span class="kw">guides</span>(<span class="dt">shape =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">size =</span> pointSize)),
           <span class="dt">color =</span> <span class="kw">guide_legend</span>(<span class="dt">override.aes =</span> <span class="kw">list</span>(<span class="dt">size =</span> pointSize))) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> textSize), 
          <span class="dt">legend.text  =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> textSize),
          <span class="dt">legend.key.size =</span> <span class="kw">unit</span>(spaceLegend, <span class="st">&quot;lines&quot;</span>))
}


create_reward_matrix_deprecated&lt;-<span class="cf">function</span>(Known_mat,weights){
  <span class="kw">set.seed</span>(<span class="dv">68864</span>)
  <span class="kw">names</span>(weights) &lt;-<span class="st"> </span><span class="kw">apply</span>(Known_mat,<span class="dv">2</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)})
  num_type &lt;-<span class="st"> </span><span class="dv">2</span>
  num_mutations &lt;-<span class="st"> </span><span class="kw">nrow</span>(Known_mat); 
  num_clones &lt;-<span class="st"> </span><span class="kw">ncol</span>(Known_mat)
  num_states &lt;-<span class="st"> </span>num_type<span class="op">^</span>num_mutations
  
  states&lt;-<span class="kw">data.frame</span>(<span class="kw">expand.grid</span>(<span class="kw">rep</span>(<span class="kw">list</span>(<span class="dv">0</span><span class="op">:</span>num_type), num_mutations)))
  state_interactions&lt;-<span class="kw">expand.grid</span>(<span class="kw">apply</span>(states[,<span class="dv">1</span><span class="op">:</span>num_mutations],<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)}),
                                  <span class="kw">apply</span>(states[,<span class="dv">1</span><span class="op">:</span>num_mutations],<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)}))
  
  states<span class="op">$</span>Reward_states&lt;-<span class="kw">ifelse</span>(<span class="kw">apply</span>(states,<span class="dv">1</span>,<span class="cf">function</span>(x){
    <span class="kw">any</span>(<span class="kw">apply</span>(Known_mat,<span class="dv">2</span>,<span class="cf">function</span>(y){ <span class="kw">all</span>(x<span class="op">==</span><span class="kw">t</span>(y))}))
  }),<span class="dv">2</span>,<span class="ot">NA</span>)
  states<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">apply</span>(states[,<span class="dv">1</span><span class="op">:</span>num_mutations],<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)})
  
  state_interactions<span class="op">$</span>possible&lt;-<span class="kw">ifelse</span>(<span class="kw">apply</span>(state_interactions,<span class="dv">1</span>,<span class="cf">function</span>(x){
    A&lt;-<span class="kw">as.numeric</span>(<span class="kw">do.call</span>(cbind,<span class="kw">strsplit</span>(<span class="kw">as.character</span>(x[<span class="dv">1</span>]),<span class="dt">split=</span><span class="st">&quot;_&quot;</span>)))
    B&lt;-<span class="kw">as.numeric</span>(<span class="kw">do.call</span>(cbind,<span class="kw">strsplit</span>(<span class="kw">as.character</span>(x[<span class="dv">2</span>]),<span class="dt">split=</span><span class="st">&quot;_&quot;</span>)))
    <span class="kw">sum</span>(<span class="kw">abs</span>(A<span class="op">-</span>B))<span class="op">&lt;=</span><span class="dv">1</span>
  }),<span class="dv">2</span>,<span class="ot">NA</span>)
  
  dat&lt;-<span class="kw">acast</span>(Var1<span class="op">~</span>Var2,<span class="dt">value.var=</span><span class="st">&quot;possible&quot;</span>,<span class="dt">data=</span><span class="kw">data.frame</span>(state_interactions,
                                                            <span class="st">&quot;value&quot;</span>=<span class="op">-</span><span class="dv">1</span>))
  <span class="kw">diag</span>(dat)&lt;-<span class="dv">0</span>
  <span class="kw">lowerTriangle</span>(dat)&lt;-<span class="ot">NA</span>
  
  test_dat&lt;-<span class="kw">do.call</span>(cbind,<span class="kw">lapply</span>(<span class="kw">colnames</span>(dat),<span class="cf">function</span>(Clone){
    <span class="cf">if</span>(Clone<span class="op">%in%</span><span class="kw">names</span>(weights)){
      output&lt;-<span class="kw">ifelse</span>(dat[,Clone]<span class="op">==</span><span class="dv">2</span>,weights[Clone],dat[,Clone])
    } <span class="cf">else</span>{
      output&lt;-dat[,Clone]
    }
    <span class="kw">return</span>(output)
  }))
  <span class="kw">colnames</span>(test_dat) &lt;-<span class="kw">rownames</span>(test_dat)
  graph&lt;-<span class="kw">graph.adjacency</span>(test_dat,<span class="dt">mode=</span><span class="st">&quot;directed&quot;</span>,<span class="dt">weighted=</span><span class="ot">TRUE</span>)
  graph_mat &lt;-<span class="st"> </span><span class="kw">get.data.frame</span>(graph)<span class="op">%&gt;%</span><span class="st"> </span><span class="kw">drop_na</span>()
  
  subgraphs&lt;-<span class="kw">make_ego_graph</span>(<span class="kw">graph_from_data_frame</span>(graph_mat,<span class="dt">directed=</span><span class="ot">TRUE</span>), <span class="dt">order=</span><span class="dv">1</span>,<span class="dt">nodes=</span><span class="kw">names</span>(weights), <span class="dt">mode=</span><span class="st">&quot;all&quot;</span>)
  subgraph_bind&lt;-<span class="kw">do.call</span>(rbind,<span class="kw">lapply</span>(subgraphs,get.data.frame)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(to,from,weight, <span class="dt">.keep_all =</span> <span class="ot">TRUE</span>)
  subgraph_subset&lt;-subgraph_bind<span class="op">%&gt;%</span><span class="kw">filter</span>(<span class="op">!</span>to<span class="op">%in%</span><span class="kw">setdiff</span>(<span class="kw">setdiff</span>(subgraph_bind<span class="op">$</span>to,subgraph_bind<span class="op">$</span>from),<span class="kw">names</span>(weights)))
  <span class="kw">return</span>(subgraph_subset)
}


create_reward_matrix&lt;-<span class="cf">function</span>(Known_mat,weights){
  
  <span class="kw">set.seed</span>(<span class="dv">68864</span>)
  <span class="kw">names</span>(weights) &lt;-<span class="st"> </span><span class="kw">apply</span>(Known_mat,<span class="dv">2</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)})
  num_type &lt;-<span class="st"> </span><span class="dv">2</span>
  num_mutations &lt;-<span class="st"> </span><span class="kw">nrow</span>(Known_mat); 
  mutant_names&lt;-<span class="kw">rownames</span>(Known_mat)
  num_clones &lt;-<span class="st"> </span><span class="kw">ncol</span>(Known_mat)
  num_states &lt;-<span class="st"> </span>num_type<span class="op">^</span>num_mutations
  
  possible_mut_list&lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">apply</span>(Known_mat,<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">list</span>(<span class="dv">0</span><span class="op">:</span><span class="kw">max</span>(<span class="kw">unique</span>(x))) }),<span class="dt">recursive =</span> <span class="ot">FALSE</span>)
  
  states&lt;-<span class="kw">data.frame</span>(<span class="kw">expand.grid</span>(possible_mut_list))
  state_interactions&lt;-<span class="kw">data.frame</span>(<span class="kw">expand.grid</span>(<span class="kw">apply</span>(states[,<span class="dv">1</span><span class="op">:</span>num_mutations],<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)}),
                                             <span class="kw">apply</span>(states[,<span class="dv">1</span><span class="op">:</span>num_mutations],<span class="dv">1</span>,<span class="cf">function</span>(x){<span class="kw">paste</span>(x,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)})))
  
  state_interactions<span class="op">$</span>possible&lt;-<span class="kw">ifelse</span>(<span class="kw">apply</span>(state_interactions,<span class="dv">1</span>,<span class="cf">function</span>(x){
    A&lt;-<span class="kw">as.numeric</span>(<span class="kw">do.call</span>(cbind,<span class="kw">strsplit</span>(<span class="kw">as.character</span>(x[<span class="dv">1</span>]),<span class="dt">split=</span><span class="st">&quot;_&quot;</span>)))
    B&lt;-<span class="kw">as.numeric</span>(<span class="kw">do.call</span>(cbind,<span class="kw">strsplit</span>(<span class="kw">as.character</span>(x[<span class="dv">2</span>]),<span class="dt">split=</span><span class="st">&quot;_&quot;</span>)))
    <span class="kw">sum</span>(<span class="kw">abs</span>(A<span class="op">-</span>B))<span class="op">&lt;=</span><span class="dv">1</span>
  }),<span class="dv">0</span>,<span class="ot">NA</span>)
  
  state_interactions<span class="op">$</span>action&lt;-<span class="kw">apply</span>(state_interactions,<span class="dv">1</span>,<span class="cf">function</span>(x){
    A&lt;-<span class="kw">as.numeric</span>(<span class="kw">do.call</span>(cbind,<span class="kw">strsplit</span>(<span class="kw">as.character</span>(x[<span class="dv">1</span>]),<span class="dt">split=</span><span class="st">&quot;_&quot;</span>)))
    B&lt;-<span class="kw">as.numeric</span>(<span class="kw">do.call</span>(cbind,<span class="kw">strsplit</span>(<span class="kw">as.character</span>(x[<span class="dv">2</span>]),<span class="dt">split=</span><span class="st">&quot;_&quot;</span>)))
    <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.na</span>(x[<span class="st">&quot;possible&quot;</span>])){
      <span class="cf">if</span>(<span class="kw">sum</span>(<span class="kw">abs</span>(B<span class="op">-</span>A))<span class="op">==</span><span class="dv">0</span>){
        <span class="kw">return</span>(<span class="st">&quot;stay&quot;</span>)
      } <span class="cf">else</span>{
        <span class="kw">return</span>(mutant_names[<span class="kw">which</span>((B<span class="op">-</span>A)<span class="op">==</span><span class="dv">1</span>)])
      }
    }
  })
  
  dat&lt;-<span class="kw">setNames</span>(state_interactions<span class="op">%&gt;%</span><span class="kw">filter</span>(action<span class="op">%in%</span><span class="kw">c</span>(mutant_names,<span class="st">&quot;stay&quot;</span>)),
                <span class="kw">c</span>(<span class="st">&quot;State&quot;</span>,<span class="st">&quot;NextState&quot;</span>,<span class="st">&quot;Reward&quot;</span>,<span class="st">&quot;Action&quot;</span>))[,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">3</span>)]
  
  dat<span class="op">$</span>Reward &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">apply</span>(dat,<span class="dv">1</span>,<span class="cf">function</span>(x){
    <span class="kw">ifelse</span>(x<span class="op">$</span>NextState<span class="op">%in%</span><span class="kw">names</span>(weights),weights[x<span class="op">$</span>NextState],x<span class="op">$</span>Reward)
  }))
  dat<span class="op">$</span>Reward &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">apply</span>(dat,<span class="dv">1</span>,<span class="cf">function</span>(x){
    <span class="kw">ifelse</span>(x<span class="op">$</span>Action<span class="op">%in%</span><span class="st">&quot;stay&quot;</span>,<span class="dv">0</span>,x<span class="op">$</span>Reward)
  }))
  dat<span class="op">$</span>State &lt;-<span class="st"> </span><span class="kw">as.character</span>(dat<span class="op">$</span>State)
  dat<span class="op">$</span>NextState &lt;-<span class="st"> </span><span class="kw">as.character</span>(dat<span class="op">$</span>NextState)
  dat<span class="op">$</span>Action &lt;-<span class="st"> </span><span class="kw">as.character</span>(dat<span class="op">$</span>Action)
  
  control &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">alpha =</span> <span class="fl">0.8</span>, <span class="dt">gamma =</span> <span class="fl">0.9</span>)
  model &lt;-<span class="st"> </span><span class="kw">ReinforcementLearning</span>(<span class="dt">data =</span> dat, <span class="dt">s =</span> <span class="st">&quot;State&quot;</span>, <span class="dt">a =</span> <span class="st">&quot;Action&quot;</span>, <span class="dt">r =</span> <span class="st">&quot;Reward&quot;</span>,  <span class="dt">s_new =</span> <span class="st">&quot;NextState&quot;</span>,  <span class="dt">iter =</span>  <span class="dv">1</span>,<span class="dt">control=</span>control)
  x&lt;-<span class="st"> </span>model<span class="op">$</span>Q
  <span class="kw">rownames</span>(x) &lt;-<span class="st"> </span><span class="kw">substring</span>(<span class="kw">rownames</span>(x),<span class="dv">2</span>)
  Q_mat &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">melt</span>(x),<span class="kw">c</span>(<span class="st">&quot;State&quot;</span>,<span class="st">&quot;Action&quot;</span>,<span class="st">&quot;Q&quot;</span>))
  set&lt;-<span class="kw">inner_join</span>(dat,Q_mat,<span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;State&quot;</span>,<span class="st">&quot;Action&quot;</span>))
  set<span class="op">$</span>Valid &lt;-<span class="st"> </span><span class="ot">TRUE</span>
  <span class="kw">return</span>(set)
  }


plot_clonal_heatmap_and_barplot_with_SD &lt;-<span class="cf">function</span>(sample,output_folder,clonal_abundance,clonal_architecture,clone_cell_count,shrink)
{
  clonal_abundance_subset &lt;-<span class="kw">data.frame</span>( clonal_abundance[[sample]])<span class="co">#%&gt;%filter(Count&gt;=5)</span>
  clones_to_use &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">as.character</span>(clonal_abundance_subset<span class="op">$</span>Clone),<span class="kw">as.character</span>(clonal_architecture[[sample]]<span class="op">$</span>Clone))
  clonal_architecture_subset &lt;-<span class="st"> </span><span class="kw">data.frame</span>(clonal_architecture[[sample]])<span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(<span class="kw">data.frame</span>(clonal_architecture[[sample]])<span class="op">$</span>Clone<span class="op">%in%</span>clones_to_use)
  clonal_abundance_subset &lt;-<span class="st"> </span><span class="kw">data.frame</span>(clonal_abundance_subset)<span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(Clone<span class="op">%in%</span>clones_to_use)
  genes_to_display &lt;-<span class="kw">unique</span>(<span class="kw">as.character</span>(clonal_architecture_subset[clonal_architecture_subset<span class="op">$</span>Genotype<span class="op">%in%</span><span class="kw">c</span>(<span class="st">&quot;Heterozygous&quot;</span>,<span class="st">&quot;Homozygous&quot;</span>),<span class="st">&quot;Mutant&quot;</span>]))
  
  <span class="cf">if</span>(shrink<span class="op">==</span><span class="ot">TRUE</span>){
    clonal_architecture_subset &lt;-clonal_architecture_subset<span class="op">%&gt;%</span><span class="kw">filter</span>(Mutant<span class="op">%in%</span>genes_to_display)
  }
  clonal_architecture_subset<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">factor</span>(clonal_architecture_subset<span class="op">$</span>Clone, <span class="dt">levels=</span><span class="kw">rev</span>(clonal_abundance_subset<span class="op">$</span>Clone))
  clonal_abundance_subset<span class="op">$</span>Clone &lt;-<span class="st"> </span><span class="kw">factor</span>(clonal_abundance_subset<span class="op">$</span>Clone, <span class="dt">levels=</span><span class="kw">levels</span>(clonal_architecture_subset<span class="op">$</span>Clone))
  gg_heatmap &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> clonal_architecture_subset, <span class="kw">aes</span>(<span class="dt">x =</span> Clone, <span class="dt">y =</span> <span class="kw">factor</span>(Mutant,<span class="dt">levels=</span><span class="kw">rev</span>(<span class="kw">levels</span>(<span class="kw">factor</span>(Mutant)))), <span class="dt">fill =</span> Genotype)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span><span class="co"># scale_y_discrete(limits = rev(levels(Mutant)))+</span>
<span class="st">    </span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;WT&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">1</span>],
                               <span class="st">&quot;Heterozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">3</span>],
                               <span class="st">&quot;Homozygous&quot;</span>=<span class="kw">brewer.pal</span>(<span class="dv">7</span>,<span class="st">&quot;Reds&quot;</span>)[<span class="dv">6</span>],
                               <span class="st">&quot;Unknown&quot;</span>=<span class="st">&quot;grey50&quot;</span>),<span class="st">&quot;Genotype&quot;</span>)  <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">6</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Mutation&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>, <span class="dt">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>,
          <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(), 
          <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
          <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>(),
          <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(),
          <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>),<span class="st">&quot;cm&quot;</span>))
  
  <span class="co"># Generate clonal abundance barplot</span>
  gg_clonal_barplot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> <span class="kw">data.frame</span>(clonal_abundance_subset), <span class="kw">aes</span>(<span class="dt">x =</span> Clone, <span class="dt">y =</span> Count)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="kw">aes</span>(<span class="dt">fill =</span> Count)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_gray</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme_classic</span>(<span class="dt">base_size=</span><span class="dv">6</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">ylim</span>(<span class="dv">0</span>,<span class="kw">max</span>(clonal_abundance_subset<span class="op">$</span>Count)<span class="op">*</span><span class="fl">1.3</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;Cell Count&quot;</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymin =</span> X2.<span class="dv">5</span>., <span class="dt">ymax =</span> X97.<span class="dv">5</span>.), <span class="dt">width =</span> <span class="fl">0.2</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>Count), <span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="dt">width=</span><span class="fl">0.9</span>), <span class="dt">vjust=</span><span class="op">-</span><span class="fl">0.25</span>,<span class="dt">size=</span><span class="dv">1</span>)<span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_distiller</span>(<span class="dt">name =</span> <span class="st">&quot;Value&quot;</span>, <span class="dt">palette =</span> <span class="st">&quot;Reds&quot;</span>, <span class="dt">direction =</span> <span class="dv">1</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>(),  <span class="dt">axis.ticks.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.line=</span><span class="kw">element_blank</span>(),
          <span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,
          <span class="dt">plot.margin=</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="op">-</span><span class="fl">0.2</span>,<span class="dv">1</span>),<span class="st">&quot;cm&quot;</span>))
  
  grob.title &lt;-<span class="st"> </span><span class="kw">textGrob</span>(<span class="kw">paste</span>(sample,<span class="st">&quot;-&quot;</span>), <span class="dt">hjust=</span> <span class="fl">0.5</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">12</span>))
  final_plot&lt;-<span class="kw">plot_grid</span>(grob.title,gg_clonal_barplot,gg_heatmap,<span class="dt">ncol=</span><span class="dv">1</span>,<span class="dt">align=</span><span class="st">&quot;hv&quot;</span>,<span class="dt">axis=</span><span class="st">&quot;l&quot;</span>,<span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.1</span>,<span class="dv">1</span>,<span class="fl">0.75</span>))
  
  <span class="kw">save_plot</span>(<span class="kw">paste</span>(output_folder,sample,<span class="st">&quot;.pdf&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>),final_plot, <span class="dt">ncol=</span><span class="dv">1</span>) <span class="co"># Open a new pdf file</span>
}

query_initiating_mutations&lt;-<span class="cf">function</span>(graph_results){
  start_index&lt;-<span class="kw">paste</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="kw">length</span>(<span class="kw">strsplit</span>(graph_results<span class="op">$</span>State[<span class="dv">1</span>],<span class="dt">split=</span><span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]])),<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>,<span class="dt">collapse=</span><span class="st">&quot;_&quot;</span>)
  possible_starting_actions&lt;-graph_results<span class="op">%&gt;%</span><span class="kw">filter</span>(State<span class="op">==</span>start_index<span class="op">&amp;</span>Action<span class="op">!=</span><span class="st">&quot;stay&quot;</span>)<span class="op">%&gt;%</span><span class="kw">pull</span>(Action)
  final_results&lt;-<span class="kw">list</span>()
  initating_action_count&lt;-<span class="dv">0</span>
  <span class="cf">for</span>(initating_action <span class="cf">in</span> possible_starting_actions){
    <span class="kw">print</span>(initating_action)
    set &lt;-<span class="st"> </span>graph_results
    initating_action_count&lt;-initating_action_count<span class="op">+</span><span class="dv">1</span>
    storage_results&lt;-<span class="st"> </span><span class="kw">list</span>()
    branches&lt;-<span class="dv">0</span>
    state_to_kill &lt;-<span class="st"> </span>set<span class="op">%&gt;%</span><span class="kw">filter</span>(State<span class="op">==</span>start_index<span class="op">&amp;</span>Action<span class="op">==</span>initating_action)<span class="op">%&gt;%</span><span class="kw">pull</span>(NextState)
    start_killed &lt;-<span class="st"> </span><span class="kw">sum</span>(set<span class="op">%&gt;%</span><span class="kw">filter</span>(State<span class="op">==</span>state_to_kill)<span class="op">%&gt;%</span><span class="kw">pull</span>(Valid))
    <span class="cf">while</span>(start_killed<span class="op">&gt;</span><span class="dv">0</span>){
      <span class="co">#print(branches)</span>
     <span class="co"># print(start_killed)</span>
      branches &lt;-<span class="st"> </span>branches <span class="op">+</span><span class="dv">1</span>
      number_of_mutations&lt;-<span class="dv">0</span>
      state_log&lt;-<span class="st"> </span><span class="kw">list</span>()
      optimal_reward&lt;-<span class="kw">list</span>()
      action_log&lt;-<span class="kw">list</span>()
      current_state&lt;-<span class="st"> </span>start_index
      indicator&lt;-<span class="ot">TRUE</span>
      nextState&lt;-<span class="dv">0</span>
      <span class="cf">while</span>(current_state<span class="op">!=</span>nextState)  {
       <span class="co"># print(number_of_mutations)</span>
        number_of_mutations &lt;-<span class="st"> </span>number_of_mutations<span class="op">+</span><span class="dv">1</span>
        <span class="cf">if</span>(number_of_mutations<span class="op">==</span><span class="dv">1</span>){
          state_log[[number_of_mutations]] &lt;-<span class="st"> </span>start_index
        }
        current_state  &lt;-<span class="st"> </span>state_log[[number_of_mutations]]
        nextState_indicator&lt;-<span class="st"> </span><span class="ot">FALSE</span>
        
        <span class="cf">while</span>(nextState_indicator<span class="op">==</span><span class="ot">FALSE</span>){
          
          <span class="cf">if</span>(number_of_mutations<span class="op">==</span><span class="dv">1</span>){
            max_potential_action_index&lt;-<span class="st">  </span>set<span class="op">%&gt;%</span>
<span class="st">              </span><span class="kw">filter</span>(State<span class="op">==</span>current_state<span class="op">&amp;</span>Action<span class="op">==</span>initating_action)
          } <span class="cf">else</span> {
            max_potential_action_index &lt;-<span class="st"> </span>set<span class="op">%&gt;%</span>
<span class="st">              </span><span class="kw">filter</span>(State<span class="op">==</span>current_state<span class="op">&amp;</span>Valid<span class="op">==</span><span class="ot">TRUE</span>)<span class="op">%&gt;%</span>
<span class="st">              </span><span class="kw">filter</span>(Q<span class="op">==</span><span class="kw">max</span>(Q))<span class="op">%&gt;%</span><span class="kw">sample_n</span>(<span class="dv">1</span>)
          }
          <span class="cf">if</span>(<span class="kw">nrow</span>(max_potential_action_index)<span class="op">==</span><span class="dv">0</span>){
            <span class="cf">break</span>
          }
          max_potential_action &lt;-<span class="st"> </span>max_potential_action_index<span class="op">%&gt;%</span><span class="kw">pull</span>(NextState)
          next_valid_action &lt;-<span class="st"> </span><span class="kw">any</span>(set<span class="op">%&gt;%</span><span class="kw">filter</span>(State<span class="op">==</span>max_potential_action<span class="op">&amp;</span>Action<span class="op">!=</span><span class="st">&quot;stay&quot;</span>)<span class="op">%&gt;%</span><span class="kw">pull</span>(Valid))  
          <span class="cf">if</span>(next_valid_action<span class="op">==</span><span class="ot">TRUE</span>){
            nextState &lt;-max_potential_action
            current_action &lt;-<span class="st">  </span>max_potential_action_index<span class="op">%&gt;%</span><span class="kw">pull</span>(Action)
            nextState_indicator<span class="op">==</span><span class="ot">TRUE</span>
            <span class="cf">break</span>
          } <span class="cf">else</span>{
            set[set<span class="op">$</span>State<span class="op">%in%</span>max_potential_action_index[<span class="st">&quot;State&quot;</span>]<span class="op">&amp;</span>
<span class="st">                  </span>set<span class="op">$</span>Action<span class="op">%in%</span>max_potential_action_index[<span class="st">&quot;Action&quot;</span>],<span class="st">&quot;Valid&quot;</span>] &lt;-<span class="st"> </span><span class="ot">FALSE</span>  
          }
        }
        <span class="cf">if</span>(<span class="kw">nrow</span>(set<span class="op">%&gt;%</span><span class="kw">filter</span>(State<span class="op">==</span>current_state<span class="op">&amp;</span>Action<span class="op">==</span>current_action))<span class="op">==</span><span class="dv">0</span>){
          optimal_reward[[number_of_mutations]] &lt;-<span class="ot">NA</span>
        } <span class="cf">else</span> {
          optimal_reward[[number_of_mutations]] &lt;-<span class="st"> </span>set<span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">filter</span>(State<span class="op">==</span>current_state<span class="op">&amp;</span>Action<span class="op">==</span>current_action)<span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">pull</span>(Reward) 
        }
        state_log[[number_of_mutations<span class="op">+</span><span class="dv">1</span>]]&lt;-<span class="st"> </span>nextState
        action_log[[number_of_mutations]] &lt;-<span class="st"> </span>current_action 
        <span class="cf">if</span>(current_action<span class="op">==</span>nextState){
          indicator<span class="op">==</span><span class="ot">FALSE</span>
          state_log[[number_of_mutations<span class="op">+</span><span class="dv">1</span>]]&lt;-<span class="ot">NULL</span>
          <span class="cf">break</span>
        }
      }
      optimal_reward[[number_of_mutations<span class="op">+</span><span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="ot">NA</span>
      action_log[[number_of_mutations<span class="op">+</span><span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="ot">NA</span>
      storage_results[[branches]] &lt;-<span class="kw">data.frame</span>(<span class="st">&quot;states&quot;</span>=<span class="kw">do.call</span>(rbind,state_log),<span class="co">#[1:(length(state_log)-1)]),</span>
                                               <span class="st">&quot;actions&quot;</span>=<span class="kw">do.call</span>(rbind,action_log),
                                               <span class="st">&quot;reward&quot;</span>=<span class="kw">do.call</span>(rbind,optimal_reward),
                                               <span class="st">&quot;nextState&quot;</span>=<span class="kw">do.call</span>(rbind,<span class="kw">c</span>(state_log[<span class="dv">2</span><span class="op">:</span><span class="kw">length</span>(state_log)],<span class="ot">NA</span>)) )
      storage_results[[branches]] &lt;-<span class="st"> </span>storage_results[[branches]]<span class="op">%&gt;%</span>
<span class="st">        </span><span class="kw">filter</span>(states<span class="op">!=</span>nextState)
      storage_results[[branches]]<span class="op">$</span>cumulative_reward &lt;-<span class="st"> </span><span class="kw">cumsum</span>(storage_results[[branches]]<span class="op">$</span>reward)
      
      <span class="co">#storage_results[[branches]] &lt;-storage_results[[branches]][1:which.max(storage_results[[branches]]$cumulative_reward), ]</span>
      set[set<span class="op">$</span>State<span class="op">%in%</span>current_state<span class="op">&amp;</span>set<span class="op">$</span>Action<span class="op">%in%</span>current_action,<span class="st">&quot;Valid&quot;</span>] &lt;-<span class="st"> </span><span class="ot">FALSE</span>
      start_killed &lt;-<span class="st"> </span><span class="kw">sum</span>(set<span class="op">%&gt;%</span><span class="kw">filter</span>(State<span class="op">==</span>state_to_kill)<span class="op">%&gt;%</span><span class="kw">pull</span>(Valid))
    }
    final_results[[initating_action_count]]&lt;-storage_results[<span class="op">!</span><span class="kw">duplicated</span>(storage_results)]
  }
  <span class="kw">names</span>(final_results)&lt;-possible_starting_actions
  <span class="kw">return</span>(final_results)
}</code></pre></div>

</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="AppendixA.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scDNA.pdf", "scDNA.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
